--- a/config.mk.in	2009-02-20 13:00:04.000000000 +0300
+++ b/config.mk.in	2009-02-24 15:26:52.000000000 +0300
@@ -173,7 +173,7 @@
   shlibfn = $(1:%=%.$(SHLIB_SUFFIX).$(MAJ).$(MIN))
   shliblefn = $(1:%=%.$(SHLIB_SUFFIX))
 #  SHLIB_CLIB = -lc
-  LDFLAGS_SHLIB = -shared -Wl,-soname,$(SONAME) $(SHLIB_CLIB)
+  LDFLAGS_SHLIB = -shared -Wl,-soname,$(SONAME) $(SHLIB_CLIB) $(LDFLAGS)
   CFLAGS_SHLIB=-fPIC
 endif
 
