Index: xine-ui-0.99.4/src/xitk/xine-toolkit/xitk.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/xine-toolkit/xitk.c	2007-12-23 01:26:05.991112183 +0100
+++ xine-ui-0.99.4/src/xitk/xine-toolkit/xitk.c	2007-12-23 01:26:10.819387329 +0100
@@ -45,11 +45,6 @@
 #include <sys/time.h>
 
 #include <locale.h>
-
-#ifdef __linux__
-#include <execinfo.h>
-#endif
-
 #include <X11/Xlib.h>
 #include <X11/Xatom.h>
 #include <X11/keysym.h>
