Index: pulseaudio-0.9.13/src/daemon/main.c
===================================================================
--- pulseaudio-0.9.13.orig/src/daemon/main.c	2008-10-03 21:16:52.000000000 +0200
+++ pulseaudio-0.9.13/src/daemon/main.c	2008-10-10 09:56:38.000000000 +0200
@@ -892,7 +892,7 @@ int main(int argc, char *argv[]) {
     win32_timer = pa_mainloop_get_api(mainloop)->time_new(pa_mainloop_get_api(mainloop), pa_gettimeofday(&win32_tv), message_cb, NULL);
 #endif
 
-    oil_init();
+    oil_init_no_optimize();
 
     if (!conf->no_cpu_limit)
         pa_assert_se(pa_cpu_limit_init(pa_mainloop_get_api(mainloop)) == 0);
