diff -Nru asterisk-1.2.14.org/codecs/lpc10/Makefile asterisk-1.2.14/codecs/lpc10/Makefile
--- asterisk-1.2.14.org/codecs/lpc10/Makefile	2005-11-29 19:24:39.000000000 +0100
+++ asterisk-1.2.14/codecs/lpc10/Makefile	2006-12-27 08:54:16.000000000 +0100
@@ -34,6 +34,7 @@
 ifneq ($(PROC),ppc)
 ifneq ($(PROC),x86_64)
 ifneq ($(PROC),alpha)
+ifneq ($(PROC),mipsel)
 #The problem with sparc is the best stuff is in newer versions of gcc (post 3.0) only.
 #This works for even old (2.96) versions of gcc and provides a small boost either way.
 #A ultrasparc cpu is really v9 but the stock debian stable 3.0 gcc doesn.t support it.
@@ -53,6 +54,7 @@
 endif
 endif
 endif
+endif
 
 LIB = $(LIB_TARGET_DIR)/liblpc10.a
 
@@ -69,7 +71,7 @@
 
 $(LIB): $(OBJ)
 	$(AR) cr $@ $(OBJ)
-	ranlib $@
+	$(RANLIB) $@
 
 clean:
 	-rm -f *.o $(LIB)
