diff -Nru asterisk-1.2.14.org/channels/chan_iax2.c asterisk-1.2.14/channels/chan_iax2.c
--- asterisk-1.2.14.org/channels/chan_iax2.c	2006-12-09 16:45:37.000000000 +0100
+++ asterisk-1.2.14/channels/chan_iax2.c	2006-12-27 08:46:38.000000000 +0100
@@ -1191,7 +1191,7 @@
 		last++;
 	else
 		last = s;
-	snprintf(s2, strlen(s) + 100, "/var/tmp/%s-%ld", last, (unsigned long)rand());
+	snprintf(s2, strlen(s) + 100, "/tmp/%s-%ld", last, (unsigned long)rand());
 	res = stat(s, &stbuf);
 	if (res < 0) {
 		ast_log(LOG_WARNING, "Failed to stat '%s': %s\n", s, strerror(errno));
