diff -Nru asterisk-1.2.14.org/pbx/Makefile asterisk-1.2.14/pbx/Makefile
--- asterisk-1.2.14.org/pbx/Makefile	2005-11-29 19:24:39.000000000 +0100
+++ asterisk-1.2.14/pbx/Makefile	2006-12-27 09:17:58.000000000 +0100
@@ -38,6 +38,9 @@
 
 KDE_CONSOLE_OBJS=pbx_kdeconsole_main.o pbx_kdeconsole.o 
 
+CFLAGS += $(EXTRA_CFLAGS)
+PBX_LIBS += $(EXTRA_PBX_MODULES)
+
 all: depend $(PBX_LIBS)
 
 clean:
@@ -59,7 +62,7 @@
 	$(CC) $(SOLINK) -o $@ $(KDE_CONSOLE_OBJS) $(KDE_LIBS)
 
 pbx_dundi.so: dundi-parser.o pbx_dundi.o
-	$(CC) $(SOLINK) -o $@ ${CYGSOLINK} pbx_dundi.o dundi-parser.o -lz ${CYGSOLIB}
+	$(CC) $(SOLINK) -o $@ ${CYGSOLINK} pbx_dundi.o dundi-parser.o $(EXTRA_LDFLAGS) -lz ${CYGSOLIB}
 
 %.moc : %.h
 	$(MOC) $< -o $@
