diff -Nru asterisk-1.2.14.org/codecs/gsm/Makefile asterisk-1.2.14/codecs/gsm/Makefile
--- asterisk-1.2.14.org/codecs/gsm/Makefile	2006-07-24 19:05:56.000000000 +0200
+++ asterisk-1.2.14/codecs/gsm/Makefile	2006-12-27 09:14:51.000000000 +0100
@@ -241,6 +241,8 @@
 ifneq ($(shell uname -m),armv4l)
 ifneq ($(shell uname -m),sparc64)
 ifneq (${PROC},arm)
+ifneq (${PROC},mipsel)
+ifneq (${PROC},mips)
 ifneq ($(shell uname -m), parisc)
 ifneq ($(shell uname -m),s390)
 GSM_SOURCES+= $(SRC)/k6opt.s
@@ -255,6 +257,8 @@
 endif
 endif
 endif
+endif
+endif
 
 TOAST_SOURCES = $(SRC)/toast.c 		\
 		$(SRC)/toast_lin.c	\
@@ -308,6 +312,8 @@
 ifneq ($(shell uname -m), alpha)
 ifneq ($(shell uname -m), sparc64)
 ifneq ($(shell uname -m), armv4l)
+ifneq (${PROC}, mipsel)
+ifneq (${PROC}, mips)
 ifneq ($(shell uname -m), parisc)
 ifneq ($(shell uname -m),s390)
 GSM_OBJECTS+= $(SRC)/k6opt.o
@@ -321,6 +327,8 @@
 endif
 endif
 endif
+endif
+endif
 
 TOAST_OBJECTS =	$(SRC)/toast.o 		\
 		$(SRC)/toast_lin.o	\
