diff -Nru asterisk-1.2.14.org/codecs/ilbc/Makefile asterisk-1.2.14/codecs/ilbc/Makefile
--- asterisk-1.2.14.org/codecs/ilbc/Makefile	2005-11-29 19:24:39.000000000 +0100
+++ asterisk-1.2.14/codecs/ilbc/Makefile	2006-12-27 09:21:07.000000000 +0100
@@ -1,5 +1,5 @@
 ARCH=$(PROC)
-CFLAGS+=-Wall -O3 -funroll-loops
+CFLAGS+=-Wall $(OPTIMIZE) -funroll-loops
 ifneq (${OSARCH},CYGWIN)
 CFLAGS += -fPIC
 endif
@@ -15,8 +15,8 @@
 
 
 $(LIB): $(OBJS)
-	ar cr $(LIB) $(OBJS)
-	ranlib $(LIB)
+	$(AR) cr $(LIB) $(OBJS)
+	$(RANLIB) $(LIB)
 
 clean:
 	rm -f $(LIB) *.o
