diff -Nru asterisk-1.2.14.org/channels/h323/Makefile asterisk-1.2.14/channels/h323/Makefile
--- asterisk-1.2.14.org/channels/h323/Makefile	2005-11-29 19:24:39.000000000 +0100
+++ asterisk-1.2.14/channels/h323/Makefile	2006-12-27 09:22:09.000000000 +0100
@@ -30,7 +30,7 @@
 	touch $(SOURCES)
 
 libchanh323.a:	$(OBJS)
-	ar crv $@ $(OBJS)
+	$(AR) crv $@ $(OBJS)
 
 Makefile.ast:	FORCE
 	@echo H323CFLAGS  = $(STDCCFLAGS) $(OPTCCFLAGS) $(CFLAGS) >$@.tmp
diff -Nru asterisk-1.2.14.org/channels/Makefile asterisk-1.2.14/channels/Makefile
--- asterisk-1.2.14.org/channels/Makefile	2006-08-17 23:57:19.000000000 +0200
+++ asterisk-1.2.14/channels/Makefile	2006-12-27 09:22:09.000000000 +0100
@@ -15,6 +15,7 @@
 #
 
 CHANNEL_LIBS=chan_sip.so chan_agent.so chan_mgcp.so chan_iax2.so chan_local.so chan_skinny.so chan_features.so
+CXXLIBS=-lstdc++
 
 ifneq (${OSARCH},CYGWIN)
 # if you really, really want to use these drivers, uncomment the line below
@@ -228,7 +229,7 @@
 
 ifeq (${OSARCH},Linux)
 chan_h323.so: chan_h323.o h323/libchanh323.a h323/Makefile.ast
-	$(CC) $(SOLINK) $(H323LDFLAGS) -o $@ $< h323/libchanh323.a $(H323LDLIBS) -lstdc++
+	$(CC) $(SOLINK) $(H323LDFLAGS) -o $@ $< h323/libchanh323.a $(H323LDLIBS) $(CXXLIBS)
 else
 chan_h323.so: chan_h323.o h323/libchanh323.a
 	$(CC) $(SOLINK) -o $@ $< h323/libchanh323.a $(CHANH323LIB) -L$(PWLIBDIR)/lib $(PTLIB) -L$(OPENH323DIR)/lib $(H323LIB) -L/usr/lib -lcrypto -lssl -lexpat
