--- libX11-1.1.5/src/util/Makefile.in.orig	2008-10-05 00:47:56.000000000 +0200
+++ libX11-1.1.5/src/util/Makefile.in	2008-10-05 01:45:07.000000000 +0200
@@ -56,11 +56,11 @@
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/src -I$(top_builddir)/include/X11
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
-COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
+COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) makekeys \
 	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
 LTCOMPILE = $(LIBTOOL) --tag=CC $(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) \
 	--mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
-	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+	$(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) makekeys
 CCLD = $(CC)
 SOURCES = makekeys.c
 DIST_SOURCES = makekeys.c
@@ -80,9 +80,9 @@
 AWK = @AWK@
 BIGFONT_CFLAGS = @BIGFONT_CFLAGS@
 BIGFONT_LIBS = @BIGFONT_LIBS@
-CC = @CC@
+CC = $(HOSTCC)
 CCDEPMODE = @CCDEPMODE@
-CFLAGS = @CFLAGS@
+CFLAGS = 
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CXX = @CXX@
@@ -114,7 +114,7 @@
 INSTALL_STRIP_PROGRAM = @INSTALL_STRIP_PROGRAM@
 KEYSYMDEF = @KEYSYMDEF@
 LAUNCHD = @LAUNCHD@
-LDFLAGS = @LDFLAGS@
+LDFLAGS = 
 LIBOBJS = @LIBOBJS@
 LIBS = @LIBS@
 LIBTOOL = @LIBTOOL@
@@ -230,7 +230,6 @@
 makekeys_CFLAGS = $(X11_CFLAGS)
 
 #override CC = gcc
-LINK = $(CC) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 EXTRA_DIST = mkks.sh
 
 # Check source code with tools like lint & sparse
