diff -Naur php-5.3.1.orig/configure php-5.3.1/configure
--- php-5.3.1.orig/configure	2009-11-18 21:11:57.000000000 +0100
+++ php-5.3.1/configure	2010-01-15 23:43:10.000000000 +0100
@@ -14712,7 +14712,7 @@
     ac_libs=$LIBS
     LIBS="$LIBS -ldl"
     if test "$cross_compiling" = yes; then
-  found=no
+  found=yes
 else
   cat > conftest.$ac_ext <<EOF
 #line 14719 "configure"
@@ -54494,7 +54494,7 @@
     _SAVE_CPPFLAGS=$CPPFLAGS
   _SAVE_LIBS=$LIBS
   CPPFLAGS="$CPPFLAGS -I$LDAP_INCDIR"
-  LIBS="$LIBS $LDAP_SHARED_LIBADD"
+  LIBS="$LIBS $LDAP_SHARED_LIBADD -lsasl2"
 
     echo $ac_n "checking for 3 arg ldap_set_rebind_proc""... $ac_c" 1>&6
 echo "configure:54501: checking for 3 arg ldap_set_rebind_proc" >&5
