Index: php-5.2.3/configure
===================================================================
--- php-5.2.3.orig/configure	2007-07-30 10:52:16.000000000 -0500
+++ php-5.2.3/configure	2007-07-30 10:52:16.000000000 -0500
@@ -14233,7 +14233,7 @@
     ac_libs=$LIBS
     LIBS="$LIBS -ldl"
     if test "$cross_compiling" = yes; then
-  found=no
+  found=yes
 else
   cat > conftest.$ac_ext <<EOF
 #line 14240 "configure"
@@ -52155,7 +52155,7 @@
     _SAVE_CPPFLAGS=$CPPFLAGS
   _SAVE_LDFLAGS=$LDFLAGS
   CPPFLAGS="$CPPFLAGS -I$LDAP_INCDIR"
-  LDFLAGS="$LDFLAGS $LDAP_SHARED_LIBADD"
+  LDFLAGS="$LDFLAGS $LDAP_SHARED_LIBADD -lsasl2"
 
     echo $ac_n "checking for 3 arg ldap_set_rebind_proc""... $ac_c" 1>&6
 echo "configure:52162: checking for 3 arg ldap_set_rebind_proc" >&5
