diff -ruN nufw-2.2.17/Makefile.am nufw-2.2.17.new/Makefile.am
--- nufw-2.2.17/Makefile.am	2008-09-29 19:06:22.000000000 +0200
+++ nufw-2.2.17.new/Makefile.am	2008-04-04 11:15:37.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = src scripts doc tests conf python
+SUBDIRS = src scripts conf python
 
 PATCH_FILES = patches/dump-connection-mark.diff 
 EXTRA_DIST = autogen.sh $(PATCH_FILES)
diff -ruN nufw-2.2.17/Makefile.in nufw-2.2.17.new/Makefile.in
--- nufw-2.2.17/Makefile.in	2008-09-29 19:06:34.000000000 +0200
+++ nufw-2.2.17.new/Makefile.in	2008-09-24 11:25:21.000000000 +0200
@@ -193,7 +193,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src scripts doc tests conf python
+SUBDIRS = src scripts conf python
 PATCH_FILES = patches/dump-connection-mark.diff 
 EXTRA_DIST = autogen.sh $(PATCH_FILES)
 all: all-recursive
