Index: nufw-2.2.21/configure
===================================================================
--- nufw-2.2.21.orig/configure	2009-02-03 22:46:52.000000000 +0100
+++ nufw-2.2.21/configure	2009-10-05 20:27:16.000000000 +0200
@@ -12546,7 +12546,7 @@
 
   if test x$libgnutls_config_prefix != x ; then
      if test x${LIBGNUTLS_CONFIG+set} != xset ; then
-        LIBGNUTLS_CONFIG=$libgnutls_config_prefix/bin/libgnutls-config
+        LIBGNUTLS_CONFIG=$STAGING_DIR/host/bin/libgnutls-config
      fi
   fi
 
