--- dillo-rcc974615ae39.orig/configure.in	2009-03-09 15:40:08.000000000 +0100
+++ dillo-rcc974615ae39/configure.in	2009-03-09 15:52:42.000000000 +0100
@@ -145,14 +145,10 @@
 dnl
 dnl For debugging and to be user friendly
 AC_MSG_CHECKING([FLTK2])
-if sh -c "fltk2-config --version" >/dev/null 2>&1
-then AC_MSG_RESULT(yes)
-     LIBFLTK_CXXFLAGS=`fltk2-config --cxxflags`
-     LIBFLTK_CFLAGS=`fltk2-config --cflags`
-     LIBFLTK_LIBS=`fltk2-config --use-images --ldflags`
-else AC_MSG_RESULT(no)
-     AC_ERROR(FLTK2 must be installed!)
-fi
+AC_MSG_RESULT(yes)
+     LIBFLTK_CXXFLAGS="-I. -I/usr/include/freetype2 -Wno-non-virtual-dtor"
+     LIBFLTK_CFLAGS="-I."
+     LIBFLTK_LIBS="-L./lib -Wl,-rpath,/usr/local/lib -lfltk2_images -lfltk2_images -lfltk2 -lX11 -lXi -lXft -lpthread -lm -lXext -lsupc++"
 
 
 dnl ----------------
