Index: faad2/libfaad/Makefile.am
===================================================================
--- faad2.orig/libfaad/Makefile.am	2008-03-20 12:54:17.000000000 +0100
+++ faad2/libfaad/Makefile.am	2008-03-20 12:54:36.000000000 +0100
@@ -3,7 +3,8 @@
 include_HEADERS = $(top_srcdir)/include/faad.h \
                   $(top_srcdir)/include/neaacdec.h
 
-libfaad_la_LDFLAGS = -lm
+libfaad_la_LDFLAGS = -version-info 1:0:0
+libfaad_la_LIBADD = -lm
 
 libfaad_la_SOURCES = bits.c cfft.c decoder.c drc.c \
 drm_dec.c error.c filtbank.c \
@@ -25,4 +26,4 @@
 sbr_qmf_c.h codebook/hcb.h \
 codebook/hcb_1.h codebook/hcb_2.h codebook/hcb_3.h codebook/hcb_4.h \
 codebook/hcb_5.h codebook/hcb_6.h codebook/hcb_7.h codebook/hcb_8.h \
-codebook/hcb_9.h codebook/hcb_10.h codebook/hcb_11.h codebook/hcb_sf.h
\ No newline at end of file
+codebook/hcb_9.h codebook/hcb_10.h codebook/hcb_11.h codebook/hcb_sf.h
