diff -Nru asterisk-1.4.18.org/channels/chan_iax2.c asterisk-1.4.18/channels/chan_iax2.c
--- asterisk-1.4.18.org/channels/chan_iax2.c	2008-01-31 20:52:49.000000000 +0100
+++ asterisk-1.4.19.1/channels/chan_iax2.c	2008-03-02 19:32:14.000000000 +0100
@@ -1554,7 +1554,7 @@
 		last++;
 	else
 		last = s;
-	snprintf(s2, strlen(s) + 100, "/var/tmp/%s-%ld", last, (unsigned long)ast_random());
+	snprintf(s2, strlen(s) + 100, "/tmp/%s-%ld", last, (unsigned long)ast_random());
 	res = stat(s, &stbuf);
 	if (res < 0) {
 		ast_log(LOG_WARNING, "Failed to stat '%s': %s\n", s, strerror(errno));
