diff -Nru asterisk-1.4.18.org/configure.ac asterisk-1.4.18/configure.ac
--- asterisk-1.4.18.org/configure.ac	2008-01-16 02:13:27.000000000 +0100
+++ asterisk-1.4.19.1/configure.ac	2008-03-02 19:38:21.000000000 +0100
@@ -470,7 +470,7 @@
    fi
 fi
 
-AST_EXT_LIB_CHECK([IKSEMEL], [iksemel], [iks_start_sasl], [iksemel.h])
+AST_EXT_LIB_CHECK([IKSEMEL], [iksemel], [iks_start_sasl], [iksemel.h], [-lgnutls -lgcrypt -lgpg-error])
 
 if test "${PBX_IKSEMEL}" = 1; then
    AST_EXT_LIB_CHECK([GNUTLS], [gnutls], [gnutls_bye], [gnutls/gnutls.h], [-lz -lgcrypt -lgpg-error])
