diff -Nru asterisk-1.4.18.org/Makefile asterisk-1.4.18/Makefile
--- asterisk-1.4.18.org/Makefile	2008-01-29 18:21:33.000000000 +0100
+++ asterisk-1.4.19.1/Makefile	2008-03-02 19:39:23.000000000 +0100
@@ -215,7 +215,7 @@
 endif
 
 ifneq ($(PROC),ultrasparc)
-  ASTCFLAGS+=$(shell if $(CC) -march=$(PROC) -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-march=$(PROC)"; fi)
+  #ASTCFLAGS+=$(shell if $(CC) -march=$(PROC) -S -o /dev/null -xc /dev/null >/dev/null 2>&1; then echo "-march=$(PROC)"; fi)
 endif
 
 ifeq ($(PROC),ppc)
