Index: weechat-0.2.5/configure
===================================================================
--- weechat-0.2.5.orig/configure	2007-06-30 17:15:36.000000000 +0200
+++ weechat-0.2.5/configure	2007-06-30 17:15:36.000000000 +0200
@@ -29349,8 +29349,7 @@
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
-See \`config.log' for more details." >&2;}
-   { (exit 1); exit 1; }; }
+See \`config.log' for more details." >&2;} }
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
@@ -29407,7 +29406,6 @@
 echo "${ECHO_T}yes" >&6; }
   fi
 fi
-
 if test "x$iconv_found" = "xno" ; then
    { { echo "$as_me:$LINENO: error:
 *** Iconv headers and/or libraries couldn't be found in your system.
@@ -29416,8 +29414,7 @@
 echo "$as_me: error:
 *** Iconv headers and/or libraries couldn't be found in your system.
 *** Try to install them with your software package manager.
-*** WeeChat can't be built without Iconv support." >&2;}
-   { (exit 1); exit 1; }; }
+*** WeeChat can't be built without Iconv support." >&2;} }
 fi
 
 # ------------------------------------------------------------------------------
