--- a/configure
+++ b/configure
@@ -30177,14 +30177,6 @@ fi
   test -n "$LUACONFIG" && break
 done
 
-        if test "x$LUACONFIG" != "x" ; then
-            { echo "$as_me:$LINENO: checking for Lua headers and librairies with lua-config" >&5
-echo $ECHO_N "checking for Lua headers and librairies with lua-config... $ECHO_C" >&6; }
-	    echo
-            LUA_CFLAGS=`$LUACONFIG --include`
-            LUA_LFLAGS=`$LUACONFIG --libs`
-	    LUA_VERSION="5.0.x"
-        fi
     fi
 
     if test "x$LUA_CFLAGS" = "x" -o "x$LUA_LFLAGS" = "x" ; then
