--- a/configure.in
+++ b/configure.in
@@ -176,21 +176,7 @@ fi
 #    WXWIDGETS_LIBS=""
 #fi
 
-if test "x$enable_gtk" = "xyes" ; then
-    AM_PATH_GTK_2_0(2.4.0, LIBGTK_FOUND=1, LIBGTK_FOUND=0)
-    if test "$LIBGTK_FOUND" = "0" ; then
-        AC_MSG_WARN([
-*** Gtk library not found!
-*** WeeChat will be built without Gtk support.])
-        enable_gtk="no"
-        not_found="$not_found gtk"
-    else
-        GTK_CFLAGS=`pkg-config --cflags gtk+-2.0`
-        GTK_LIBS=`pkg-config --libs gtk+-2.0`
-        AC_SUBST(GTK_CFLAGS)
-        AC_SUBST(GTK_LIBS)
-    fi
-fi
+enable_gtk="no"
 
 # ------------------------------------------------------------------------------
 #                                  iconv
@@ -586,20 +572,11 @@ fi
 # ------------------------------------------------------------------------------
 
 if test "x$enable_gnutls" = "xyes" ; then
-    found_gnutls="no"
-    AM_PATH_LIBGNUTLS( 1.0.0, found_gnutls=yes, AC_MSG_WARN([[
-*** libgnutls was not found. You may want to get it from ftp://ftp.gnutls.org/pub/gnutls/
-*** WeeChat will be built without GnuTLS support.]]))
-    if test "x$found_gnutls" = "xyes" ; then
-        GNUTLS_CFLAGS=`libgnutls-config --cflags`
-        GNUTLS_LFLAGS=`libgnutls-config --libs`
-        AC_SUBST(GNUTLS_CFLAGS)
-        AC_SUBST(GNUTLS_LFLAGS)
-        AC_DEFINE(HAVE_GNUTLS)
-    else
-        enable_gnutls="no"
-        not_found="$not_found gnutls"
-    fi
+    GNUTLS_CFLAGS=`pkg-config --cflags gnutls`
+    GNUTLS_LFLAGS=`pkg-config --libs gnutls`
+    AC_SUBST(GNUTLS_CFLAGS)
+    AC_SUBST(GNUTLS_LFLAGS)
+    AC_DEFINE(HAVE_GNUTLS)
 fi
 
 # ------------------------------------------------------------------------------
