diff --git a/configure b/configure
index f388ffe..c9ec352 100755
--- a/configure
+++ b/configure
@@ -31400,8 +31400,8 @@ echo "$as_me: WARNING:
 
 
     if test "x$found_gnutls" = "xyes" ; then
-        GNUTLS_CFLAGS=`libgnutls-config --cflags`
-        GNUTLS_LFLAGS=`libgnutls-config --libs`
+        GNUTLS_CFLAGS=`pkg-config gnutls --cflags`
+        GNUTLS_LFLAGS=`pkg-config gnutls --libs`
 
 
         cat >>confdefs.h <<\_ACEOF
