--- httpd-2.2.4.orig/build/mkconfNW.awk	2007-07-02 03:30:22.000000000 +0200
+++ httpd-2.2.4/build/mkconfNW.awk	2007-07-15 21:29:01.000000000 +0200
@@ -23,7 +23,7 @@
     A["sysconfdir"] = "conf"
     A["iconsdir"] = "icons"
     A["manualdir"] = "manual"
-    A["runtimedir"] = "logs"
+    A["runtimedir"] = "log"
     A["errordir"] = "error"
     A["proxycachedir"] = "proxy"
 
--- httpd-2.2.4.orig/config.layout	2007-07-02 03:30:22.000000000 +0200
+++ httpd-2.2.4/config.layout	2007-07-15 22:03:15.000000000 +0200
@@ -28,8 +28,8 @@
     cgidir:        ${datadir}/cgi-bin
     includedir:    ${prefix}/include
     localstatedir: ${prefix}
-    runtimedir:    ${localstatedir}/logs
-    logfiledir:    ${localstatedir}/logs
+    runtimedir:    ${localstatedir}/log
+    logfiledir:    ${localstatedir}/log
     proxycachedir: ${localstatedir}/proxy
 </Layout>
 
@@ -150,7 +150,7 @@
     includedir:    ${prefix}/include
     localstatedir: /var${prefix}
     runtimedir:    ${localstatedir}/run
-    logfiledir:    ${localstatedir}/logs
+    logfiledir:    ${localstatedir}/log
     proxycachedir: ${localstatedir}/proxy
 </Layout>
 
