Index: strongswan-4.3.7/src/libstrongswan/utils/identification.c
===================================================================
--- strongswan-4.3.7.orig/src/libstrongswan/utils/identification.c	2010-11-04 01:34:00.492000001 +0100
+++ strongswan-4.3.7/src/libstrongswan/utils/identification.c	2010-11-04 01:34:06.791999987 +0100
@@ -810,7 +810,7 @@
 	return print_in_hook(dst, len, "%*s", spec->width, buf);
 }
 
-METHOD(identification_t, clone, identification_t*,
+METHOD(identification_t, clone_, identification_t*,
 	private_identification_t *this)
 {
 	private_identification_t *clone = malloc_thing(private_identification_t);
@@ -842,7 +842,7 @@
 			.get_encoding = _get_encoding,
 			.get_type = _get_type,
 			.create_part_enumerator = _create_part_enumerator,
-			.clone = _clone,
+			.clone = _clone_,
 			.destroy = _destroy,
 		},
 		.type = type,
