Index: strongswan-4.3.7/src/pluto/adns.c
===================================================================
--- strongswan-4.3.7.orig/src/pluto/adns.c	2010-11-04 01:34:39.497000001 +0100
+++ strongswan-4.3.7/src/pluto/adns.c	2010-11-04 01:35:10.609000001 +0100
@@ -179,7 +179,7 @@
 	res_query(dname, class, type, answer, anslen)
 # define res_nclose(statp) res_close()
 
-static struct __res_state *statp = &_res;
+#define statp	((struct __res_state *)(&_res))
 
 #else /* !OLD_RESOLVER */
 
