diff -urN batman-adv-r963/batman-adv-kernelland/batman-core/Makefile.kbuild batman-adv-r963.new/batman-adv-kernelland/batman-core/Makefile.kbuild
--- batman-adv-r963/batman-adv-kernelland/batman-core/Makefile.kbuild	2008-01-28 08:47:47.000000000 +0100
+++ batman-adv-r963.new/batman-adv-kernelland/batman-core/Makefile.kbuild	2008-03-02 18:35:39.000000000 +0100
@@ -23,7 +23,7 @@
 -include $(TOPDIR)/Rules.make
 endif
 
-CFLAGS += -DREVISION_VERSION=\"\ rv$(REVISION)\"
+EXTRA_CFLAGS += -DREVISION_VERSION=\"\ rv$(REVISION)\"
 
 obj-m += batman-adv-core.o
 batman-adv-core-objs := batman-adv-main.o batman-adv-proc.o batman-adv-send.o batman-adv-log.o batman-adv-routing.o batman-adv-interface.o batman-adv-device.o batman-adv-ttable.o bitarray.o hash.o ring_buffer.o
