--- a/configure
+++ b/configure
@@ -1058,7 +1058,7 @@ LIBGCONF_CFLAGS
 LIBGCONF_LIBS
 DBUS_GLIB_CFLAGS
 DBUS_GLIB_LIBS'
-ac_subdirs_all='third-party/libevent'
+#ac_subdirs_all='third-party/libevent'
 
 # Initialize some variables set by options.
 ac_init_help=
@@ -19881,7 +19881,7 @@ fi
 
 
 
-subdirs="$subdirs third-party/libevent"
+#subdirs="$subdirs third-party/libevent"
 
 libevent_source=bundled
 { $as_echo "$as_me:$LINENO: checking for evutil_vsnprintf in -levent" >&5
--- a/third-party/Makefile.am
+++ b/third-party/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = libnatpmp miniupnp libevent dht
+SUBDIRS = libnatpmp miniupnp dht
 
 EXTRA_DIST = \
   macosx-libevent-config.h \
--- a/third-party/Makefile.in
+++ b/third-party/Makefile.in
@@ -278,7 +278,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 transmissionlocaledir = @transmissionlocaledir@
-SUBDIRS = libnatpmp miniupnp libevent dht
+SUBDIRS = libnatpmp miniupnp dht
 EXTRA_DIST = \
   macosx-libevent-config.h \
   macosx-libevent-event-config.h
