--- a/Makefile
+++ b/Makefile
@@ -61,7 +61,8 @@
 	-L$(MESSAGES)     -lMsg            \
 	-lClntOptions -lOptions -lLowLevel \
 	-L$(MISC)         -lMisc           \
-	-L$(POSLIB)       $(CLNTPOSLIB)
+	-L$(POSLIB)       $(CLNTPOSLIB) \
+	$(CLNT_LIBS)
 
 server: $(SERVERBIN)
 
@@ -93,7 +94,8 @@
 	-L$(MISC)        -lMisc            \
 	-L$(LOWLEVEL)    -lLowLevel        \
 	-L$(IFACEMGR)    -lIfaceMgr        \
-	-L$(POSLIB)      $(SRVPOSLIB)
+	-L$(POSLIB)      $(SRVPOSLIB) \
+	$(SRV_LIBS)
 
 
 relay: $(RELAYBIN)
@@ -113,7 +115,8 @@
 	-L$(MESSAGES)    -lMsg          \
 	-L$(MISC)        -lMisc         \
 	-L$(OPTIONS)     -lOptions      \
-	-lMisc -lIfaceMgr -lLowLevel -lRelTransMgr -lRelCfgMgr -lRelMsg -lRelOptions -lOptions
+	-lMisc -lIfaceMgr -lLowLevel -lRelTransMgr -lRelCfgMgr -lRelMsg -lRelOptions -lOptions \
+	$(SRV_LIBS)
 
 requestor: $(REQUESTORBIN)
 $(REQUESTORBIN): includes commonlibs Requestor $(REQUESTORDIRS)
