Index: lcd4linux-847/drv_RouterBoard.c
===================================================================
--- lcd4linux-847.orig/drv_RouterBoard.c	2007-12-01 17:15:10.000000000 +0100
+++ lcd4linux-847/drv_RouterBoard.c	2007-12-01 17:15:10.000000000 +0100
@@ -106,7 +106,7 @@
 #include <string.h>
 #include <errno.h>
 #include <unistd.h>
-#include <sys/io.h>
+#include <asm/io.h>
 
 #include "debug.h"
 #include "cfg.h"
