diff -urN icecast-2.3.2/src/format_flac.c icecast-2.3.2.new/src/format_flac.c
--- icecast-2.3.2/src/format_flac.c	2006-09-21 03:49:16.000000000 +0200
+++ icecast-2.3.2.new/src/format_flac.c	2008-09-29 19:43:54.000000000 +0200
@@ -18,7 +18,7 @@
 #endif
 
 #include <stdlib.h>
-#include <ogg/ogg.h>
+#include <tremor/ogg.h>
 #include <string.h>
 
 typedef struct source_tag source_t;
diff -urN icecast-2.3.2/src/format_midi.c icecast-2.3.2.new/src/format_midi.c
--- icecast-2.3.2/src/format_midi.c	2006-09-21 03:49:16.000000000 +0200
+++ icecast-2.3.2.new/src/format_midi.c	2008-09-29 19:44:15.000000000 +0200
@@ -18,7 +18,7 @@
 #endif
 
 #include <stdlib.h>
-#include <ogg/ogg.h>
+#include <tremor/ogg.h>
 #include <string.h>
 
 typedef struct source_tag source_t;
diff -urN icecast-2.3.2/src/format_ogg.c icecast-2.3.2.new/src/format_ogg.c
--- icecast-2.3.2/src/format_ogg.c	2008-04-19 15:45:30.000000000 +0200
+++ icecast-2.3.2.new/src/format_ogg.c	2008-09-29 19:44:32.000000000 +0200
@@ -24,7 +24,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <ogg/ogg.h>
+#include <tremor/ogg.h>
 
 #include "refbuf.h"
 #include "source.h"
diff -urN icecast-2.3.2/src/format_ogg.h icecast-2.3.2.new/src/format_ogg.h
--- icecast-2.3.2/src/format_ogg.h	2006-09-21 03:49:16.000000000 +0200
+++ icecast-2.3.2.new/src/format_ogg.h	2008-09-29 19:44:38.000000000 +0200
@@ -18,7 +18,7 @@
 #ifndef __FORMAT_OGG_H__
 #define __FORMAT_OGG_H__
 
-#include <ogg/ogg.h>
+#include <tremor/ogg.h>
 #include "refbuf.h"
 #include "format.h"
 
diff -urN icecast-2.3.2/src/format_speex.c icecast-2.3.2.new/src/format_speex.c
--- icecast-2.3.2/src/format_speex.c	2007-10-04 04:07:57.000000000 +0200
+++ icecast-2.3.2.new/src/format_speex.c	2008-09-29 19:44:56.000000000 +0200
@@ -18,7 +18,7 @@
 #endif
 
 #include <stdlib.h>
-#include <ogg/ogg.h>
+#include <tremor/ogg.h>
 #include <speex/speex_header.h>
 
 typedef struct source_tag source_t;
diff -urN icecast-2.3.2/src/format_theora.c icecast-2.3.2.new/src/format_theora.c
--- icecast-2.3.2/src/format_theora.c	2006-09-21 03:49:16.000000000 +0200
+++ icecast-2.3.2.new/src/format_theora.c	2008-09-29 19:46:36.000000000 +0200
@@ -18,7 +18,7 @@
 #endif
 
 #include <stdlib.h>
-#include <ogg/ogg.h>
+#include <tremor/ogg.h>
 #include <theora/theora.h>
 
 typedef struct source_tag source_t;
diff -urN icecast-2.3.2/src/format_vorbis.c icecast-2.3.2.new/src/format_vorbis.c
--- icecast-2.3.2/src/format_vorbis.c	2007-09-13 00:40:55.000000000 +0200
+++ icecast-2.3.2.new/src/format_vorbis.c	2008-09-29 19:47:25.000000000 +0200
@@ -18,8 +18,8 @@
 #endif
 
 #include <stdlib.h>
-#include <ogg/ogg.h>
-#include <vorbis/codec.h>
+#include <tremor/ogg.h>
+#include <tremor/ivorbiscodec.h>
 #include <memory.h>
 #include <string.h>
 
diff -urN icecast-2.3.2/src/source.c icecast-2.3.2.new/src/source.c
--- icecast-2.3.2/src/source.c	2008-05-15 15:25:04.000000000 +0200
+++ icecast-2.3.2.new/src/source.c	2008-09-29 19:47:40.000000000 +0200
@@ -19,7 +19,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/types.h>
-#include <ogg/ogg.h>
+#include <tremor/ogg.h>
 #include <errno.h>
 
 #ifndef _WIN32
--- icecast-2.3.2/configure	2008-05-27 02:21:57.000000000 +0200
+++ icecast-2.3.2.new/configure	2008-09-29 20:05:43.000000000 +0200
@@ -22199,7 +22199,7 @@
 if test "${xt_cv_lib_ogg+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
-  OGG_LIBS="-logg"
+  OGG_LIBS="-lvorbisidec"
 
 #
 # check if the installed Ogg is sufficiently new.
@@ -22261,7 +22261,7 @@
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include <ogg/ogg.h>
+#include <tremor/ogg.h>
 int
 main ()
 {
@@ -22410,9 +22410,9 @@
 
 fi
 
-VORBIS_LIBS="-lvorbis"
-VORBISFILE_LIBS="-lvorbisfile"
-VORBISENC_LIBS="-lvorbisenc"
+VORBIS_LIBS="-lvorbisidec"
+VORBISFILE_LIBS="-lvorbisidec"
+VORBISENC_LIBS="-lvorbisidec"
 
 xt_save_LIBS="$LIBS"
 xt_save_LDFLAGS="$LDFLAGS"
@@ -22537,14 +22537,13 @@
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 
-#include <vorbis/codec.h>
-#include <vorbis/vorbisenc.h>
+#include <tremor/ivorbiscodec.h>
 
 int
 main ()
 {
 
-struct ovectl_ratemanage_arg a;
+/*struct ovectl_ratemanage_arg a;*/
 
   ;
   return 0;
--- icecast-2.3.2/src/format_kate.c	2008-04-18 18:29:28.000000000 +0200
+++ icecast-2.3.2.new/src/format_kate.c	2008-10-01 08:27:00.000000000 +0200
@@ -19,7 +19,7 @@
 
 #include <stdlib.h>
 #include <string.h>
-#include <ogg/ogg.h>
+#include <tremor/ogg.h>
 #ifdef HAVE_KATE
 #include <kate/oggkate.h>
 #endif
