diff -urN freeradius-1.1.6.old/src/include/modpriv.h freeradius-1.1.6/src/include/modpriv.h
--- freeradius-1.1.6.old/src/include/modpriv.h	2008-02-16 11:27:01.000000000 -0500
+++ freeradius-1.1.6/src/include/modpriv.h	2008-02-16 11:26:24.000000000 -0500
@@ -4,7 +4,7 @@
  * Version: $Id: modpriv.h,v 1.3 2004/02/26 19:04:19 aland Exp $ */
 #include "radiusd.h"
 #include "modules.h"
-#include "ltdl.h"
+#include "../../libltdl/ltdl.h"
 
 /*
  *	Keep track of which modules we've loaded.
diff -urN freeradius-1.1.6/src/main/modules.c freeradius-1.1.6.old/src/main/modules.c
--- freeradius-1.1.6/src/main/modules.c	2008-02-16 11:42:36.000000000 -0500
+++ freeradius-1.1.6.old/src/main/modules.c	2008-02-16 11:42:14.000000000 -0500
@@ -36,7 +36,7 @@
 #include "modules.h"
 #include "modcall.h"
 #include "conffile.h"
-#include "ltdl.h"
+#include "../../libltdl/ltdl.h"
 #include "rad_assert.h"
 
 /*
