diff -urN gpsd-2.39/drivers.c gpsd-2.39.new/drivers.c
--- gpsd-2.39/drivers.c	2009-03-18 18:37:56.000000000 +0100
+++ gpsd-2.39.new/drivers.c	2009-07-29 00:22:54.000000000 +0200
@@ -165,9 +165,13 @@
 	/* probe for the FV-18 -- expect $PFEC,GPint followed by data */
 	(void)nmea_send(session, "$PFEC,GPint");
 	break;
+    case 3:
+	/* probe for the Trimble Copernicus */ 
+	(void)nmea_send(session->gpsdata.gps_fd, "$PTNLSNM,0139,01");
+	break;
 #endif /* NMEA_ENABLE */
 #ifdef EVERMORE_ENABLE
-    case 3:
+    case 4:
 	/* Enable checksum and GGA(1s), GLL(0s), GSA(1s), GSV(1s), RMC(1s), VTG(0s), PEMT101(1s) */
 	/* EverMore will reply with: \x10\x02\x04\x38\x8E\xC6\x10\x03 */
 	(void)gpsd_write(session,
@@ -175,31 +179,31 @@
 	break;
 #endif /* EVERMORE_ENABLE */
 #ifdef ITRAX_ENABLE
-    case 4:
+    case 5:
 	/* probe for iTrax, looking for "$PFST,OK" */
 	(void)nmea_send(session, "$PFST");
 	break;
 #endif /* ITRAX_ENABLE */
 #ifdef GPSCLOCK_ENABLE
-    case 5:
+    case 6:
 	/* probe for Furuno Electric GH-79L4-N (GPSClock); expect $PFEC,GPssd */
 	(void)nmea_send(session, "$PFEC,GPsrq");
 	break;
 #endif /* GPSCLOCK_ENABLE */
 #ifdef ASHTECH_ENABLE
-    case 6:
+    case 7:
 	/* probe for Ashtech -- expect $PASHR,RID */
 	(void)nmea_send(session, "$PASHQ,RID");
 	break;
 #endif /* ASHTECH_ENABLE */
 #ifdef UBX_ENABLE
-    case 7:
+    case 8:
 	/* probe for UBX -- query software version */
 	(void)ubx_write(session, 0x0au, 0x04, NULL, 0);
 	break;
 #endif /* UBX_ENABLE */
 #ifdef MKT3301_ENABLE
-    case 8:
+    case 9:
 	/* probe for MKT-3301 -- expect $PMTK705 */
 	(void)nmea_send(session, "$PMTK605");
 	break;
