Index: pulseaudio-0.9.22/src/pulsecore/memblock.c
===================================================================
--- pulseaudio-0.9.22.orig/src/pulsecore/memblock.c	2011-03-08 09:25:27.000000000 +0100
+++ pulseaudio-0.9.22/src/pulsecore/memblock.c	2011-03-08 09:27:21.000000000 +0100
@@ -54,7 +54,7 @@
  * stored in SHM and our OS does not commit the memory before we use
  * it for the first time. */
 #define PA_MEMPOOL_SLOTS_MAX 1024
-#define PA_MEMPOOL_SLOT_SIZE (64*1024)
+#define PA_MEMPOOL_SLOT_SIZE 64
 
 #define PA_MEMEXPORT_SLOTS_MAX 128
 
