--- ORBit2-2.14.19/configure.in	2010-07-01 15:57:01.000000000 +0200
+++ ORBit2-2.14.19.new/configure.in	2012-12-11 14:02:51.624767158 +0100
@@ -213,25 +213,6 @@
 AC_C_BIGENDIAN
 
 
-dnl
-dnl   So many people with broken linking setups asked about this
-dnl it turned out to be necessary to check for this.
-dnl
-dofus_save_libs="$LIBS"
-LIBS="$ORBIT_LIBS $LIBS"
-AC_TRY_RUN([ #include <stdio.h>
-             main ()
-             {
-               return 0;
-             } ],,
-           AC_MSG_ERROR([
-Linking is badly borked on your system. Please ensure your library path is correct
-Check config.log for details - check near the end of the log above 'failed program was'
-Alternatively ensure that your /etc/ld.so.conf (and/or LD_LIBRARY_PATH) includes the
-prefix you're compiling on: '${prefix}' ]), 
-AC_MSG_WARN([Could not check for borked linking while cross-compiling]))
-LIBS="$dofus_save_libs"
-
 orig_CPPFLAGS=$CPPFLAGS
 CPPFLAGS="$CPPFLAGS $ORBIT_CFLAGS"
 AC_CHECK_ALIGNOF(CORBA_octet)
