diff -urN nfs-utils-1.0.7/tools/getkversion/getkversion.c nfs-utils-1.0.7.new/tools/getkversion/getkversion.c
--- nfs-utils-1.0.7/tools/getkversion/getkversion.c	1999-10-19 01:21:12.000000000 +0200
+++ nfs-utils-1.0.7.new/tools/getkversion/getkversion.c	2007-01-05 23:25:14.000000000 +0100
@@ -12,6 +12,10 @@
 int
 main(void)	/* This is for Dan Popp ;) */
 {
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,19)
+        printf("%s\n", LINUX_VERSION_CODE);
+#else
 	printf("%s\n", UTS_RELEASE);
+#endif  
 	return 0;
 }
