--- a/tools/rpcgen/rpc_main.c
+++ b/tools/rpcgen/rpc_main.c
@@ -545,13 +545,6 @@ s_output(int argc, char **argv, char *in
 	  timerflag = 1;
 	}
 
-#ifndef linux
-	if( !tirpcflag && inetdflag )
-	  f_print(fout, "#include <sys/ttycom.h>/* TIOCNOTTY */\n");
-#else
-	if( !tirpcflag )
-	  f_print(fout, "#include <sys/ttycom.h>/* TIOCNOTTY */\n");
-#endif
 	if( Cflag && (inetdflag || pmflag ) ) {
 	  f_print(fout, "#ifdef __cplusplus\n");
 	  f_print(fout, "#include <sysent.h> /* getdtablesize, open */\n"); 
--- a/tools/rpcgen/rpc_svcout.c
+++ b/tools/rpcgen/rpc_svcout.c
@@ -788,7 +788,6 @@ write_rpc_svc_fg(char *infile, char *sp)
 	else {
 	  f_print(fout, "%si = open(\"/dev/tty\", 2);\n", sp);
 	  f_print(fout, "%sif (i >= 0) {\n", sp);
-	  f_print(fout, "%s\t(void) ioctl(i, TIOCNOTTY, (char *)NULL);\n", sp);;
 	  f_print(fout, "%s\t(void) close(i);\n", sp);
 	  f_print(fout, "%s}\n", sp);
 	}
