Index: lighttpd-1.4.19/configure
===================================================================
--- lighttpd-1.4.19.orig/configure	2008-03-11 20:34:42.000000000 +0100
+++ lighttpd-1.4.19/configure	2008-03-11 20:34:42.000000000 +0100
@@ -23971,7 +23971,7 @@
 { echo "$as_me:$LINENO: result: $WITH_PCRE" >&5
 echo "${ECHO_T}$WITH_PCRE" >&6; }
 
-if test "x$cross_compiling" = xno -a "$WITH_PCRE" != "no"; then
+if test "$WITH_PCRE" != "no"; then
   # Extract the first word of "pcre-config", so it can be a program name with args.
 set dummy pcre-config; ac_word=$2
 { echo "$as_me:$LINENO: checking for $ac_word" >&5
Index: lighttpd-1.4.19/configure.in
===================================================================
--- lighttpd-1.4.19.orig/configure.in	2008-03-11 20:34:42.000000000 +0100
+++ lighttpd-1.4.19/configure.in	2008-03-11 20:34:42.000000000 +0100
@@ -251,7 +251,7 @@
     [WITH_PCRE=$withval],[WITH_PCRE=yes])
 AC_MSG_RESULT([$WITH_PCRE])
 
-if test "x$cross_compiling" = xno -a "$WITH_PCRE" != "no"; then
+if test "$WITH_PCRE" != "no"; then
   AC_PATH_PROG(PCRECONFIG, pcre-config)
 
   if test x"$PCRECONFIG" != x; then
