diff -urN empcd/empcd.h empcd.new/empcd.h
--- empcd/empcd.h	2005-11-29 01:35:30.000000000 +0100
+++ empcd.new/empcd.h	2007-06-14 11:54:14.000000000 +0200
@@ -24,6 +24,10 @@
 #include <pwd.h>
 
 /* Linux specific... */
+#include <linux/version.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,0)
+#define EVIOCGRAB             _IOW('E', 0x90, int)                    /* Grab/Release device */
+#endif
 #include <linux/input.h>
 
 #ifndef NULL
