diff --git a/Makefile.in b/Makefile.in
--- a/Makefile.in
+++ b/Makefile.in
@@ -338,7 +338,7 @@ EXTRA_DIST = ChangeLog.pre-2-14 ChangeLog.pre-2-12 ChangeLog.pre-2-10 \
 	gio-2.0-uninstalled.pc.in gio-unix-2.0-uninstalled.pc.in
 TEST_PROGS = 
 AUTOMAKE_OPTIONS = 1.7
-SUBDIRS = . m4macros glib gobject gmodule gthread gio tests po docs
+SUBDIRS = . m4macros glib gobject gmodule gthread gio po
 DIST_SUBDIRS = $(SUBDIRS) build
 bin_SCRIPTS = glib-gettextize
 AM_CPPFLAGS = \
