diff -urN glib-2.16.5.orig/gio/Makefile.in glib-2.16.5/gio/Makefile.in
--- glib-2.16.5.orig/gio/Makefile.in	2008-09-02 01:43:09.000000000 +0200
+++ glib-2.16.5/gio/Makefile.in	2008-09-02 01:47:01.000000000 +0200
@@ -165,7 +165,7 @@
   distclean-recursive maintainer-clean-recursive
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = xdgmime inotify . fam tests
+DIST_SUBDIRS = xdgmime inotify . fam
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 ALLOCA = @ALLOCA@
@@ -380,7 +380,7 @@
 	pltcheck.sh $(NULL)
 TEST_PROGS = 
 NULL = 
-SUBDIRS = $(am__append_1) $(am__append_2) . $(am__append_4) tests
+SUBDIRS = $(am__append_1) $(am__append_2) . $(am__append_4)
 @MS_LIB_AVAILABLE_TRUE@noinst_DATA = gio-2.0.lib
 @HAVE_GNUC_VISIBILITY_TRUE@@OS_LINUX_TRUE@TESTS = abicheck.sh pltcheck.sh
 AM_CPPFLAGS = \
diff -urN glib-2.16.5.orig/glib/Makefile.in glib-2.16.5/glib/Makefile.in
--- glib-2.16.5.orig/glib/Makefile.in	2008-09-02 01:43:17.000000000 +0200
+++ glib-2.16.5/glib/Makefile.in	2008-09-02 01:44:48.000000000 +0200
@@ -396,8 +396,8 @@
 @ENABLE_REGEX_TRUE@gregex_c = gregex.c
 @ENABLE_REGEX_FALSE@gregex_h = 
 @ENABLE_REGEX_TRUE@gregex_h = gregex.h
-SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre . tests
-DIST_SUBDIRS = libcharset gnulib pcre update-pcre tests
+SUBDIRS = libcharset $(PRINTF_SUBDIR) $(MAYBE_PCRE) update-pcre .
+DIST_SUBDIRS = libcharset gnulib pcre update-pcre
 AM_CPPFLAGS = \
 	-I$(top_srcdir) 		\
 	$(pcre_inc) 			\
diff -urN glib-2.16.5.orig/gobject/Makefile.in glib-2.16.5/gobject/Makefile.in
--- glib-2.16.5.orig/gobject/Makefile.in	2008-09-02 01:43:06.000000000 +0200
+++ glib-2.16.5/gobject/Makefile.in	2008-09-02 01:46:05.000000000 +0200
@@ -343,7 +343,7 @@
 # GObject - GLib Type, Object, Parameter and Signal Library
 # Copyright (C) 1997,98,99,2000 Tim Janik and Red Hat, Inc.
 #
-SUBDIRS = . tests
+SUBDIRS = .
 AM_CPPFLAGS = \
 	-DG_LOG_DOMAIN=\"GLib-GObject\"		\
 	-I$(top_srcdir)				\
diff -urN glib-2.16.5.orig/Makefile.in glib-2.16.5/Makefile.in
--- glib-2.16.5.orig/Makefile.in	2008-09-02 01:42:56.000000000 +0200
+++ glib-2.16.5/Makefile.in	2008-09-02 01:44:05.000000000 +0200
@@ -342,7 +342,7 @@
 	gio-2.0-uninstalled.pc.in gio-unix-2.0-uninstalled.pc.in
 TEST_PROGS = 
 AUTOMAKE_OPTIONS = 1.7
-SUBDIRS = . m4macros glib gmodule gthread gobject gio tests po docs
+SUBDIRS = . m4macros glib gmodule gthread gobject gio po
 DIST_SUBDIRS = $(SUBDIRS) build
 bin_SCRIPTS = glib-gettextize
 AM_CPPFLAGS = \
