diff -Nru asterisk-addons-1.4.8.org/menuselect/Makefile asterisk-addons-1.4.8/menuselect/Makefile
--- asterisk-addons-1.4.8.org/menuselect/Makefile	2009-01-08 22:31:41.000000000 +0100
+++ asterisk-addons-1.4.8/menuselect/Makefile	2009-06-04 22:21:23.000000000 +0200
@@ -52,7 +52,7 @@
 $(OBJS) menuselect_gtk.o menuselect_curses.o menuselect_stub.o: autoconfig.h menuselect.h
 
 makeopts autoconfig.h: autoconfig.h.in makeopts.in
-	@./configure $(CONFIGURE_SILENT) CC= LD= AR= CFLAGS=
+	@./configure $(CONFIGURE_SILENT) CC= LD= AR= CFLAGS= LDFLAGS=
 
 menuselect gmenuselect: mxml/libmxml.a
 
