Index: gnupg-1.4.7/Makefile.in
===================================================================
--- gnupg-1.4.7.orig/Makefile.in	2007-09-07 18:53:42.000000000 +0200
+++ gnupg-1.4.7/Makefile.in	2007-09-07 18:53:42.000000000 +0200
@@ -311,7 +311,7 @@
 AUTOMAKE_OPTIONS = dist-bzip2 filename-length-max=99
 @CROSS_COMPILING_FALSE@checks = checks
 @CROSS_COMPILING_TRUE@checks = 
-SUBDIRS = m4 intl zlib util mpi cipher tools g10 keyserver po doc ${checks}
+SUBDIRS = m4 intl zlib util mpi cipher tools g10 keyserver ${checks}
 EXTRA_DIST = scripts/config.rpath  PROJECTS BUGS config.h.in autogen.sh
 DISTCLEANFILES = 
 all: config.h
