diff -urN pygtk-2.10.6.orig/Makefile.am pygtk-2.10.6/Makefile.am
--- pygtk-2.10.6.orig/Makefile.am	2008-05-24 15:18:49.000000000 +0100
+++ pygtk-2.10.6/Makefile.am	2008-05-24 15:20:50.000000000 +0100
@@ -105,7 +105,7 @@
 # code generator
 .defs.c:
 	(cd $(srcdir)\
-	 && $(PYTHON) codegen/codegen.py \
+	 && $(PYTHON_HOST) codegen/codegen.py \
 	    --py_ssize_t-clean \
 	    --register pango-types.defs \
 	    --register atk-types.defs \
diff -urN pygtk-2.10.6.orig/Makefile.in pygtk-2.10.6/Makefile.in
--- pygtk-2.10.6.orig/Makefile.in	2008-05-24 15:18:49.000000000 +0100
+++ pygtk-2.10.6/Makefile.in	2008-05-24 15:20:39.000000000 +0100
@@ -1085,7 +1085,7 @@
 # code generator
 .defs.c:
 	(cd $(srcdir)\
-	 && $(PYTHON) codegen/codegen.py \
+	 && $(PYTHON_HOST) codegen/codegen.py \
 	    --py_ssize_t-clean \
 	    --register pango-types.defs \
 	    --register atk-types.defs \
diff -urN pygtk-2.10.6.orig/gtk/Makefile.am pygtk-2.10.6/gtk/Makefile.am
--- pygtk-2.10.6.orig/gtk/Makefile.am	2008-05-24 15:18:49.000000000 +0100
+++ pygtk-2.10.6/gtk/Makefile.am	2008-05-24 15:21:10.000000000 +0100
@@ -170,7 +170,7 @@
 
 
 .defs.c: 
-	($(PYTHON) $(top_srcdir)/codegen/codegen.py \
+	($(PYTHON_HOST) $(top_srcdir)/codegen/codegen.py \
 	   $(PYGTK_CODEGEN_DEFINES) \
 	    -I $(srcdir) \
 	    --py_ssize_t-clean \
diff -urN pygtk-2.10.6.orig/gtk/Makefile.in pygtk-2.10.6/gtk/Makefile.in
--- pygtk-2.10.6.orig/gtk/Makefile.in	2008-05-24 15:18:49.000000000 +0100
+++ pygtk-2.10.6/gtk/Makefile.in	2008-05-24 15:21:20.000000000 +0100
@@ -1056,7 +1056,7 @@
 	gtk-types.defs
 
 .defs.c: 
-	($(PYTHON) $(top_srcdir)/codegen/codegen.py \
+	($(PYTHON_HOST) $(top_srcdir)/codegen/codegen.py \
 	   $(PYGTK_CODEGEN_DEFINES) \
 	    -I $(srcdir) \
 	    --py_ssize_t-clean \
