diff -u --recursive gst-plugins-base-0.10.25-vanilla/Makefile.in gst-plugins-base-0.10.25/Makefile.in
--- gst-plugins-base-0.10.25-vanilla/Makefile.in	2009-11-14 22:55:44.146292714 -0500
+++ gst-plugins-base-0.10.25/Makefile.in	2009-11-14 22:56:11.830296238 -0500
@@ -397,7 +397,6 @@
 	gst-libs 		\
 	gst sys $(SUBDIRS_EXT) 	\
 	tools 			\
-	tests 			\
 	docs			\
 	po 			\
 	common 			\
@@ -409,7 +408,6 @@
 	gst-libs		\
 	gst sys ext		\
 	tools 			\
-	tests 			\
 	po 			\
 	common 			\
 	m4
