diff -urN gst-plugins-base-0.10.21/Makefile.am gst-plugins-base-0.10.21.new/Makefile.am
--- gst-plugins-base-0.10.21/Makefile.am	2007-12-18 00:41:14.000000000 +0100
+++ gst-plugins-base-0.10.21.new/Makefile.am	2009-03-11 10:23:14.000000000 +0100
@@ -11,14 +11,12 @@
 	gst sys $(SUBDIRS_EXT) 	\
 	tools 			\
 	tests 			\
-	docs			\
 	po 			\
 	common 			\
 	m4 			\
 	pkgconfig
 
 DIST_SUBDIRS = 			\
-	docs			\
 	gst-libs		\
 	gst sys ext		\
 	tools 			\
diff -urN gst-plugins-base-0.10.21/Makefile.in gst-plugins-base-0.10.21.new/Makefile.in
--- gst-plugins-base-0.10.21/Makefile.in	2009-03-11 10:22:56.000000000 +0100
+++ gst-plugins-base-0.10.21.new/Makefile.in	2009-03-11 10:23:08.000000000 +0100
@@ -359,14 +359,12 @@
 	gst-libs 		\
 	gst sys $(SUBDIRS_EXT) 	\
 	tools 			\
-	docs			\
 	po 			\
 	common 			\
 	m4 			\
 	pkgconfig
 
 DIST_SUBDIRS = \
-	docs			\
 	gst-libs		\
 	gst sys ext		\
 	tools 			\
