diff -u --recursive gst-plugins-base-0.10.25-vanilla/common/shave-libtool.in gst-plugins-base-0.10.25/common/shave-libtool.in
--- gst-plugins-base-0.10.25-vanilla/common/shave-libtool.in	2009-11-15 07:25:57.034292287 -0500
+++ gst-plugins-base-0.10.25/common/shave-libtool.in	2009-11-15 07:26:23.829341026 -0500
@@ -41,12 +41,9 @@
 
 case "$mode" in
 compile)
-    # shave will be called and print the actual CC/CXX/LINK line
-    preserved_args="$preserved_args --shave-mode=$mode"
     pass_though=1
     ;;
 link)
-    preserved_args="$preserved_args --shave-mode=$mode"
     Q="  LINK  "
     ;;
 *)
