diff -ruN python-e_dbus-r38564.orig/setup.py python-e_dbus-r38564/setup.py
--- python-e_dbus-r38564.orig/setup.py	2009-01-13 10:14:45.000000000 +0100
+++ python-e_dbus-r38564/setup.py	2009-01-13 10:15:46.000000000 +0100
@@ -1,10 +1,7 @@
 import sys
 import os
 
-from ez_setup import use_setuptools
-use_setuptools("0.6c9")
-
-from setuptools import setup, Extension
+from distutils.core import setup, Extension
 import subprocess
 import shlex
 
