diff -ruN python-e_dbus-r37637.orig/setup.py python-e_dbus-r37637/setup.py
--- python-e_dbus-r37637.orig/setup.py	2009-01-04 12:18:59.000000000 +0100
+++ python-e_dbus-r37637/setup.py	2009-01-04 12:21:26.000000000 +0100
@@ -1,10 +1,7 @@
 import sys
 import os
 
-from ez_setup import use_setuptools
-use_setuptools("0.6c3")
-
-from setuptools import setup, Extension
+from distutils.core import setup, Extension
 import subprocess
 import shlex
 
