diff -u --recursive vips-7.26.1-vanilla/configure.in vips-7.26.1/configure.in
--- vips-7.26.1-vanilla/configure.in	2011-08-10 21:31:05.386968954 -0500
+++ vips-7.26.1/configure.in	2011-08-10 21:31:50.747488580 -0500
@@ -142,7 +142,6 @@
 AC_PROG_AWK
 AC_PROG_CC
 AC_PROG_CC_STDC
-AC_PROG_CXX
 AC_C_CONST
 AC_C_RESTRICT
 AC_PROG_RANLIB
@@ -604,7 +603,6 @@
 # in AC_OUTPUT, but that seems to break for some combinations of sh/m4
 AC_OUTPUT([
 	vips-7.26.pc
-	vipsCC-7.26.pc
 	Makefile 
 	libvips/include/vips/version.h 
 	libvips/include/Makefile 
@@ -612,7 +610,6 @@
 	libvips/Makefile 
 	libvips/arithmetic/Makefile 
 	libvips/boolean/Makefile 
-	libvips/cimg/Makefile 
 	libvips/colour/Makefile 
 	libvips/conversion/Makefile 
 	libvips/convolution/Makefile 
@@ -629,9 +626,6 @@
 	libvips/relational/Makefile 
 	libvips/resample/Makefile 
 	libvips/video/Makefile 
-	libvipsCC/include/Makefile 
-	libvipsCC/include/vips/Makefile 
-	libvipsCC/Makefile 
 	tools/Makefile 
 	tools/batch_crop 
 	tools/batch_image_convert 
@@ -639,7 +633,6 @@
 	tools/light_correct 
 	tools/shrink_width 
 	swig/Makefile 
-	swig/vipsCC/Makefile 
 	swig/python/setup.py 
 	man/Makefile
 	doc/Makefile
diff -u --recursive vips-7.26.1-vanilla/libvips/Makefile.am vips-7.26.1/libvips/Makefile.am
--- vips-7.26.1-vanilla/libvips/Makefile.am	2011-08-10 21:31:05.982949772 -0500
+++ vips-7.26.1/libvips/Makefile.am	2011-08-10 21:32:37.969906173 -0500
@@ -1,14 +1,3 @@
-# only build in the cimg dir if C++ is enabled
-if ENABLE_CXX
-C_COMPILE_DIR = cimg
-C_DIST_DIR =
-C_LIB = cimg/libcimg.la
-else
-C_COMPILE_DIR =
-C_DIST_DIR = cimg
-C_LIB = 
-endif
-
 SUBDIRS = \
 	include \
 	arithmetic \
@@ -18,7 +7,6 @@
 	conversion \
 	deprecated \
 	convolution \
-	$(C_COMPILE_DIR) \
 	format \
 	freq_filt \
 	histograms_lut \
@@ -35,9 +23,6 @@
 lib_LTLIBRARIES = libvips.la
 
 libvips_la_SOURCES = dummy.c 
-if ENABLE_CXX
-libvips_la_SOURCES += dummy2.cc
-endif
 
 # DLLs need dependant libs there too ... put @VIPS_LIBS@ at the end
 libvips_la_LIBADD = \
@@ -48,7 +33,6 @@
 	conversion/libconversion.la \
 	convolution/libconvolution.la \
 	deprecated/libdeprecated.la \
-	$(C_LIB) \
 	format/libformat.la \
 	freq_filt/libfreq_filt.la \
 	histograms_lut/libhistograms_lut.la \
@@ -62,13 +46,6 @@
 	video/libvideo.la \
 	@VIPS_LIBS@
 
-if ENABLE_CXX
-libvips_la_LIBADD += @VIPS_CXX_LIBS@
-endif
-
 libvips_la_LDFLAGS = \
 	-no-undefined \
 	-version-info @LIBRARY_CURRENT@:@LIBRARY_REVISION@:@LIBRARY_AGE@ 
-
-EXTRA_DIST = \
-	$(C_DIST_DIR)
diff -u --recursive vips-7.26.1-vanilla/libvips/resample/Makefile.am vips-7.26.1/libvips/resample/Makefile.am
--- vips-7.26.1-vanilla/libvips/resample/Makefile.am	2011-08-10 21:31:06.658928005 -0500
+++ vips-7.26.1/libvips/resample/Makefile.am	2011-08-10 21:32:59.648166658 -0500
@@ -2,22 +2,6 @@
 # you'd think we could just define a couple of variables, but that seems to
 # confuse libtool and make it link the library with g++
 # instead, have two completely different paths
-if ENABLE_CXX
-
-libresample_la_SOURCES = \
-	im_affine.c \
-	interpolate.c \
-	im_shrink.c \
-	im_rightshift_size.c \
-	transform.c \
-	resample_dispatch.c \
-	bicubic.cpp \
-	lbb.cpp \
-	nohalo.cpp \
-	vsqbs.cpp \
-	templates.h 
-
-else
 
 libresample_la_SOURCES = \
 	im_affine.c \
@@ -27,15 +11,6 @@
 	transform.c \
 	resample_dispatch.c 
 
-EXTRA_DIST = \
-	bicubic.cpp \
-	lbb.cpp \
-	nohalo.cpp \
-	vsqbs.cpp \
-	templates.h 
-
-endif
-
 noinst_LTLIBRARIES = libresample.la
 
 INCLUDES = -I${top_srcdir}/libvips/include @VIPS_CFLAGS@ @VIPS_INCLUDES@ 
