diff -u --recursive --new-file vips-7.22.2-vanilla/configure.in vips-7.22.2/configure.in
--- vips-7.22.2-vanilla/configure.in	2010-09-13 16:03:33.945285002 -0500
+++ vips-7.22.2/configure.in	2010-09-13 16:13:36.848285002 -0500
@@ -142,6 +142,7 @@
 AC_PROG_AWK
 AC_PROG_CC
 AC_PROG_CC_STDC
+AC_PROG_CXX
 AC_C_CONST
 AC_C_RESTRICT
 AC_PROG_RANLIB
@@ -157,7 +158,6 @@
   AS_HELP_STRING([--enable-cxx], [build C++ components (default: enabled)]))
 
 if test x"$enable_cxx" != "xno"; then
-  AC_PROG_CXX
   AC_DEFINE(ENABLE_CXX,1,[build C++ components])
   AM_CONDITIONAL(ENABLE_CXX, true)
   enable_cxx=yes
@@ -273,9 +273,6 @@
   enable_threads=yes
 fi
 
-# check for gtk-doc
-GTK_DOC_CHECK(1.9)
-
 # optional supporting libraries
 
 # we can wrap fftw3 and fftw2 ... but just look for fftw3, since we can do
@@ -633,9 +630,6 @@
 	swig/vipsCC/Makefile 
 	swig/python/setup.py 
 	man/Makefile
-	doc/Makefile
-	doc/reference/Makefile
-	doc/reference/libvips-docs.sgml
 	po/Makefile.in
 ])

diff -u --recursive --new-file vips-7.22.2-vanilla/Makefile.am vips-7.22.2/Makefile.am
--- vips-7.22.2-vanilla/Makefile.am	2010-09-13 16:03:33.946285002 -0500
+++ vips-7.22.2/Makefile.am	2010-09-13 16:10:27.306284973 -0500
@@ -24,7 +24,6 @@
 	tools \
 	man \
 	po \
-	doc \
 	$(C_COMPILE_DIR) \
 	$(P_COMPILE_DIR) 
