diff -u --recursive vips-7.26.1-vanilla/libvips/format/fits.c vips-7.26.1/libvips/format/fits.c
--- vips-7.26.1-vanilla/libvips/format/fits.c	2011-08-10 21:31:05.971950126 -0500
+++ vips-7.26.1/libvips/format/fits.c	2011-08-10 22:12:08.478853668 -0500
@@ -56,8 +56,6 @@
 #endif /*HAVE_CONFIG_H*/
 #include <vips/intl.h>
 
-#ifdef HAVE_CFITSIO
-
 #include <stdio.h>
 #include <assert.h>
 #include <stdlib.h>
@@ -67,6 +65,18 @@
 #include <vips/internal.h>
 #include <vips/debug.h>
 
+#ifndef HAVE_CFITSIO
+
+int
+im_fits2vips( const char *filename, IMAGE *im )
+{
+        im_error( "im_fits2vips", "%s",
+	         _( "FITS support disabled" ) );
+	return( -1 );
+}
+
+#else /* HAVE_CFITSIO */
+
 #include <fitsio.h>
 
 #ifdef WITH_DMALLOC
