diff -urN mjpg-streamer-r74/Makefile mjpg-streamer-r74.new/Makefile
--- mjpg-streamer-r74/Makefile	2009-02-08 14:51:37.000000000 +0100
+++ mjpg-streamer-r74.new/Makefile	2009-02-08 14:52:48.000000000 +0100
@@ -31,7 +31,6 @@
 PLUGINS = input_uvc.so
 PLUGINS += output_file.so
 PLUGINS += output_http.so
-PLUGINS += input_testpicture.so
 PLUGINS += output_autofocus.so
 PLUGINS += input_gspcav1.so
 PLUGINS += input_file.so
