diff -urN expat-1.95.8.orig/conftools/config.sub expat-1.95.8/conftools/config.sub
--- expat-1.95.8.orig/conftools/config.sub	2002-06-02 00:29:30.000000000 +0200
+++ expat-1.95.8/conftools/config.sub	2008-04-14 18:27:28.000000000 +0200
@@ -251,7 +251,7 @@
 	| a29k \
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
-	| arc | arm | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr \
+	| arc | arm | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
 	| c4x | clipper \
 	| d10v | d30v | dsp16xx \
 	| fr30 \
@@ -305,7 +305,7 @@
 	| alpha64-* | alpha64ev[4-8]-* | alpha64ev56-* | alpha64ev6[78]-* \
 	| alphapca5[67]-* | alpha64pca5[67]-* | arc-* \
 	| arm-*  | armbe-* | armle-* | armv*-* \
-	| avr-* \
+	| avr-* | avr32-* \
 	| bs2000-* \
 	| c[123]* | c30-* | [cjt]90-* | c54x-* \
 	| clipper-* | cydra-* \
