--- a/doc/click-elem2man
+++ b/doc/click-elem2man
@@ -227,14 +227,15 @@ sub nroff_fixfP ($$) {
 }
 
 sub nroff_manref ($$$$) {
-    map { $_ = "\"$_\"" if (s/\"/\\(dq/g) } @_;
+    my @x = @_;
+    map { $_ = "\"$_\"" if (s/\"/\\(dq/g) } @x;
     my($x);
-    if ($_[0] eq "") {
-	$x = "\n.M " . $_[1] . " " . $_[2];
+    if ($x[0] eq "") {
+	$x = "\n.M " . $x[1] . " " . $x[2];
     } else {
-	$x = "\n.RM " . $_[0] . " " . $_[1] . " " . $_[2];
+	$x = "\n.RM " . $x[0] . " " . $x[1] . " " . $x[2];
     }
-    $x .= " " . $_[3] if $_[3] ne "";
+    $x .= " " . $x[3] if $x[3] ne "";
     $x . "\n";
 }
 
