---
 m4/click.m4 |    8 --------
 1 file changed, 8 deletions(-)

--- wing-20100903.orig/m4/click.m4
+++ wing-20100903/m4/click.m4
@@ -235,14 +235,6 @@ AC_DEFUN([CLICK_CHECK_BUILD_DYNAMIC_LINK
     if test "x$ac_build_have_dlopen" = xyes -a "x$ac_build_have_dlfcn_h" = xyes; then
 	ac_build_have_dynamic_linking=yes
     fi
-    if test "x$ac_build_have_dynamic_linking" != "x$ac_have_dynamic_linking"; then
-	AC_MSG_ERROR([
-=========================================
-
-Build system and host system don't have the same dynamic linking state!
-
-=========================================])
-    fi
     AC_SUBST(BUILD_DL_LIBS)
     eval "$saver"
 ])
