Index: postgresql-8.2.5/src/port/Makefile
===================================================================
--- postgresql-8.2.5.orig/src/port/Makefile	2007-09-20 10:18:17.000000000 -0500
+++ postgresql-8.2.5/src/port/Makefile	2007-09-20 10:18:17.000000000 -0500
@@ -43,6 +43,7 @@
 
 libpgport.a: $(LIBOBJS)
 	$(AR) $(AROPT) $@ $^
+	$(RANLIB) libpgport.a
 
 thread.o: thread.c
 	$(CC) $(CFLAGS) $(CPPFLAGS) $(PTHREAD_CFLAGS) -c $<
