Index: postgresql-8.2.5/src/template/linux
===================================================================
--- postgresql-8.2.5.orig/src/template/linux	2007-09-20 10:18:17.000000000 -0500
+++ postgresql-8.2.5/src/template/linux	2007-09-20 10:18:17.000000000 -0500
@@ -1,4 +1,4 @@
 # $PostgreSQL: pgsql/src/template/linux,v 1.28 2006/03/11 04:38:41 momjian Exp $
 
 # Force _GNU_SOURCE on; plperl is broken with Perl 5.8.0 otherwise
-CPPFLAGS="-D_GNU_SOURCE"
+CPPFLAGS="$CPPFLAGS -D_GNU_SOURCE"
