Index: postgresql-8.2.5/src/bin/psql/print.h
===================================================================
--- postgresql-8.2.5.orig/src/bin/psql/print.h	2007-09-20 10:18:16.000000000 -0500
+++ postgresql-8.2.5/src/bin/psql/print.h	2007-09-20 10:18:17.000000000 -0500
@@ -88,10 +88,6 @@
 
 void		setDecimalLocale(void);
 
-#ifndef __CYGWIN__
-#define DEFAULT_PAGER "more"
-#else
 #define DEFAULT_PAGER "less"
-#endif
 
 #endif   /* PRINT_H */
