Index: postgresql-8.2.5/src/backend/utils/adt/float.c
===================================================================
--- postgresql-8.2.5.orig/src/backend/utils/adt/float.c	2007-09-20 10:18:16.000000000 -0500
+++ postgresql-8.2.5/src/backend/utils/adt/float.c	2007-09-20 10:18:17.000000000 -0500
@@ -110,7 +110,9 @@
 static int	float8_cmp_internal(float8 a, float8 b);
 
 #ifndef HAVE_CBRT
-static double cbrt(double x);
+#undef cbrt
+#define cbrt(x) pg_cbrt(x)
+static double pg_cbrt(double x);
 #endif   /* HAVE_CBRT */
 
 
@@ -2831,7 +2833,7 @@
 #ifndef HAVE_CBRT
 
 static double
-cbrt(double x)
+pg_cbrt(double x)
 {
 	int			isneg = (x < 0.0);
 	double		absx = fabs(x);
