diff --git a/configure b/configure
index 205be9c..df159e3 100755
--- a/configure
+++ b/configure
@@ -20668,8 +20668,6 @@ cat >>confdefs.h <<\_ACEOF
 #define PCAP_TIMEOUT_IGNORED
 _ACEOF
 
-
-    extra_incl=-I/usr/include/pcap
     ;;
   *-hpux10*|*-hpux11*)
 
@@ -26209,7 +26207,6 @@ if test "x$enable_flexresp" != "xno" -a "x$enable_flexresp" = "xyes"; then
     fi
 
     if test `libnet-config --cflags | wc -c` = "1"; then
-        CPPFLAGS="${CPPFLAGS} -I/usr/local/include -I/sw/include"
         LIBNET_CONFIG_BROKEN_CFLAGS="yes"
     fi
 
@@ -26220,7 +26217,6 @@ echo "$as_me: WARNING: libnet-config --libs is broken on your system.  If you" >
 echo "$as_me: WARNING: are using a precompiled package please notify the" >&2;}
         { echo "$as_me:$LINENO: WARNING: maintainer." >&5
 echo "$as_me: WARNING: maintainer." >&2;}
-        LDFLAGS="${LDFLAGS} -L/usr/local/lib -L/sw/lib"
         LIBS="${LIBS} -lnet"
     fi
 
@@ -26795,7 +26791,6 @@ if test "x$enable_react" != "xno" -a "x$enable_react" = "xyes"; then
     fi
 
     if test `libnet-config --cflags | wc -c` = "1"; then
-        CPPFLAGS="${CPPFLAGS} -I/usr/local/include -I/sw/include"
         LIBNET_CONFIG_BROKEN_CFLAGS="yes"
     fi
 
@@ -26806,7 +26801,6 @@ echo "$as_me: WARNING: libnet-config --libs is broken on your system.  If you" >
 echo "$as_me: WARNING: are using a precompiled package please notify the" >&2;}
         { echo "$as_me:$LINENO: WARNING: maintainer." >&5
 echo "$as_me: WARNING: maintainer." >&2;}
-        LDFLAGS="${LDFLAGS} -L/usr/local/lib -L/sw/lib"
         LIBS="${LIBS} -lnet"
     fi
 
