---
 configure.in |   46 ++++++++++++++++++++++++++++++++++------------
 1 file changed, 34 insertions(+), 12 deletions(-)

--- a/configure.in
+++ b/configure.in
@@ -2653,6 +2653,7 @@ AC_CHECK_LIB(c, inet_aton, [$ac_cv_prog_
 # On Tru64, chflags seems to be present, but calling it will
 # exit Python
 AC_MSG_CHECKING(for chflags)
+AC_CACHE_VAL(ac_cv_have_chflags, [
 AC_TRY_RUN([
 #include <sys/stat.h>
 #include <unistd.h>
@@ -2662,12 +2663,19 @@ int main(int argc, char*argv[])
     return 1;
   return 0;
 }
-],AC_DEFINE(HAVE_CHFLAGS, 1, Define to 1 if you have the `chflags' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+],
+ac_cv_have_chflags=yes,
+ac_cv_have_chflags=no,
+ac_cv_have_chflags=no)])
+AC_MSG_RESULT($ac_cv_have_chflags)
+if test "$ac_cv_have_chflags" = yes
+then
+  AC_DEFINE(HAVE_CHFLAGS, 1,
+  [Define to 1 if you have the `chflags' function.])
+fi
 
 AC_MSG_CHECKING(for lchflags)
+AC_CACHE_VAL(ac_cv_have_lchflags, [
 AC_TRY_RUN([
 #include <sys/stat.h>
 #include <unistd.h>
@@ -2677,10 +2685,16 @@ int main(int argc, char*argv[])
     return 1;
   return 0;
 }
-],AC_DEFINE(HAVE_LCHFLAGS, 1, Define to 1 if you have the `lchflags' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+],
+ac_cv_have_lchflags=yes,
+ac_cv_have_lchflags=no,
+ac_cv_have_lchflags=no)])
+AC_MSG_RESULT($ac_cv_have_lchflags)
+if test "$ac_cv_have_lchflags" = yes
+then
+  AC_DEFINE(HAVE_CHFLAGS, 1,
+  [Define to 1 if you have the `lchflags' function.])
+fi
 
 dnl Check if system zlib has *Copy() functions
 dnl
@@ -3781,6 +3795,7 @@ else
 fi
 
 AC_MSG_CHECKING(for %zd printf() format support)
+AC_CACHE_VAL(ac_cv_py_format_size_t, [
 AC_TRY_RUN([#include <stdio.h>
 #include <stddef.h>
 #include <string.h>
@@ -3814,10 +3829,17 @@ int main()
 	return 1;
 
     return 0;
-}],
-[AC_MSG_RESULT(yes)
- AC_DEFINE(PY_FORMAT_SIZE_T, "z", [Define to printf format modifier for Py_ssize_t])],
- AC_MSG_RESULT(no))
+}
+],
+ac_cv_py_format_size_t=yes,
+ac_cv_py_format_size_t=no,
+ac_cv_py_format_size_t=no)])
+AC_MSG_RESULT($ac_cv_py_format_size_t)
+if test "$ac_cv_pyt_format_t" = yes
+then
+  AC_DEFINE(PY_FORMAT_SIZE_T, "z",
+  [Define to printf format modifier for Py_ssize_t])
+fi
 
 AC_CHECK_TYPE(socklen_t,,
   AC_DEFINE(socklen_t,int,
