diff -ruNb Python-2.5.pris/Python/import.c Python-2.5/Python/import.c
--- Python-2.5.pris/Python/import.c	2006-09-10 23:06:23.000000000 -0500
+++ Python-2.5/Python/import.c	2007-04-19 17:09:30.209982000 -0500
@@ -853,8 +853,12 @@
 write_compiled_module(PyCodeObject *co, char *cpathname, time_t mtime)
 {
 	FILE *fp;
+   char *py_dontcompile = getenv("PYTHON_DONTCOMPILE");
 
+   if (!py_dontcompile) 
 	fp = open_exclusive(cpathname);
+   else
+     fp = NULL;
 	if (fp == NULL) {
 		if (Py_VerboseFlag)
 			PySys_WriteStderr(
