Index: curl-7.19.5/configure
===================================================================
--- curl-7.19.5.orig/configure	2009-05-25 10:09:06.000000000 +0400
+++ curl-7.19.5/configure	2009-05-25 10:09:06.000000000 +0400	
@@ -25173,16 +25173,6 @@
 rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
   fi
 
-  if test "$OPENSSL_ENABLED" = "1"; then
-    if test -n "$LIB_OPENSSL"; then
-
-       LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$LIB_OPENSSL"
-       export LD_LIBRARY_PATH
-       { $as_echo "$as_me:$LINENO: Added $LIB_OPENSSL to LD_LIBRARY_PATH" >&5
-$as_echo "$as_me: Added $LIB_OPENSSL to LD_LIBRARY_PATH" >&6;}
-    fi
-  fi
-
 fi
 
 
@@ -26177,19 +26167,6 @@
 fi
 
 
-      if test "x$USE_GNUTLS" = "xyes"; then
-        { $as_echo "$as_me:$LINENO: detected GnuTLS version $version" >&5
-$as_echo "$as_me: detected GnuTLS version $version" >&6;}
-
-        if test -n "$gtlslib"; then
-
-          LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$gtlslib"
-          export LD_LIBRARY_PATH
-          { $as_echo "$as_me:$LINENO: Added $gtlslib to LD_LIBRARY_PATH" >&5
-$as_echo "$as_me: Added $gtlslib to LD_LIBRARY_PATH" >&6;}
-        fi
-      fi
-
     fi
 
   fi
Index: curl-7.19.5/configure.ac
===================================================================
--- curl-7.19.5.orig/configure.ac	2009-05-25 10:40:49.000000000 +0400
+++ curl-7.19.5/configure.ac	2009-05-25 10:40:49.000000000 +0400
@@ -1340,18 +1340,6 @@
     ])
   fi
 
-  if test "$OPENSSL_ENABLED" = "1"; then
-    if test -n "$LIB_OPENSSL"; then
-       dnl when the ssl shared libs were found in a path that the run-time
-       dnl linker doesn't search through, we need to add it to LD_LIBRARY_PATH
-       dnl to prevent further configure tests to fail due to this
-
-       LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$LIB_OPENSSL"
-       export LD_LIBRARY_PATH
-       AC_MSG_NOTICE([Added $LIB_OPENSSL to LD_LIBRARY_PATH])
-    fi
-  fi
-
 fi
 
 dnl **********************************************************************
@@ -1615,21 +1603,6 @@
          CPPFLAGS="$CLEANCPPFLAGS"
        ])
 
-      if test "x$USE_GNUTLS" = "xyes"; then
-        AC_MSG_NOTICE([detected GnuTLS version $version])
-
-        if test -n "$gtlslib"; then
-          dnl when shared libs were found in a path that the run-time
-          dnl linker doesn't search through, we need to add it to
-          dnl LD_LIBRARY_PATH to prevent further configure tests to fail
-          dnl due to this
-
-          LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$gtlslib"
-          export LD_LIBRARY_PATH
-          AC_MSG_NOTICE([Added $gtlslib to LD_LIBRARY_PATH])
-        fi
-      fi
-
     fi
 
   fi dnl GNUTLS not disabled
