--- apcupsd-3.14.2/Makefile.in-orig	2008-11-29 00:54:14.000000000 -0800
+++ apcupsd-3.14.2/Makefile.in	2008-11-29 00:59:59.000000000 -0800
@@ -372,7 +372,7 @@
 	(cd examples; $(MAKE) clean)
 
 clean: example_clean targetclean
-	find . -type f -size 0 -exec $(RMF) {} \;
+	find . -type f -size 0 ! -name '.prepared_*' -exec $(RMF) {} \;
 
 example_distclean: example_clean
 	(cd examples; $(MAKE) distclean)
