Index: usbutils-002/Makefile.am
===================================================================
--- usbutils-002.orig/Makefile.am	2011-04-14 14:33:22.000000000 +0300
+++ usbutils-002/Makefile.am	2011-04-14 14:33:32.000000000 +0300
@@ -6,9 +6,6 @@
 
 data_DATA = \
 	usb.ids
-if HAVE_ZLIB
-data_DATA += usb.ids.gz
-endif
 
 bin_PROGRAMS = \
 	lsusb
@@ -35,6 +32,12 @@
 lsusb_LDADD = \
 	$(LIBUSB_LIBS)
 
+if HAVE_ZLIB
+data_DATA += usb.ids.gz
+lsusb_CPPFLAGS += -DHAVE_LIBZ
+lsusb_LDADD += -lz
+endif
+
 man_MANS = \
 	lsusb.8	\
 	usb-devices.1
