diff --git a/xfsettingsd/registry.c b/xfsettingsd/registry.c
index e868685..d2da4a1 100644
--- a/xfsettingsd/registry.c
+++ b/xfsettingsd/registry.c
@@ -302,13 +302,15 @@ xsettings_registry_store_xrdb(XSettingsRegistry *registry)
     if (G_LIKELY (filename))
     {
         /* create file contents */
+        const gchar *xft_rgba = g_value_get_string (&properties[XSETTING_ENTRY_XFT_RGBA].value);
+        
         string = g_string_sized_new (80);
         g_string_append_printf (string, "Xft.antialias: %d\n"
                                         "Xft.hinting: %d\n"
                                         "Xft.rgba: %s\n",
                                         g_value_get_int (&properties[XSETTING_ENTRY_XFT_ANTIALIAS].value),
                                         g_value_get_int (&properties[XSETTING_ENTRY_XFT_HINTING].value),
-                                        g_value_get_string (&properties[XSETTING_ENTRY_XFT_RGBA].value));
+                                        xft_rgba ? xft_rgba : "");
         
         if (g_value_get_int (&properties[XSETTING_ENTRY_XFT_HINTING].value))
             g_string_append_printf (string, "Xft.hintstyle: %s\n", g_value_get_string (&properties[XSETTING_ENTRY_XFT_HINTSTYLE].value));
@@ -355,10 +357,12 @@ xsettings_registry_store_xrdb(XSettingsRegistry *registry)
     if (G_LIKELY (filename))
     {
         /* build file contents */
+        const gchar *cursor_theme_name = g_value_get_string (&properties[XSETTING_ENTRY_GTK_CURSORTHEMENAME].value);
+        
         contents = g_strdup_printf ("Xcursor.theme: %s\n"
                                     "Xcursor.theme_core: true\n"
                                     "Xcursor.size: %d\n",
-                                    g_value_get_string (&properties[XSETTING_ENTRY_GTK_CURSORTHEMENAME].value),
+                                    cursor_theme_name ? cursor_theme_name : "",
                                     g_value_get_int (&properties[XSETTING_ENTRY_GTK_CURSORTHEMESIZE].value));
         
         /* write the contents to the file */
