diff -urN radvd-1.0/scanner.l radvd-1.0.new/scanner.l
--- radvd-1.0/scanner.l	2006-08-24 13:39:00.000000000 +0200
+++ radvd-1.0.new/scanner.l	2006-12-09 17:40:06.000000000 +0100
@@ -34,7 +34,7 @@
 addr3		({hexdigit}{1,4}":"){7}{hexdigit}{1,4}
 addr		({addr1}|{addr2}|{addr3}|"::")
 whitespace	([ \t])+
-string		[a-z]([a-z]|{digit}|_)*([:.]{digit}+)?
+string		[a-z]([a-z]|{digit}|_|\-)*([:.]{digit}+)?
 %%
 
 #.*$			{/* ignore comments */}
