--- a/source3/smbd/server_exit.c
+++ b/source3/smbd/server_exit.c
@@ -132,7 +132,9 @@ static void exit_server_common(enum serv
 #endif
 
 	if (am_parent) {
+#ifdef WKSSVC_SUPPORT
 		rpc_wkssvc_shutdown();
+#endif
 #ifdef ACTIVE_DIRECTORY
 		rpc_dssetup_shutdown();
 #endif
--- a/source3/rpc_server/rpc_ep_setup.c
+++ b/source3/rpc_server/rpc_ep_setup.c
@@ -982,6 +982,7 @@ static bool dssetup_init_cb(void *ptr)
 }
 #endif
 
+#ifdef WKSSVC_SUPPORT
 static bool wkssvc_init_cb(void *ptr)
 {
 	struct dcesrv_ep_context *ep_ctx =
@@ -1029,6 +1030,7 @@ static bool wkssvc_init_cb(void *ptr)
 
 	return true;
 }
+#endif
 
 bool dcesrv_ep_setup(struct tevent_context *ev_ctx,
 		     struct messaging_context *msg_ctx)
@@ -1202,12 +1204,14 @@ bool dcesrv_ep_setup(struct tevent_conte
 	}
 #endif
 
+#ifdef WKSSVC_SUPPORT
 	wkssvc_cb.init         = wkssvc_init_cb;
 	wkssvc_cb.shutdown     = NULL;
 	wkssvc_cb.private_data = ep_ctx;
 	if (!NT_STATUS_IS_OK(rpc_wkssvc_init(&wkssvc_cb))) {
 		return false;
 	}
+#endif
 
 	return true;
 }
--- a/source3/librpc/rpc/rpc_common.c
+++ b/source3/librpc/rpc/rpc_common.c
@@ -109,9 +109,11 @@ static bool initialize_interfaces(void)
 	if (!smb_register_ndr_interface(&ndr_table_srvsvc)) {
 		return false;
 	}
+#ifdef WKSSVC_SUPPORT
 	if (!smb_register_ndr_interface(&ndr_table_wkssvc)) {
 		return false;
 	}
+#endif
 #ifdef WINREG_SUPPORT
 	if (!smb_register_ndr_interface(&ndr_table_winreg)) {
 		return false;
--- a/source3/rpcclient/rpcclient.c
+++ b/source3/rpcclient/rpcclient.c
@@ -634,7 +634,9 @@ static struct cmd_set *rpcclient_command
 	epmapper_commands,
 	shutdown_commands,
  	test_commands,
+#ifdef WKSSVC_SUPPORT
 	wkssvc_commands,
+#endif
 	ntsvcs_commands,
 	drsuapi_commands,
 	eventlog_commands,
