--- a/gcc/configure
+++ b/gcc/configure
@@ -4766,7 +4766,7 @@ if test "${with_sysroot+set}" = set; the
  esac
 
  TARGET_SYSTEM_ROOT_DEFINE='-DTARGET_SYSTEM_ROOT=\"$(TARGET_SYSTEM_ROOT)\"'
- CROSS_SYSTEM_HEADER_DIR='$(TARGET_SYSTEM_ROOT)$(NATIVE_SYSTEM_HEADER_DIR)'
+ CROSS_SYSTEM_HEADER_DIR='$(TARGET_SYSTEM_ROOT)/include'
 
  if test "x$exec_prefix" = xNONE; then
   if test "x$prefix" = xNONE; then
--- a/gcc/configure.ac
+++ b/gcc/configure.ac
@@ -576,7 +576,7 @@ AC_ARG_WITH(sysroot,
  esac
    
  TARGET_SYSTEM_ROOT_DEFINE='-DTARGET_SYSTEM_ROOT=\"$(TARGET_SYSTEM_ROOT)\"'
- CROSS_SYSTEM_HEADER_DIR='$(TARGET_SYSTEM_ROOT)$(NATIVE_SYSTEM_HEADER_DIR)'
+ CROSS_SYSTEM_HEADER_DIR='$(TARGET_SYSTEM_ROOT)/include'
 
  if test "x$exec_prefix" = xNONE; then
   if test "x$prefix" = xNONE; then
