diff -urN xorg-server-X11R7.2-1.2.0.orig/hw/xnest/Makefile.in xorg-server-X11R7.2-1.2.0/hw/xnest/Makefile.in
--- xorg-server-X11R7.2-1.2.0.orig/hw/xnest/Makefile.in	2007-09-15 16:52:43.000000000 +0200
+++ xorg-server-X11R7.2-1.2.0/hw/xnest/Makefile.in	2007-09-15 17:04:07.000000000 +0200
@@ -598,7 +598,7 @@
 Xnest_SOURCES = $(SRCS)
 Xnest_LDADD = $(XORG_CORE_LIBS) \
               $(XNEST_LIBS) \
-              $(XNESTMODULES_LIBS)
+              $(XNESTMODULES_LIBS)  -lz -lfreetype -lfontenc -shared
 
 Xnest_LDFLAGS = 
 AM_CFLAGS = -DHAVE_XNEST_CONFIG_H \
