diff -urN xorg-server-X11R7.2-1.4.0.orig/hw/xfree86/utils/gtf/Makefile.in xorg-server-X11R7.2-1.4.0/hw/xfree86/utils/gtf/Makefile.in
--- xorg-server-X11R7.2-1.4.0.orig/hw/xfree86/utils/gtf/Makefile.in	2007-09-14 21:23:18.000000000 +0200
+++ xorg-server-X11R7.2-1.4.0/hw/xfree86/utils/gtf/Makefile.in	2007-09-14 21:23:42.000000000 +0200
@@ -576,7 +576,7 @@
 xpconfigdir = @xpconfigdir@
 gtf_SOURCES = gtf.c
 gtf_CFLAGS = $(XORG_CFLAGS)
-gtf_LDADD = -lm
+gtf_LDADD = -lm -lnotimpl
 appmandir = $(APP_MAN_DIR)
 appman_PRE = gtf.man
 appman_DATA = $(appman_PRE:man=@APP_MAN_SUFFIX@)
