diff -ruN openldap-2.4.16-orig/clients/tools/Makefile.in openldap-2.4.16/clients/tools/Makefile.in
--- openldap-2.4.16-orig/clients/tools/Makefile.in
+++ openldap-2.4.16/clients/tools/Makefile.in
@@ -120,7 +120,7 @@
 	-$(MKDIR) $(DESTDIR)$(bindir)
 	@(								\
 	    for prg in $(PROGRAMS); do					\
-		$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 $$prg$(EXEEXT)	\
+		$(LTINSTALL) $(INSTALLFLAGS) -m 755 $$prg$(EXEEXT)	\
 		    $(DESTDIR)$(bindir);				\
 	    done							\
 	)
diff -ruN openldap-2.4.16-orig/servers/slapd/Makefile.in openldap-2.4.16-1/servers/slapd/Makefile.in
--- openldap-2.4.16-orig/servers/slapd/Makefile.in
+++ openldap-2.4.16/servers/slapd/Makefile.in
@@ -375,7 +375,7 @@
 install-slapd: FORCE
 	-$(MKDIR) $(DESTDIR)$(libexecdir)
 	-$(MKDIR) $(DESTDIR)$(localstatedir)/run
-	$(LTINSTALL) $(INSTALLFLAGS) $(STRIP) -m 755 \
+	$(LTINSTALL) $(INSTALLFLAGS) -m 755 \
 		slapd$(EXEEXT) $(DESTDIR)$(libexecdir)
 	@for i in $(SUBDIRS); do \
 	    if test -d $$i && test -f $$i/Makefile ; then \
