--- old/dnsmasq.init     2008-10-05 22:12:42.000000000 +0200
+++ new/dnsmasq.init  2009-03-25 16:50:48.000000000 +0100
@@ -52,6 +52,7 @@
	append_bool "$cfg" dbus "-1"
	append_bool "$cfg" boguspriv "-b"
	append_bool "$cfg" expandhosts "-E"
+	append_bool "$cfg" nonwildcard "-z"

	append_parm "$cfg" dnsforwardmax "-0"
	append_parm "$cfg" port "-p"
@@ -63,6 +64,7 @@
	append_parm "$cfg" "local" "-S"
	append_parm "$cfg" "leasefile" "-l"
	append_parm "$cfg" "resolvfile" "-r"
+	append_parm "$cfg" "interfaces" "-i"

	config_get leasefile $cfg leasefile
	[ -e "$leasefile" ] || touch "$leasefile"
--- old/dhcp.conf       2009-04-01 15:13:53.000000000 +0200
+++ new/dhcp.conf       2009-04-01 15:15:02.000000000 +0200
@@ -11,6 +11,8 @@
	option readethers	1
	option leasefile	'/tmp/dhcp.leases'
	option resolvfile	'/tmp/resolv.conf.auto'
+	option nonwildcard	1
+	option interfaces	br-lan

 config dhcp lan
 	option interface	lan
