--- a/Makefile
+++ b/Makefile
@@ -27,16 +27,12 @@
 	$(CC) $(CFLAGS) -D'VERSION=$(VERSION)' -c $<
 
 
-sslh: $(OBJS) sslh-fork sslh-select
+sslh: $(OBJS) sslh-fork
 
 sslh-fork: $(OBJS) sslh-fork.o Makefile common.h
 	$(CC) $(CFLAGS) -D'VERSION=$(VERSION)' -o sslh-fork sslh-fork.o $(OBJS) $(LIBS)
 	strip sslh-fork
 
-sslh-select: $(OBJS) sslh-select.o Makefile common.h 
-	$(CC) $(CFLAGS) -D'VERSION=$(VERSION)' -o sslh-select sslh-select.o $(OBJS) $(LIBS)
-	strip sslh-select
-
 
 $(MAN): sslh.pod Makefile
 	pod2man --section=8 --release=$(VERSION) --center=" " sslh.pod | gzip -9 - > $(MAN)
@@ -58,7 +54,7 @@
 	update-rc.d sslh remove
 
 clean:
-	rm -f sslh-fork sslh-select $(MAN) *.o
+	rm -f sslh-fork $(MAN) *.o
 
 tags:
 	ctags -T *.[ch]
