--- cups-1.1.23/systemv/Makefile.old	Sun Oct 15 03:33:34 2006
+++ cups-1.1.23/systemv/Makefile	Sun Oct 15 03:33:55 2006
@@ -77,7 +77,7 @@
 	$(INSTALL_BIN) lp $(BINDIR)
 	$(INSTALL_BIN) lpoptions $(BINDIR)
 	$(INSTALL_BIN) lpstat $(BINDIR)
-	-$(INSTALL_BIN) -m 4755 -o $(CUPS_USER) -g $(CUPS_GROUP) lppasswd $(BINDIR)
+	-$(INSTALL_BIN) lppasswd $(BINDIR)
 	if test ! -x $(BINDIR)/lppasswd; then \
 		chmod 755 $(BINDIR)/lppasswd; \
 	fi
