diff -pruN clearsilver-0.10.5.orig/cgi/cgi.c clearsilver-0.10.5/cgi/cgi.c
--- clearsilver-0.10.5.orig/cgi/cgi.c	2008-05-29 17:59:56.000000000 +0200
+++ clearsilver-0.10.5/cgi/cgi.c	2008-05-29 18:02:51.000000000 +0200
@@ -642,7 +642,7 @@ NEOERR *cgi_parse (CGI *cgi)
 
   if (!strcmp(method, "POST"))
   {
-    if (type && !strcmp(type, "application/x-www-form-urlencoded"))
+    if (type && !strncmp(type, "application/x-www-form-urlencoded", 33))
     {
       err = _parse_post_form(cgi);
       if (err != STATUS_OK) return nerr_pass (err);
