--- boost_1_38_0/boost/python/detail/config.hpp	2007-11-25 19:07:19.000000000 +0100
+++ boost_1_38_0/boost/python/detail/config.hpp	2010-02-26 02:06:26.000000000 +0100
@@ -111,7 +111,7 @@
 # define BOOST_PYTHON_OFFSETOF(s_name, s_member) \
         ((size_t)__INTADDR__(&(((s_name *)0)->s_member)))
 #else
-# define BOOST_PYTHON_OFFSETOF offsetof
+# define BOOST_PYTHON_OFFSETOF __builtin_offsetof
 #endif
 
 //  enable automatic library variant selection  ------------------------------// 
