--- a/specs/troffrules.in	2010-01-15 01:11:36.000000000 +0000
+++ b/specs/troffrules.in	2011-01-24 21:37:01.000000000 +0000
@@ -49,7 +49,9 @@
 
 .ms.ps:
 	-$(AM_V_GEN) $(GROFF) -Tps $(GROFF_FLAGS) $< 2> index.$@.raw > $@
-	@if grep '^[^1-9.]' index.$@.raw | grep -v warning; then exit 1; \
+	@if grep '^[^1-9.]' index.$@.raw | grep -v warning | \
+	    grep -v 'table wider than line width'; \
+	 then exit 1; \
 	 else test $$? -le 1; fi
 
 .ms.txt:
