diff --git a/src/lib/ecore_evas/Makefile.am b/src/lib/ecore_evas/Makefile.am
index 2152d47..39b13df 100644
--- a/src/lib/ecore_evas/Makefile.am
+++ b/src/lib/ecore_evas/Makefile.am
@@ -1,7 +1,7 @@
 MAINTAINERCLEANFILES = Makefile.in
 
 if BUILD_ECORE_X
 ECORE_X_INC = -I$(top_srcdir)/src/lib/ecore_x
-ECORE_X_LIB = $(top_builddir)/src/lib/ecore_x/libecore_x.la
+ECORE_X_LIB = $(top_builddir)/src/lib/ecore_x/libecore_x.la -lX11
 else
 ECORE_X_INC =
