--- a/lib/Makefile.in
+++ b/lib/Makefile.in
@@ -176,6 +176,7 @@ SASL_KRB_LIB = @SASL_KRB_LIB@
 SASL_MECHS = @SASL_MECHS@
 SASL_STATIC_LIBS = @SASL_STATIC_LIBS@
 SASL_STATIC_OBJS = @SASL_STATIC_OBJS@
+SASL_STATIC_SRC = $(patsubst %.o, %.c, $(SASL_STATIC_OBJS))
 SASL_STATIC_SRCS = @SASL_STATIC_SRCS@
 SASL_UTIL_HEADERS_EXTRA = @SASL_UTIL_HEADERS_EXTRA@
 SASL_UTIL_LIBS_EXTRA = @SASL_UTIL_LIBS_EXTRA@
@@ -646,7 +647,7 @@ libsasl2.a: libsasl2.la $(SASL_STATIC_OB
 	rm -f $@
 	ln -s .libs/$@ $@
 
-$(SASL_STATIC_OBJS): linksrcs
+$(SASL_STATIC_SRC): linksrcs
 
 linksrcs:
 	-ln -s $(SASL_STATIC_SRCS) .
