--- a/Makefile
+++ b/Makefile
@@ -4,14 +4,15 @@
 # (c) 2007-2011 Thomas Bernard
 # http://miniupnp.free.fr/libnatpmp.html
 
-OS = $(shell uname -s)
+OS = Linux
 CC = gcc
 INSTALL = install
 
 # APIVERSION is used in soname
 APIVERSION = 1
 LDFLAGS = -Wl,--no-undefined
-CFLAGS = -O -fPIC -Wall -DENABLE_STRNATPMPERR
+COPTS = -O
+CFLAGS = $(COPTS) -fPIC -Wall -DENABLE_STRNATPMPERR
 
 LIBOBJS = natpmp.o getgateway.o
 
