diff -ruN python-evas-r38564.orig/setup.py python-evas-r38564/setup.py
--- python-evas-r38564.orig/setup.py	2009-01-13 10:17:27.000000000 +0100
+++ python-evas-r38564/setup.py	2009-01-13 10:24:35.000000000 +0100
@@ -1,10 +1,7 @@
 import sys
 import os
 
-from ez_setup import use_setuptools
-use_setuptools('0.6c9')
-
-from setuptools import setup, find_packages, Extension
+from distutils.core import setup, Extension
 from distutils.command.install_headers import install_headers
 from distutils.sysconfig import get_python_inc
 import subprocess
@@ -142,7 +139,7 @@
       long_description=long_description,
       keywords='wrapper binding enlightenment graphics raster evas canvas',
       classifiers=trove_classifiers,
-      packages=find_packages(),
+      packages=['evas'],
       headers=headers,
       ext_modules=[evasmodule],
       zip_safe=False,
