--- a/extensions/GNUmakefile.in
+++ b/extensions/GNUmakefile.in
@@ -2,8 +2,8 @@
 
 top_srcdir      := @top_srcdir@
 srcdir          := @srcdir@
-abstop_srcdir   := $(shell readlink -f ${top_srcdir})
-abssrcdir       := $(shell readlink -f ${srcdir})
+abstop_srcdir   := $(shell cd ${top_srcdir} && pwd)
+abssrcdir       := $(shell cd ${srcdir} && pwd)
 
 ifeq (${abstop_srcdir},)
 $(error Path resolution of ${top_srcdir} failed)
--- a/extensions/ipset/GNUmakefile.in
+++ b/extensions/ipset/GNUmakefile.in
@@ -3,8 +3,8 @@
 top_srcdir      := @top_srcdir@
 srcdir          := @srcdir@
 datarootdir     := @datarootdir@
-abstop_srcdir   := $(shell readlink -f ${top_srcdir})
-abssrcdir       := $(shell readlink -f ${srcdir})
+abstop_srcdir   := $(shell cd ${top_srcdir} && pwd)
+abssrcdir       := $(shell cd ${srcdir} && pwd)
 
 ifeq (${abstop_srcdir},)
 $(error Path resolution of ${top_srcdir} failed)
