diff -ru chrony-1.21.orig/io_linux.h chrony-1.21/io_linux.h
--- chrony-1.21.orig/io_linux.h	2005-08-11 22:32:54.000000000 +0200
+++ chrony-1.21/io_linux.h	2006-08-08 22:24:43.000000000 +0200
@@ -24,6 +24,15 @@
 #define CHRONY_IOC_NONE        1U
 #define CHRONY_IOC_READ        2U
 #define CHRONY_IOC_WRITE       4U
+#elif defined(__mips__) || (__powerpc__) || (__arm__)
+#define CHRONY_IOC_NRBITS	8
+#define CHRONY_IOC_TYPEBITS	8
+#define CHRONY_IOC_SIZEBITS	13
+#define CHRONY_IOC_DIRBITS	3
+
+#define CHRONY_IOC_NONE        1U
+#define CHRONY_IOC_READ        2U
+#define CHRONY_IOC_WRITE       4U
 #else
 #error "I don't know the values of the _IOC_* constants for your architecture"
 #endif
