--- chrony-1.21/addrfilt.c.orig	2005-08-11 22:32:54.000000000 +0200
+++ chrony-1.21/addrfilt.c	2005-09-04 11:05:54.000000000 +0200
@@ -45,7 +45,7 @@
 
 struct _TableNode;
 
-typedef struct _TableNode ExtendedTable[TABLE_SIZE];
+typedef struct _TableNode *ExtendedTable;
 
 typedef enum {DENY, ALLOW, AS_PARENT} State;
 
@@ -124,7 +124,7 @@
 
   if (node->extended == NULL) {
 
-    node->extended = MallocNew(ExtendedTable);
+    node->extended = MallocArray(ExtendedTable, TABLE_SIZE);
 
     for (i=0; i<TABLE_SIZE; i++) {
       child_node = &((*(node->extended))[i]);
