diff -urN libnetfilter_queue-0.0.13/Makefile.am libnetfilter_queue-0.0.13.new/Makefile.am
--- libnetfilter_queue-0.0.13/Makefile.am	2006-12-21 12:33:43.000000000 +0100
+++ libnetfilter_queue-0.0.13.new/Makefile.am	2008-02-05 11:50:33.000000000 +0100
@@ -2,7 +2,7 @@
 
 EXTRA_DIST = $(man_MANS) 
 
-SUBDIRS = include src utils
+SUBDIRS = include src
 
 man_MANS = #nfnetlink_queue.3 nfnetlink_queue.7
 
diff -urN libnetfilter_queue-0.0.13/Makefile.in libnetfilter_queue-0.0.13.new/Makefile.in
--- libnetfilter_queue-0.0.13/Makefile.in	2006-12-21 12:40:47.000000000 +0100
+++ libnetfilter_queue-0.0.13.new/Makefile.in	2008-02-05 11:50:27.000000000 +0100
@@ -149,7 +149,7 @@
 
 EXTRA_DIST = $(man_MANS) 
 
-SUBDIRS = include src utils
+SUBDIRS = include src
 
 man_MANS = #nfnetlink_queue.3 nfnetlink_queue.7
 
