diff -urN bluez-utils-3.24.orig/input/device.c bluez-utils-3.24/input/device.c
--- bluez-utils-3.24/input/device.orig	2007-12-08 20:04:42.000000000 +0100
+++ bluez-utils-3.24/input/device.c	2008-01-16 09:20:41.000000000 +0100
@@ -25,6 +25,8 @@
 #include <config.h>
 #endif
 
+#define ENOKEY          161     /* Required key not available */
+
 #include <stdlib.h>
 #include <errno.h>
 #include <fcntl.h>
diff -urN bluez-utils-3.24.orig/input/storage.c bluez-utils-3.24/input/storage.c
--- bluez-utils-3.24/input/storage.orig	2007-10-05 09:17:06.000000000 +0200
+++ bluez-utils-3.24/input/storage.c	2008-01-16 09:26:23.000000000 +0100
@@ -25,6 +25,8 @@
 #include <config.h>
 #endif
 
+#define ENOKEY          161     /* Required key not available */
+
 #include <stdio.h>
 #include <errno.h>
 #include <ctype.h>
