diff -urN gtk+-2.14.3.orig/gdk/Makefile.in gtk+-2.14.3/gdk/Makefile.in
--- gtk+-2.14.3.orig/gdk/Makefile.in	2008-10-03 23:14:22.000000000 +0200
+++ gtk+-2.14.3/gdk/Makefile.in	2008-10-03 23:19:30.000000000 +0200
@@ -424,8 +424,8 @@
 	&& DISPLAY=:$$XID && export DISPLAY
 
 
-SUBDIRS = $(gdktarget) . tests
-DIST_SUBDIRS = win32 x11 quartz directfb tests
+SUBDIRS = $(gdktarget) .
+DIST_SUBDIRS = x11 directfb
 
 INCLUDES = \
 	-DG_LOG_DOMAIN=\"Gdk\"		\
diff -urN gtk+-2.14.3.orig/gtk/Makefile.in gtk+-2.14.3/gtk/Makefile.in
--- gtk+-2.14.3.orig/gtk/Makefile.in	2008-10-03 23:14:06.000000000 +0200
+++ gtk+-2.14.3/gtk/Makefile.in	2008-10-03 23:20:08.000000000 +0200
@@ -465,8 +465,8 @@
 @USE_QUARTZ_TRUE@GTK_PRINT_PREVIEW_COMMAND = "open -a /Applications/Preview.app %f"
 @USE_QUARTZ_FALSE@GTK_PRINT_PREVIEW_COMMAND = "evince --unlink-tempfile --preview --print-settings %s %f"
 
-SUBDIRS = theme-bits . tests
-DIST_SUBDIRS = theme-bits tests
+SUBDIRS = theme-bits .
+DIST_SUBDIRS = theme-bits
 @HAVE_CUPS_FALSE@GTK_PRINT_BACKENDS = file,lpr
 
 @HAVE_CUPS_TRUE@GTK_PRINT_BACKENDS = file,cups
diff -urN gtk+-2.14.3.orig/Makefile.in gtk+-2.14.3/Makefile.in
--- gtk+-2.14.3.orig/Makefile.in	2008-10-03 23:13:54.000000000 +0200
+++ gtk+-2.14.3/Makefile.in	2008-10-03 23:18:13.000000000 +0200
@@ -516,7 +516,7 @@
 	&& DISPLAY=:$$XID && export DISPLAY
 
 
-SRC_SUBDIRS = gdk-pixbuf gdk gtk modules demos tests perf contrib
+SRC_SUBDIRS = gdk-pixbuf gdk gtk modules contrib
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros
 
 # require automake 1.4
