diff --git a/linux/include/openswan/ipsec_kversion.h b/linux/include/openswan/ipsec_kversion.h
index 8ebf99a..27fe871 100644
--- a/linux/include/openswan/ipsec_kversion.h
+++ b/linux/include/openswan/ipsec_kversion.h
@@ -332,7 +332,7 @@
 
 #endif
 
-#ifndef NETDEV_TX_BUSY
+#if !defined(NETDEV_TX_BUSY) && LINUX_VERSION_CODE < KERNEL_VERSION(2,6,32)
 # ifdef NETDEV_XMIT_CN
 #  define NETDEV_TX_BUSY NETDEV_XMIT_CN
 # else
