diff -ruN dropbear-0.48-old/scp.c dropbear-0.48-new/scp.c
--- dropbear-0.48-old/scp.c	2006-03-09 16:06:03.000000000 +0100
+++ dropbear-0.48-new/scp.c	2006-03-11 01:19:26.000000000 +0100
@@ -78,6 +78,26 @@
 #include "scpmisc.h"
 #include "progressmeter.h"
 
+#ifndef TIMEVAL_TO_TIMESPEC
+#define	TIMEVAL_TO_TIMESPEC(tv, ts) {					\
+	(ts)->tv_sec = (tv)->tv_sec;					\
+	(ts)->tv_nsec = (tv)->tv_usec * 1000;				\
+}
+#endif
+
+#ifndef timersub
+#define timersub(tvp, uvp, vvp)					 \
+	do {								\
+		(vvp)->tv_sec = (tvp)->tv_sec - (uvp)->tv_sec;	  \
+		(vvp)->tv_usec = (tvp)->tv_usec - (uvp)->tv_usec;   \
+		if ((vvp)->tv_usec < 0) {			   \
+			(vvp)->tv_sec--;				\
+			(vvp)->tv_usec += 1000000;		  \
+		}						   \
+	} while (/* CONSTCOND */ 0)
+#endif /* timersub */
+
+
 void bwlimit(int);
 
 /* Struct for addargs */
