---
 src/Makefile.in |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- gpxview-0.9.1.orig/src/Makefile.in
+++ gpxview-0.9.1/src/Makefile.in
@@ -33,15 +33,15 @@ OBJS = main.o gpx.o goto.o gps.o misc.o
 
 DEFS = -DVERSION=\"$(VERSION)\" -DPREFIX=\"$(prefix)\" -DICONPATH=\"$(datadir)/$(APP)/\" -DAPP=\"$(APP)\"
 CFLAGS = -Wall -O2 $(DEFS) @CFLAGS@
-LDFLAGS = @LIBS@ -lz
+LDFLAGS = @LDFLAGS@ @LIBS@ -lz
 
 all: $(APP)
 
 %.o: %.c Makefile
-	gcc $(CFLAGS) -o $@ -c $<
+	$(CC) $(CFLAGS) -o $@ -c $<
 
 $(APP): $(OBJS)
-	gcc -o $@ $(OBJS) $(LDFLAGS)
+	$(CC) $(CFLAGS) -o $@ $(OBJS) $(LDFLAGS)
 
 install:
 	install -d $(DESTDIR)$(prefix)$(bindir)
