diff -Nru asterisk-1.6.2.0-beta2.org/build_tools/menuselect-deps.in asterisk-1.6.2.0-beta2/build_tools/menuselect-deps.in
--- asterisk-1.6.2.0-beta2.org/build_tools/menuselect-deps.in	2009-01-15 21:18:53.000000000 +0100
+++ asterisk-1.6.2.0-beta2/build_tools/menuselect-deps.in	2009-06-04 19:37:42.000000000 +0200
@@ -11,7 +11,7 @@
 GTK=@PBX_GTK@
 H323=@PBX_H323@
 HOARD=@PBX_HOARD@
-ICONV=@PBX_ICONV@
+ICONV=0
 IKSEMEL=@PBX_IKSEMEL@
 IMAP_TK=@PBX_IMAP_TK@
 IODBC=@PBX_IODBC@
