--- wifidog-1.1.5/src/firewall.c	2008-08-29 22:44:57.000000000 +0200
+++ wifidog-1.1.5-fixed/src/firewall.c	2008-08-29 22:45:46.000000000 +0200
@@ -135,7 +135,7 @@
 
 	 /* Find ip, copy mac in reply */
 	 reply = NULL;
-    while (!feof(proc) && (fscanf(proc, " %15[0-9.] %*s %*s %17[A-F0-9:] %*s %*s", ip, mac) == 2)) {
+    while (!feof(proc) && (fscanf(proc, " %15[0-9.] %*s %*s %17[A-Fa-f0-9:] %*s %*s", ip, mac) == 2)) {
 		  if (strcmp(ip, req_ip) == 0) {
 				reply = safe_strdup(mac);
 				break;
