--- a/include/cstring
+++ b/include/cstring
@@ -47,31 +47,6 @@ namespace std{
 	using ::strstr;
 	using ::strtok;
 	using ::strxfrm;
-
-
-	//Extra definitions required in c++ spec
-
-
-	inline void* memchr(void* s, int c, size_t n){
-		return memchr(const_cast<const void *>(s), c, n);
-	}
-
-	inline char* strchr(char* s, int c){
-		return strchr(const_cast<const char *>(s), c);
-	}
-
-	inline char* strpbrk(char* s1, const char* s2){
-		return strpbrk(const_cast<const char *>(s1), s2);
-	}
-
-	inline char* strrchr(char* s, int c){
-		return strrchr(const_cast<const char *>(s), c);
-	}
-
-	inline char* strstr(char* s1, const char* s2){
-		return strstr(const_cast<const char *>(s1), s2);
-	}
-
 }
 
 #endif
