diff -urN pygobject-2.14.0.orig/gobject/Makefile.in pygobject-2.14.0/gobject/Makefile.in
--- pygobject-2.14.0.orig/gobject/Makefile.in	2008-05-22 21:54:05.000000000 +0100
+++ pygobject-2.14.0/gobject/Makefile.in	2008-05-22 21:58:47.000000000 +0100
@@ -1040,6 +1040,7 @@
 	rm -f constants.py
 	cp $(srcdir)/constants.py.in constants.py
 	chmod 644 constants.py
+	gcc $(GLIB_CFLAGS) generate-constants.c -o generate-constants
 	$(top_builddir)/gobject/generate-constants$(EXEEXT) >> constants.py
 	chmod 444 constants.py
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
