Index: squid-2.6.STABLE14/src/ssl_support.c
===================================================================
--- squid-2.6.STABLE14.orig/src/ssl_support.c	2007-07-25 16:04:15.000000000 -0500
+++ squid-2.6.STABLE14/src/ssl_support.c	2007-07-25 16:04:15.000000000 -0500
@@ -442,10 +442,12 @@
     ERR_clear_error();
     debug(83, 1) ("Initialising SSL.\n");
     switch (version) {
+#ifndef OPENSSL_NO_SSL2
     case 2:
 	debug(83, 5) ("Using SSLv2.\n");
 	method = SSLv2_server_method();
 	break;
+#endif
     case 3:
 	debug(83, 5) ("Using SSLv3.\n");
 	method = SSLv3_server_method();
