--- a/install.sh
+++ b/install.sh
@@ -337,13 +337,15 @@ cd manpages
 
 for f in *.5; do
     gzip -c $f > $f.gz
-    run_install -D -m 644 $f.gz ${PREFIX}/usr/share/man/man5/$f.gz
+    run_install -d ${PREFIX}/usr/share/man/man5
+    run_install -m 644 $f.gz ${PREFIX}/usr/share/man/man5/$f.gz
     echo "Man page $f.gz installed to ${PREFIX}/usr/share/man/man5/$f.gz"
 done
 
 for f in *.8; do
     gzip -c $f > $f.gz
-    run_install -D -m 644 $f.gz ${PREFIX}/usr/share/man/man8/$f.gz
+    run_install -d ${PREFIX}/usr/share/man/man8
+    run_install -m 644 $f.gz ${PREFIX}/usr/share/man/man8/$f.gz
     echo "Man page $f.gz installed to ${PREFIX}/usr/share/man/man8/$f.gz"
 done
 
