--- a/Makefile.in.orig	2010-03-13 22:20:04.000000000 +0100
+++ b/Makefile.in	2010-06-03 15:08:35.000000000 +0200
@@ -240,7 +240,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = foreign 1.2h
-SUBDIRS = config m4 term src docs $(LISPDIR) man demo tutorial share
+SUBDIRS = config m4 term src $(LISPDIR) share
 EXTRA_DIST = BUGS CodeStyle Copyright FAQ.pdf GNUmakefile INSTALL INSTALL.gnu \
 Makefile.maint PATCHLEVEL PGPKEYS PORTING README README.1ST \
 VERSION configure.vms pm3d
