--- a/drivers/mtd/devices/m25p80.c
+++ b/drivers/mtd/devices/m25p80.c
@@ -652,6 +652,8 @@ static struct flash_info __devinitdata m
 	{ "s25sl12801", 0x012018, 0x0301, 64 * 1024, 256, },
 	{ "s25fl129p0", 0x012018, 0x4d00, 256 * 1024, 64, },
 	{ "s25fl129p1", 0x012018, 0x4d01, 64 * 1024, 256, },
+	{ "s25fl016k", 0xef4015, 0, 64 * 1024, 32, SECT_4K },
+	{ "s25fl064k", 0xef4017, 0, 64 * 1024, 128, SECT_4K },
 
 	/* SST -- large erase sizes are "overlays", "sectors" are 4K */
 	{ "sst25vf040b", 0xbf258d, 0, 64 * 1024, 8, SECT_4K, },
