
--- a/source/configure
+++ b/source/configure
@@ -22548,6 +22548,17 @@ rm -f conftest.err conftest.$ac_ext
 { $as_echo "$as_me:$LINENO: result: $ac_header_preproc" >&5
 $as_echo "$ac_header_preproc" >&6; }
 
+# Force disable RPC
+$as_echo "test=$ac_header" >&6
+if test "$ac_header" = "rpc/rpc.h"; then
+    if test "$ac_header_compiler" = "yes" -o "$ac_header_preproc" = "yes"; then
+        $as_echo "RPC support force disabled by OpenWRT patch" >&5
+        $as_echo "RPC support force disabled by OpenWRT patch" >&6
+    fi
+fi
+ac_header_compiler=no
+ac_header_preproc=no
+
 # So?  What about this header?
 case $ac_header_compiler:$ac_header_preproc:$ac_c_preproc_warn_flag in
   yes:no: )
