diff -urN libsigc++-2.0.17/Makefile.am libsigc++-2.0.17.new/Makefile.am
--- libsigc++-2.0.17/Makefile.am	2005-03-09 10:26:51.000000000 +0100
+++ libsigc++-2.0.17.new/Makefile.am	2007-08-29 12:41:29.000000000 +0200
@@ -4,7 +4,7 @@
 EXTRA_DIST = autogen.sh sigc++config.h.in libsigc++-2.0.spec.in
 # ACLOCAL_FLAGS = -I scripts
 
-SUBDIRS      = sigc++ tests docs examples scripts MSVC_Net2003
+SUBDIRS      = sigc++ scripts MSVC_Net2003
 DIST_SUBDIRS = $(SUBDIRS)
 
 sigc_configdir = $(libdir)/sigc++-2.0/include
diff -urN libsigc++-2.0.17/Makefile.in libsigc++-2.0.17.new/Makefile.in
--- libsigc++-2.0.17/Makefile.in	2005-12-20 09:42:07.000000000 +0100
+++ libsigc++-2.0.17.new/Makefile.in	2007-08-29 12:41:20.000000000 +0200
@@ -205,7 +205,7 @@
 # other stuff here
 EXTRA_DIST = autogen.sh sigc++config.h.in libsigc++-2.0.spec.in
 # ACLOCAL_FLAGS = -I scripts
-SUBDIRS = sigc++ tests docs examples scripts MSVC_Net2003
+SUBDIRS = sigc++ scripts MSVC_Net2003
 DIST_SUBDIRS = $(SUBDIRS)
 sigc_configdir = $(libdir)/sigc++-2.0/include
 sigc_config_DATA = sigc++config.h
