--- a/Makefile.am
+++ b/Makefile.am
@@ -4,7 +4,7 @@
 EXTRA_DIST = autogen.sh sigc++config.h.in libsigc++-2.0.spec.in
 # ACLOCAL_FLAGS = -I scripts
 
-SUBDIRS      = sigc++ tests docs examples scripts MSVC_Net2003
+SUBDIRS      = sigc++ scripts MSVC_Net2003
 DIST_SUBDIRS = $(SUBDIRS)
 
 sigc_configdir = $(libdir)/sigc++-2.0/include
--- a/Makefile.in
+++ b/Makefile.in
@@ -206,7 +206,7 @@
 # other stuff here
 EXTRA_DIST = autogen.sh sigc++config.h.in libsigc++-2.0.spec.in
 # ACLOCAL_FLAGS = -I scripts
-SUBDIRS = sigc++ tests docs examples scripts MSVC_Net2003
+SUBDIRS = sigc++ scripts MSVC_Net2003
 DIST_SUBDIRS = $(SUBDIRS)
 sigc_configdir = $(libdir)/sigc++-2.0/include
 sigc_config_DATA = sigc++config.h
