Index: libtorrent-0.12.6_r1209/rak/allocators.h
===================================================================
--- libtorrent-0.12.6_r1209.orig/rak/allocators.h	2011-04-13 16:33:28.179339669 +0200
+++ libtorrent-0.12.6_r1209/rak/allocators.h	2011-04-13 16:33:48.454840573 +0200
@@ -41,6 +41,7 @@
 
 #include <cstddef>
 #include <limits>
+#include <stdint.h>  /* for uintptr_t */
 #include <stdlib.h>
 #include <sys/types.h>
 
