diff -urN libsigc++-2.2.3/Makefile.am libsigc++-2.2.3.new/Makefile.am
--- libsigc++-2.2.3/Makefile.am	2008-10-14 17:06:54.000000000 +0200
+++ libsigc++-2.2.3.new/Makefile.am	2008-12-14 19:24:11.000000000 +0100
@@ -4,7 +4,7 @@
 EXTRA_DIST = autogen.sh sigc++config.h.in libsigc++-2.0.spec.in
 # ACLOCAL_FLAGS = -I scripts
 
-SUBDIRS      = sigc++ tests docs examples scripts MSVC_Net2005 MSVC_Net2008
+SUBDIRS      = sigc++ scripts
 DIST_SUBDIRS = $(SUBDIRS)
 
 sigc_configdir = $(libdir)/sigc++-2.0/include
diff -urN libsigc++-2.2.3/Makefile.in libsigc++-2.2.3.new/Makefile.in
--- libsigc++-2.2.3/Makefile.in	2008-10-14 17:52:18.000000000 +0200
+++ libsigc++-2.2.3.new/Makefile.in	2008-12-14 19:24:05.000000000 +0100
@@ -210,7 +210,7 @@
 # other stuff here
 EXTRA_DIST = autogen.sh sigc++config.h.in libsigc++-2.0.spec.in
 # ACLOCAL_FLAGS = -I scripts
-SUBDIRS = sigc++ tests docs examples scripts MSVC_Net2005 MSVC_Net2008
+SUBDIRS = sigc++ scripts
 DIST_SUBDIRS = $(SUBDIRS)
 sigc_configdir = $(libdir)/sigc++-2.0/include
 sigc_config_DATA = sigc++config.h
