--- freeradius-1.1.6-old/src/modules/rules.mak	2006-07-06 18:42:57.000000000 +0200
+++ freeradius-1.1.6-new/src/modules/rules.mak	2007-04-13 03:17:33.000000000 +0200
@@ -62,10 +62,10 @@
 #
 #######################################################################
 %.lo: %.c
-	$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) $(RLM_CFLAGS) -c $<
+	$(LIBTOOL) --mode=compile $(CC) $(CFLAGS) $(CPPFLAGS) $(RLM_CFLAGS) -c $<
 
 %.lo: %.cpp
-	$(LIBTOOL) --mode=compile $(CXX) $(CFLAGS) $(RLM_CFLAGS) -c $<
+	$(LIBTOOL) --mode=compile $(CXX) $(CFLAGS) $(CPPFLAGS) $(RLM_CFLAGS) -c $<
 
 ifneq ($(TARGET),)
 #######################################################################
