---
 Makefile.pre.in |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- Python-2.6.4.orig/Makefile.pre.in
+++ Python-2.6.4/Makefile.pre.in
@@ -1118,11 +1118,11 @@ TAGS::
 # Sanitation targets -- clean leaves libraries, executables and tags
 # files, which clobber removes as well
 pycremoval:
-	find $(srcdir) -name '*.py[co]' -exec rm -f {} ';'
+	find $(srcdir) ! -path './ipkg-install/*' -name '*.py[co]' -exec rm -f {} ';'
 
 clean: pycremoval
-	find . -name '*.o' -exec rm -f {} ';'
-	find . -name '*.s[ol]' -exec rm -f {} ';'
+	find . ! -path './ipkg-install/*' -name '*.o' -exec rm -f {} ';'
+	find . ! -path './ipkg-install/*' -name '*.s[ol]' -exec rm -f {} ';'
 	find $(srcdir)/build -name 'fficonfig.h' -exec rm -f {} ';' || true
 	find $(srcdir)/build -name 'fficonfig.py' -exec rm -f {} ';' || true
 	-rm -f Lib/lib2to3/*Grammar*.pickle
