diff -urN bind-9.3.2-P1/lib/dns/Makefile.in bind-9.3.2-P1.new/lib/dns/Makefile.in
--- bind-9.3.2-P1/lib/dns/Makefile.in	2004-12-09 05:07:15.000000000 +0100
+++ bind-9.3.2-P1.new/lib/dns/Makefile.in	2006-10-16 21:38:46.000000000 +0200
@@ -156,7 +156,7 @@
 	./gen -s ${srcdir} > code.h
 
 gen: gen.c
-	${CC} ${ALL_CFLAGS} ${LDFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
+	${CC} ${ALL_CFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
 
 rbtdb64.@O@: rbtdb.c
 
