--- umurmur-0.1.3/src/Makefile.old	2009-08-26 22:00:32.000000000 +0200
+++ umurmur-0.1.3/src/Makefile	        2009-11-20 23:13:26.141571090 +0100
@@ -5,18 +5,18 @@
 CFLAGS:=$(CFLAGS) -Wall -Os -g
 LDFLAGS:=$(LDFLAGS) -lcrypto -lssl -lconfig
 
-umurmurd:$(OBJS) depend.mak
+umurmurd:$(OBJS)
 	$(CC) $(LDFLAGS) $(OBJS) -o umurmurd
 
 # remove object files and executable when user executes "make clean"
 clean:
 	rm *.o umurmurd 
 
-all: umurmur
+all: umurmurd
 
 $(OBJS): Makefile
 
 depend.mak:
 	$(CC) -M $(SRCS) > depend.mak
 
-include depend.mak
\ No newline at end of file
+
