diff -urN tinyproxy-1.6.3/src/reqs.c tinyproxy-1.6.3.new/src/reqs.c
--- tinyproxy-1.6.3/src/reqs.c	2004-08-06 18:56:55.000000000 +0200
+++ tinyproxy-1.6.3.new/src/reqs.c	2007-08-30 16:50:31.000000000 +0200
@@ -726,7 +726,7 @@
 					    request->host);
 
 			indicate_http_error(connptr, 403, "Filtered",
-					    "detail", "The request you made has been filted",
+					    "detail", "The request you made has been filtered",
 					    "url", url,
 					    NULL);
 
