Index: pulseaudio-0.9.10/configure
===================================================================
--- pulseaudio-0.9.10.orig/configure	2008-04-11 13:28:12.000000000 +0200
+++ pulseaudio-0.9.10/configure	2008-04-11 13:28:12.000000000 +0200
@@ -30281,8 +30281,7 @@
 else
 
    { { echo "$as_me:$LINENO: error: *** libatomic-ops headers not found" >&5
-echo "$as_me: error: *** libatomic-ops headers not found" >&2;}
-   { (exit 1); exit 1; }; }
+echo "$as_me: error: *** libatomic-ops headers not found" >&2;} }
 
 fi
 
