Index: pulseaudio-0.9.10/src/daemon/caps.c
===================================================================
--- pulseaudio-0.9.10.orig/src/daemon/caps.c	2008-04-11 13:28:12.000000000 +0200
+++ pulseaudio-0.9.10/src/daemon/caps.c	2008-04-11 13:28:13.000000000 +0200
@@ -131,9 +131,8 @@
     return 0;
 }
 
-int pa_drop_caps(void) {
+void pa_drop_caps(void) {
     pa_drop_root();
-    return 0;
 }
 
 #endif
