Index: pulseaudio-0.9.13/src/tests/rtstutter.c
===================================================================
--- pulseaudio-0.9.13.orig/src/tests/rtstutter.c	2008-08-19 23:25:05.000000000 +0200
+++ pulseaudio-0.9.13/src/tests/rtstutter.c	2008-10-08 16:23:38.000000000 +0200
@@ -53,7 +53,6 @@ static void* work(void *p) {
 
     CPU_ZERO(&mask);
     CPU_SET((size_t) PA_PTR_TO_INT(p), &mask);
-    pa_assert_se(pthread_setaffinity_np(pthread_self(), sizeof(mask), &mask) == 0);
 
     for (;;) {
         struct timespec now, end;
Index: pulseaudio-0.9.13/src/tests/rtstutter.c
===================================================================
--- pulseaudio-0.9.13.orig/src/tests/rtstutter.c	2008-08-19 23:25:05.000000000 +0200
+++ pulseaudio-0.9.13/src/tests/rtstutter.c	2008-10-08 16:23:38.000000000 +0200
@@ -53,7 +53,6 @@ static void* work(void *p) {
 
     CPU_ZERO(&mask);
     CPU_SET((size_t) PA_PTR_TO_INT(p), &mask);
-    pa_assert_se(pthread_setaffinity_np(pthread_self(), sizeof(mask), &mask) == 0);
 
     for (;;) {
         struct timespec now, end;
