--- squid-2.6.STABLE12/src/Makefile.in.orig	2007-05-09 16:04:01.000000000 -0500
+++ squid-2.6.STABLE12/src/Makefile.in	2007-05-09 16:06:31.000000000 -0500
@@ -1210,7 +1210,7 @@
 cf_parser.h: cf.data cf_gen$(EXEEXT)
 	rm -f cf_gen
 	gcc -c ctype.c
-	gcc -o cf_gen cf_gen.c -DNODEBUG=1 -I./ -I../ -I../include/ $(CPPFLAGS) -L../lib -lmiscutil$(shell uname -m) -lm -lnsl -Wl,ctype.o
+	gcc -o cf_gen cf_gen.c -DNODEBUG=1 -I./ -I../ -I../include/ -L../lib -lmiscutil$(shell uname -m) -lm -lnsl -Wl,ctype.o
 	./cf_gen cf.data 
 
 cf_gen_defines.h: $(srcdir)/cf_gen_defines $(srcdir)/cf.data.pre
