diff -urN libtheora-1.0beta3/Makefile.am libtheora-1.0beta3.new/Makefile.am
--- libtheora-1.0beta3/Makefile.am	2008-04-16 22:25:55.000000000 +0200
+++ libtheora-1.0beta3.new/Makefile.am	2008-10-11 15:42:38.000000000 +0200
@@ -8,7 +8,7 @@
 EXAMPLES_DIR =
 endif
 
-SUBDIRS = lib include doc tests m4 $(EXAMPLES_DIR)
+SUBDIRS = lib include m4 $(EXAMPLES_DIR)
 
 
 # we include the whole debian/ dir in EXTRA_DIST because there's a problem
diff -urN libtheora-1.0beta3/Makefile.in libtheora-1.0beta3.new/Makefile.in
--- libtheora-1.0beta3/Makefile.in	2008-04-16 22:26:15.000000000 +0200
+++ libtheora-1.0beta3.new/Makefile.in	2008-10-11 15:42:32.000000000 +0200
@@ -79,7 +79,7 @@
   distclean-recursive maintainer-clean-recursive
 ETAGS = etags
 CTAGS = ctags
-DIST_SUBDIRS = lib include doc tests m4 examples
+DIST_SUBDIRS = lib include m4 examples
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 distdir = $(PACKAGE)-$(VERSION)
 top_distdir = $(distdir)
