diff -Nru aiccu.old/unix-console/Makefile aiccu/unix-console/Makefile
--- aiccu.old/unix-console/Makefile	2006-07-25 12:20:48.000000000 +0300
+++ aiccu/unix-console/Makefile	2006-11-04 00:44:03.000000000 +0200
@@ -37,12 +37,10 @@
 # This is a console client
 CFLAGS	+= -D AICCU_CONSOLE
 
-# GnuTLS Support ?
+# GnuTLS Support ? Removed for OpenWRT.
 # Used by TIC to secure that communication
-# Currently defaultly builds only on Linux, but other platforms might easily also support it
 ifeq ($(shell uname | grep -c "Linux"),1)
-CFLAGS	+= -D _LINUX -D AICCU_GNUTLS
-LDFLAGS += -lgnutls
+CFLAGS	+= -D _LINUX
 endif
 
 # Linux
