Index: rtorrent-0.8.6_r1209/src/display/canvas.h
===================================================================
--- rtorrent-0.8.6_r1209.orig/src/display/canvas.h	2011-04-13 17:29:03.218889497 +0200
+++ rtorrent-0.8.6_r1209/src/display/canvas.h	2011-04-13 17:29:22.814923068 +0200
@@ -48,7 +48,7 @@
 public:
   typedef std::vector<Attributes> attributes_list;
 
-  Canvas(int x = 0, int y = 0, int width = 0, int height = 0);
+  Canvas(int x = 0, int y = 0, int width = 1, int height = 1);
   ~Canvas() { delwin(m_window); }
 
   void                refresh()                                               { wnoutrefresh(m_window); }
