--- a/src/command_download.cc
+++ b/src/command_download.cc
@@ -310,7 +310,7 @@
   download->download()->signal_network_log(sigc::mem_fun(control->core(), &core::Manager::push_log_complete));
   download->download()->signal_storage_error(sigc::mem_fun(control->core(), &core::Manager::push_log_complete));
 
-  if (!rpc::call_command_string("get_tracker_dump").empty())
+  if (!rpc::call_command_string("get_log.tracker").empty())
     download->download()->signal_tracker_dump(sigc::ptr_fun(&core::receive_tracker_dump));
 
   return torrent::Object();
--- a/src/core/manager.cc
+++ b/src/core/manager.cc
@@ -77,7 +77,7 @@
 
 void
 receive_tracker_dump(const std::string& url, const char* data, size_t size) {
-  const std::string& filename = rpc::call_command_string("get_tracker_dump");
+  const std::string& filename = rpc::call_command_string("get_log.tracker");
 
   if (filename.empty())
     return;
@@ -94,7 +94,7 @@
 
 void
 Manager::handshake_log(const sockaddr* sa, int msg, int err, const torrent::HashString* hash) {
-  if (!rpc::call_command_value("get_handshake_log"))
+  if (!rpc::call_command_value("get_log.handshake"))
     return;
   
   std::string peer;
