diff --git a/src/tslib.c b/src/tslib.c
index e7a2766..884167d 100644
--- a/src/tslib.c
+++ b/src/tslib.c
@@ -234,7 +234,9 @@ xf86TslibControlProc(DeviceIntPtr device, int what)
 
 		if (InitValuatorClassDeviceStruct(device,
 						  2,
+#if GET_ABI_MAJOR(ABI_XINPUT_VERSION) < 3
 						  xf86GetMotionEvents,
+#endif
 						  0, Absolute) == FALSE) {
 			ErrorF("unable to allocate Valuator class device\n");
 			return !Success;
