diff -pruN gettext-0.17.orig/gettext-tools/gnulib-lib/Makefile.in gettext-0.17/gettext-tools/gnulib-lib/Makefile.in
--- gettext-0.17.orig/gettext-tools/gnulib-lib/Makefile.in	2009-05-19 17:17:24.413508248 +0200
+++ gettext-0.17/gettext-tools/gnulib-lib/Makefile.in	2009-05-19 17:19:06.457520598 +0200
@@ -1330,7 +1330,6 @@ install-libLTLIBRARIES: $(lib_LTLIBRARIE
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(libLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(libdir)/$$f'"; \
-	    $(LIBTOOL) --mode=install $(libLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(libdir)/$$f"; \
 	  else :; fi; \
 	done
 
diff -pruN gettext-0.17.orig/gettext-tools/libgettextpo/Makefile.in gettext-0.17/gettext-tools/libgettextpo/Makefile.in
--- gettext-0.17.orig/gettext-tools/libgettextpo/Makefile.in	2009-05-19 17:17:24.125764150 +0200
+++ gettext-0.17/gettext-tools/libgettextpo/Makefile.in	2009-05-19 17:23:11.101467088 +0200
@@ -953,7 +953,6 @@ install-libLTLIBRARIES: $(lib_LTLIBRARIE
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(libLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(libdir)/$$f'"; \
-	    $(LIBTOOL) --mode=install $(libLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(libdir)/$$f"; \
 	  else :; fi; \
 	done
 
diff -pruN gettext-0.17.orig/gettext-tools/src/Makefile.in gettext-0.17/gettext-tools/src/Makefile.in
--- gettext-0.17.orig/gettext-tools/src/Makefile.in	2009-05-19 17:17:24.718249813 +0200
+++ gettext-0.17/gettext-tools/src/Makefile.in	2009-05-19 17:24:34.473723917 +0200
@@ -1337,7 +1337,6 @@ install-libLTLIBRARIES: $(lib_LTLIBRARIE
 	  if test -f $$p; then \
 	    f=$(am__strip_dir) \
 	    echo " $(LIBTOOL) --mode=install $(libLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) '$$p' '$(DESTDIR)$(libdir)/$$f'"; \
-	    $(LIBTOOL) --mode=install $(libLTLIBRARIES_INSTALL) $(INSTALL_STRIP_FLAG) "$$p" "$(DESTDIR)$(libdir)/$$f"; \
 	  else :; fi; \
 	done
 
@@ -1369,7 +1368,6 @@ install-binPROGRAMS: $(bin_PROGRAMS)
 	  ; then \
 	    f=`echo "$$p1" | sed 's,^.*/,,;$(transform);s/$$/$(EXEEXT)/'`; \
 	   echo " $(INSTALL_PROGRAM_ENV) $(LIBTOOL) --mode=install $(binPROGRAMS_INSTALL) '$$p' '$(DESTDIR)$(bindir)/$$f'"; \
-	   $(INSTALL_PROGRAM_ENV) $(LIBTOOL) --mode=install $(binPROGRAMS_INSTALL) "$$p" "$(DESTDIR)$(bindir)/$$f" || exit 1; \
 	  else :; fi; \
 	done
 
