--- a/gettext-tools/gnulib-lib/spawn.in.h
+++ b/gettext-tools/gnulib-lib/spawn.in.h
@@ -31,8 +31,10 @@
 
 /* Get definitions of 'struct sched_param' and 'sigset_t'.
    But avoid namespace pollution on glibc systems.  */
-#ifndef __GLIBC__
+#ifndef __GLIBC__ || __UCLIBC__
 # include <sched.h>
+#endif
+#ifndef __GLIBC__
 # include <signal.h>
 #endif
 
@@ -88,7 +90,11 @@ typedef struct
   pid_t _pgrp;
   sigset_t _sd;
   sigset_t _ss;
+#ifdef __UCLIBC__
+  struct __sched_param _sp;
+#else
   struct sched_param _sp;
+#endif
   int _policy;
   int __pad[16];
 } posix_spawnattr_t;
