diff -ur libgphoto.old/Makefile.am libgphoto.dev/Makefile.am
--- libgphoto.old/Makefile.am	2006-04-19 03:54:32.000000000 +0200
+++ libgphoto.dev/Makefile.am	2006-10-03 14:40:18.000000000 +0200
@@ -8,7 +8,7 @@
 EXTRA_DIST = HACKING MAINTAINERS makefile.os2 os2mak.inc TESTERS CHANGES
 
 # Note: @subdirs@ lists all the directories from AC_CONFIG_SUBDIRS()
-SUBDIRS = @subdirs@ libgphoto2 camlibs tests po packaging doc
+SUBDIRS = @subdirs@ libgphoto2 camlibs po packaging doc
 
 doc_DATA = AUTHORS COPYING README CHANGES NEWS
 
diff -ur libgphoto.old/Makefile.in libgphoto.dev/Makefile.in
--- libgphoto.old/Makefile.in	2006-06-19 21:15:59.000000000 +0200
+++ libgphoto.dev/Makefile.in	2006-10-03 14:40:23.000000000 +0200
@@ -297,7 +297,7 @@
 EXTRA_DIST = HACKING MAINTAINERS makefile.os2 os2mak.inc TESTERS CHANGES
 
 # Note: @subdirs@ lists all the directories from AC_CONFIG_SUBDIRS()
-SUBDIRS = @subdirs@ libgphoto2 camlibs tests po packaging doc
+SUBDIRS = @subdirs@ libgphoto2 camlibs po packaging doc
 doc_DATA = AUTHORS COPYING README CHANGES NEWS
 
 ########################################################################
