diff -urN speex-1.2beta3.2/libspeex/Makefile.am speex-1.2beta3.2.new/libspeex/Makefile.am
--- speex-1.2beta3.2/libspeex/Makefile.am	2008-04-10 04:36:35.000000000 +0200
+++ speex-1.2beta3.2.new/libspeex/Makefile.am	2008-07-10 22:27:47.000000000 +0200
@@ -4,7 +4,7 @@
 
 EXTRA_DIST=echo_diagnostic.m
 
-INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir) @OGG_CFLAGS@
+INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir)
 
 lib_LTLIBRARIES = libspeex.la libspeexdsp.la
 
diff -urN speex-1.2beta3.2/libspeex/Makefile.in speex-1.2beta3.2.new/libspeex/Makefile.in
--- speex-1.2beta3.2/libspeex/Makefile.in	2008-04-10 18:07:52.000000000 +0200
+++ speex-1.2beta3.2.new/libspeex/Makefile.in	2008-07-10 22:31:58.000000000 +0200
@@ -263,7 +263,7 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 EXTRA_DIST = echo_diagnostic.m
-INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir) @OGG_CFLAGS@
+INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir)
 lib_LTLIBRARIES = libspeex.la libspeexdsp.la
 
 # Sources for compilation in the library
