--- alsa-lib-1.0.10/include/alsa-symbols.h.orig	2006-03-01 21:59:33.500097817 +0100
+++ alsa-lib-1.0.10/include/alsa-symbols.h	2006-03-01 22:00:22.323650362 +0100
@@ -57,7 +57,7 @@
 	__asm__ (".set " #name "," #real);		\
 	__asm__ (".set ." #name ",." #real)
 #else
-#ifdef __alpha__
+#if defined(__alpha__) || defined(__mips__) || defined(__arm__)
 #define use_default_symbol_version(real, name, version) \
         __asm__ (".weak " #name); \
         __asm__ (#name " = " #real)
