diff -urN aircrack-ng-0.6.2/src/aireplay-ng.c aircrack-ng-0.6.2.new/src/aireplay-ng.c
--- aircrack-ng-0.6.2/src/aireplay-ng.c	2006-09-24 20:28:42.000000000 +0200
+++ aircrack-ng-0.6.2.new/src/aireplay-ng.c	2007-01-05 22:54:08.000000000 +0100
@@ -29,6 +29,10 @@
 #include <sys/wait.h>
 #include <sys/time.h>
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,19)
+#include <net/if.h>
+#endif
 #include <netpacket/packet.h>
 #include <linux/if_ether.h>
 #include <linux/wireless.h>
diff -urN aircrack-ng-0.6.2/src/airodump-ng.c aircrack-ng-0.6.2.new/src/airodump-ng.c
--- aircrack-ng-0.6.2/src/airodump-ng.c	2006-09-09 16:38:26.000000000 +0200
+++ aircrack-ng-0.6.2.new/src/airodump-ng.c	2007-01-05 22:55:00.000000000 +0100
@@ -25,6 +25,10 @@
 #include <sys/wait.h>
 #include <sys/time.h>
 
+#include <linux/version.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,19)
+#include <net/if.h>
+#endif
 #include <netpacket/packet.h>
 #include <linux/if_ether.h>
 #include <linux/wireless.h>
