--- lm-sensors-2.10.1.orig/Makefile.orig	2006-09-25 00:30:10.000000000 +0200
+++ lm-sensors-2.10.1/Makefile	2006-12-09 21:13:50.000000000 +0100
@@ -45,7 +43,7 @@
 # may be used mistakenly. Note: This should point to the directory
 # *above* the linux/ subdirectory, so to /usr/local/include in the
 # above example.
-I2C_HEADERS := /usr/local/include
+I2C_HEADERS := $(STAGING_DIR)/usr/include
 #I2C_HEADERS := $(LINUX_HEADERS)
 
 ifneq ($(MAKECMDGOALS),user)
@@ -75,10 +73,10 @@
 DESTDIR :=
 
 # This is the prefix that will be used for almost all directories below.
-PREFIX := /usr/local
+PREFIX := /usr
 
 # Your C compiler
-CC := gcc
+CC := $(TARGET_CC)
 
 # This is the main modules directory into which the modules will be installed.
 # The magic invocation will return something like this:
@@ -117,7 +115,7 @@
 # library files (both static and shared) will be installed.
 LIBDIR := $(PREFIX)/lib
 
-EXLDFLAGS := -Wl,-rpath,$(LIBDIR)
+EXLDFLAGS := -Wl,-rpath,$(LIBDIR),-rpath-link,$(STAGING_DIR)/usr/lib
 
 # You should not need to change this. It is the directory into which the
 # executable program files will be installed. BINDIR for programs that are
--- lm-sensors-2.10.1.orig/lib/Module.mk	2006-09-25 00:30:10.000000000 +0200
+++ lm-sensors-2.10.1/lib/Module.mk	2006-12-09 21:14:20.000000000 +0100
@@ -66,7 +66,7 @@
 # How to create the shared library
 ifdef SYSFS_SUPPORT
 $(MODULE_DIR)/$(LIBSHLIBNAME): $(LIBSHOBJECTS)
-	$(CC) -shared -Wl,-soname,$(LIBSHSONAME) -o $@ $^ -lc -lm -lsysfs
+	$(CC) -shared -L$(STAGING_DIR)/usr/lib -Wl,-soname,$(LIBSHSONAME) -o $@ $^ -lc -lm -lsysfs
 else
 $(MODULE_DIR)/$(LIBSHLIBNAME): $(LIBSHOBJECTS)
 	$(CC) -shared -Wl,-soname,$(LIBSHSONAME) -o $@ $^ -lc -lm
