diff -urN sablevm-classpath-1.11.3.orig/lib/Makefile.am sablevm-classpath-1.11.3/lib/Makefile.am
--- sablevm-classpath-1.11.3.orig/lib/Makefile.am       2005-03-24 15:58:58.000000000 -0500
+++ sablevm-classpath-1.11.3/lib/Makefile.am    2005-12-05 23:38:23.208240037 -0500
@@ -93,7 +93,7 @@
 
 libclasspath.jar: classes compile-classes
        if test "x$(FASTJAR)" != "x"; then \
-           $(FASTJAR) $(COMPRESSION_FLAGS) -Mcf libclasspath.jar gnu java javax org > /dev/null; \
+           $(FASTJAR) -Mcf$(COMPRESSION_FLAGS) libclasspath.jar gnu java javax org > /dev/null; \
        elif test "x$(ZIP)" != "x"; then \
            $(ZIP) $(COMPRESSION_FLAGS) -r -D libclasspath.jar gnu java javax org > /dev/null; fi
 
@@ -112,7 +112,7 @@
            cp -f $(top_srcdir)/resource/$$p resources_tmp/$$p; \
        done
        if test x"$(FASTJAR)" != "x"; then cd resources_tmp && \
-           $(FASTJAR) $(COMPRESSION_FLAGS) -Mcf ../resources.jar gnu java javax org > /dev/null; \
+           $(FASTJAR) -Mcf$(COMPRESSION_FLAGS) ../resources.jar gnu java javax org > /dev/null; \
        elif test x"$(ZIP)" != "x"; then cd resources_tmp && \
            $(ZIP) $(COMPRESSION_FLAGS) -r -D ../resources.jar gnu java javax org > /dev/null; fi
        rm -rf resources_tmp
diff -urN sablevm-classpath-1.11.3.orig/lib/Makefile.in sablevm-classpath-1.11.3/lib/Makefile.in
--- sablevm-classpath-1.11.3.orig/lib/Makefile.in       2005-03-24 16:01:29.000000000 -0500
+++ sablevm-classpath-1.11.3/lib/Makefile.in    2005-12-05 23:38:03.514467812 -0500
@@ -501,7 +501,7 @@
 
 libclasspath.jar: classes compile-classes
        if test "x$(FASTJAR)" != "x"; then \
-           $(FASTJAR) $(COMPRESSION_FLAGS) -Mcf libclasspath.jar gnu java javax org > /dev/null; \
+           $(FASTJAR) -Mcf$(COMPRESSION_FLAGS) libclasspath.jar gnu java javax org > /dev/null; \
        elif test "x$(ZIP)" != "x"; then \
            $(ZIP) $(COMPRESSION_FLAGS) -r -D libclasspath.jar gnu java javax org > /dev/null; fi
 
@@ -520,7 +520,7 @@
            cp -f $(top_srcdir)/resource/$$p resources_tmp/$$p; \
        done
        if test x"$(FASTJAR)" != "x"; then cd resources_tmp && \
-           $(FASTJAR) $(COMPRESSION_FLAGS) -Mcf ../resources.jar gnu java javax org > /dev/null; \
+           $(FASTJAR) -Mcf$(COMPRESSION_FLAGS) ../resources.jar gnu java javax org > /dev/null; \
        elif test x"$(ZIP)" != "x"; then cd resources_tmp && \
            $(ZIP) $(COMPRESSION_FLAGS) -r -D ../resources.jar gnu java javax org > /dev/null; fi
        rm -rf resources_tmp