--- a/motion.c
+++ b/motion.c
@@ -1611,6 +1611,7 @@
 		 * httpd-control snaps are always enabled.
 		 */
 
+#ifdef HAVE_FFMPEG
 		/* time_current_frame is used both for snapshot and timelapse features */
 		time_current_frame = cnt->currenttime;
 
@@ -1620,7 +1621,7 @@
 			event(cnt, EVENT_IMAGE_SNAPSHOT, cnt->current_image->image, NULL, NULL, &cnt->current_image->timestamp_tm);
 			cnt->snapshot = 0;
 		}
-
+#endif /* HAVE_FFMPEG */
 
 	/***** MOTION LOOP - TIMELAPSE FEATURE SECTION *****/
 
@@ -1695,9 +1696,9 @@
 			event(cnt, EVENT_TIMELAPSEEND, NULL, NULL, NULL, cnt->currenttime_tm);
 
 
+		time_last_frame = time_current_frame;
 #endif /* HAVE_FFMPEG */
 
-		time_last_frame = time_current_frame;
 
 
 	/***** MOTION LOOP - VIDEO LOOPBACK SECTION *****/
