--- miniupnpd/Makefile.openwrt.orig	2006-11-26 16:44:21.000000000 -0600
+++ miniupnpd/Makefile.openwrt	2006-11-28 19:48:06.000000000 -0600
@@ -11,7 +11,7 @@
 CFLAGS += -D_GNU_SOURCE
 RM = rm -f
 INSTALL = install
-IPTABLESINC=$(BUILD_DIR)/iptables-1.3.3/include
+IPTABLESINC=$(STAGING_DIR)/usr/include
 
 BASEOBJS = miniupnpd.o upnphttp.o upnpdescgen.o upnpsoap.o \
            upnpreplyparse.o minixml.o \
@@ -19,7 +19,7 @@
 
 LNXOBJS = linux/getifstats.o linux/iptcrdr.o
 
-LIBS = $(BUILD_DIR)/iptables-1.3.3/libiptc/libiptc.a
+LIBS = $(STAGING_DIR)/usr/lib/libiptc.a
 
 TESTUPNPDESCGENOBJS = testupnpdescgen.o upnpdescgen.o
 
@@ -36,6 +36,7 @@
 	@echo no install for now...
 
 miniupnpd:	$(BASEOBJS) $(LNXOBJS) $(LIBS)
+	$(CC) $(CFLAGS) -o $@ $(BASEOBJS) $(LNXOBJS) $(LIBS)
 
 testupnpdescgen:	$(TESTUPNPDESCGENOBJS)
 
