diff -ur miniupnpd.old/Makefile.openwrt miniupnpd.dev/Makefile.openwrt
--- miniupnpd.old/Makefile.openwrt	2007-03-24 18:44:00.062407944 +0100
+++ miniupnpd.dev/Makefile.openwrt	2007-03-24 18:43:43.659901504 +0100
@@ -2,7 +2,6 @@
 # http://miniupnp.free.fr/
 # Author : Thomas Bernard
 # for use with GNU Make
-include $(TOPDIR)/rules.mk
 #CFLAGS = -Wall -O -D_GNU_SOURCE -g
 #CFLAGS = -Wall -Os -D_GNU_SOURCE
 #CC = mipsel-linux-gcc
@@ -11,17 +10,15 @@
 CFLAGS += -D_GNU_SOURCE
 RM = rm -f
 INSTALL = install
-IPTABLESINC=$(BUILD_DIR)/iptables-1.3.3/include
+IPTABLESINC=$(STAGING_DIR)/usr/include
 
 BASEOBJS = miniupnpd.o upnphttp.o upnpdescgen.o upnpsoap.o \
            upnpreplyparse.o minixml.o \
 		   upnpredirect.o getifaddr.o daemonize.o upnpglobalvars.o \
-		   options.o
+		   options.o upnppermissions.o
 
 LNXOBJS = linux/getifstats.o linux/iptcrdr.o
 
-LIBS = $(BUILD_DIR)/iptables-1.3.3/libiptc/libiptc.a
-
 TESTUPNPDESCGENOBJS = testupnpdescgen.o upnpdescgen.o
 
 EXECUTABLES = miniupnpd testupnpdescgen
@@ -38,6 +35,7 @@
 	@echo no install for now...
 
 miniupnpd:	$(BASEOBJS) $(LNXOBJS) $(LIBS)
+	$(CC) $(CFLAGS) -o $@ $(BASEOBJS) $(LNXOBJS) $(LIBS)
 
 testupnpdescgen:	$(TESTUPNPDESCGENOBJS)
 
@@ -69,3 +67,4 @@
 linux/iptcrdr.o: linux/iptcrdr.h
 testupnpdescgen.o: upnpdescgen.h
 upnpdescgen.o: upnpdescgen.h miniupnpdpath.h upnpglobalvars.h
+upnppermissions.o: upnppermissions.c
