--- a/netfilter/iptcrdr.c
+++ b/netfilter/iptcrdr.c
@@ -424,10 +424,20 @@ delete_redirect_and_filter_rules(unsigne
 		if(h)
 		{
 			r = delete_rule_and_commit(index, h, miniupnpd_nat_chain, "delete_redirect_rule");
+#ifdef IPTABLES_143
+			iptc_free(h);
+#else
+			iptc_free(&h);
+#endif
 		}
 		if((r == 0) && (h = iptc_init("filter")))
 		{
 			r = delete_rule_and_commit(index, h, miniupnpd_forward_chain, "delete_filter_rule");
+#ifdef IPTABLES_143
+			iptc_free(h);
+#else
+			iptc_free(&h);
+#endif
 		}
 	}
 	del_redirect_desc(eport, proto);
