--- miniupnpd.orig/Makefile.openwrt	2006-11-23 06:31:06.000000000 -0600
+++ miniupnpd/Makefile.openwrt	2006-11-26 17:16:20.000000000 -0600
@@ -2,12 +2,13 @@
 # http://miniupnp.free.fr/
 # Author : Thomas Bernard
 # for use with GNU Make
-#CFLAGS = -Wall -O -D_GNU_SOURCE -g
-CFLAGS = -Wall -Os -D_GNU_SOURCE
-CC = mipsel-linux-gcc
+include $(TOPDIR)/rules.mk
+CC = $(TARGET_CC)
+CFLAGS = $(TARGET_CFLAGS)
+CFLAGS += -D_GNU_SOURCE
 RM = rm -f
 INSTALL = install
-IPTABLESINC=/home/nanard/code/openwrt/iptables-1.3.3/include
+IPTABLESINC=$(STAGING_DIR)/usr/include
 
 BASEOBJS = miniupnpd.o upnphttp.o upnpdescgen.o upnpsoap.o \
            ../miniupnpc/upnpreplyparse.o ../miniupnpc/minixml.o \
@@ -15,7 +16,7 @@
 
 LNXOBJS = linux/getifstats.o linux/iptcrdr.o
 
-LIBS = /home/nanard/code/openwrt/iptables-1.3.3/libiptc/libiptc.a
+LIBS = $(STAGING_DIR)/usr/lib/libiptc.a
 
 TESTUPNPDESCGENOBJS = testupnpdescgen.o upnpdescgen.o
 
@@ -32,6 +33,7 @@
 	@echo no install for now...
 
 miniupnpd:	$(BASEOBJS) $(LNXOBJS) $(LIBS)
+	$(CC) $(CFLAGS) -o $@ $(BASEOBJS) $(LNXOBJS) $(LIBS)
 
 testupnpdescgen:	$(TESTUPNPDESCGENOBJS)
 
