--- miniupnpd-1.0-RC4/config.h.openwrt.orig	2006-12-02 09:25:26.000000000 -0600
+++ miniupnpd-1.0-RC4/config.h.openwrt	2007-04-26 14:20:19.000000000 -0500
@@ -5,9 +5,13 @@
 #ifndef __CONFIG_H__
 #define __CONFIG_H__
 
+#define UPNP_VERSION   "20070228"
+
 #define OS_NAME	"OpenWRT"
 /* OS/version */
 #define OS_VERSION	"OpenWRT/WhiteRussian"
 #define OS_URL	"http://openwrt.org/"
 
+#define LOG_MINIUPNPD  LOG_DAEMON
+
 #endif
