--- linuz/drivers/mtd/chips/cfi_cmdset_0002.c	2004-12-03 05:56:56.000000000 +0300
+++ linux/drivers/mtd/chips/cfi_cmdset_0002.c	2006-09-03 16:54:11.773175416 +0400
@@ -129,7 +129,8 @@
 	} /* CFI mode */
 
 	for (i=0; i< cfi->numchips; i++) {
-		cfi->chips[i].word_write_time = 1<<cfi->cfiq->WordWriteTimeoutTyp;
+		cfi->chips[i].word_write_time = 1<<	/* mirrorbit hack */
+			min(cfi->cfiq->WordWriteTimeoutTyp,cfi->cfiq->WordWriteTimeoutMax);
 		cfi->chips[i].buffer_write_time = 1<<cfi->cfiq->BufWriteTimeoutTyp;
 		cfi->chips[i].erase_time = 1<<cfi->cfiq->BlockEraseTimeoutTyp;
 	}		
