--- a/configure
+++ b/configure
@@ -21579,7 +21579,7 @@ for ac_header in \
 	openssl/ssl.h \
 	openssl/crypto.h \
 	openssl/err.h \
-	openssl/engine.h
+	
 do
 as_ac_Header=`echo "ac_cv_header_$ac_header" | $as_tr_sh`
 if { as_var=$as_ac_Header; eval "test \"\${$as_var+set}\" = set"; }; then
--- a/configure.in
+++ b/configure.in
@@ -644,7 +644,7 @@ if test "x$WITH_OPENSSL" = xyes; then
 	openssl/ssl.h \
 	openssl/crypto.h \
 	openssl/err.h \
-	openssl/engine.h,
+	, 
 	[],
 	OPENSSL_LIBS=
     )
