diff -urN rrdcollect-0.2.3.old/src/rrdcollect.c rrdcollect-0.2.3/src/rrdcollect.c
--- rrdcollect-0.2.3.old/src/rrdcollect.c	2007-06-06 12:06:01.000000000 -0400
+++ rrdcollect-0.2.3/src/rrdcollect.c	2007-06-06 12:07:05.000000000 -0400
@@ -58,7 +58,7 @@
 const char *workdir = NULL;
 int step = 60;
 int debugoutput = OUTPUT_SYSLOG;
-int debuglevel = LOG_DEBUG;  /* from syslog.h */
+int debuglevel = LOG_NOTICE;  /* from syslog.h */
 /* FIXME - it should be default LOG_NOTICE or LOG_WARNING, but first
    option for change it should be added */
 
@@ -160,7 +160,7 @@
 {
 	optind = opterr = 0;
 	
-	send_log(LOG_DEBUG,"rrdlib_update(%i,'%s')",argc,*argv);
+	/* send_log(LOG_DEBUG,"rrdlib_update(%i,'%s')",argc,*argv); */
 	rrd_update(argc, argv);
 
 	if (rrd_test_error()) {
@@ -479,7 +479,7 @@
 	setitimer(ITIMER_REAL, &period, NULL);
 
 	for (;;) {
-		send_log(LOG_DEBUG,"tick");
+		/* send_log(LOG_DEBUG,"tick"); */
 		if (action_request) {
 			switch (action_request) {
 				case SIGINT:
