--- cups-1.4.2.orig/ppdc/Makefile	2010-01-28 18:06:41.000000000 +0100
+++ cups-1.4.2/ppdc/Makefile	2010-01-28 20:25:09.000000000 +0100
@@ -239,12 +239,7 @@
 
 genstrings:		genstrings.o libcupsppdc.a ../cups/libcups.a \
 			sample.drv ../data/media.defs
-	echo Linking $@...
-	$(CXX) $(ARCHFLAGS) $(LDFLAGS) -o genstrings genstrings.o \
-		libcupsppdc.a ../cups/libcups.a $(LIBGSSAPI) $(SSLLIBS) \
-		$(DNSSDLIBS) $(COMMONLIBS) $(LIBZ)
-	echo Generating localization strings...
-	./genstrings >sample.c
+	echo Not Linking $@...
 
 
 #
