diff -urN xlogo-X11R7.0-1.0.1.orig/Makefile.in xlogo-X11R7.0-1.0.1/Makefile.in
--- xlogo-X11R7.0-1.0.1.orig/Makefile.in	2007-09-17 23:28:25.000000000 +0200
+++ xlogo-X11R7.0-1.0.1/Makefile.in	2007-09-17 23:29:17.000000000 +0200
@@ -177,7 +177,7 @@
 bin_PROGRAMS = xlogo
 
 xlogo_CFLAGS = $(XLOGO_CFLAGS)
-xlogo_LDADD = $(XLOGO_LIBS)
+xlogo_LDADD = $(XLOGO_LIBS) -lXrender
 
 xlogo_SOURCES = \
         Logo.c \
