--- a/Makefile.in
+++ b/Makefile.in
@@ -270,7 +270,7 @@ sys_symbol_underscore = @sys_symbol_unde
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 AUTOMAKE_OPTIONS = foreign
-SUBDIRS = liboil testsuite examples doc
+SUBDIRS = liboil testsuite
 EXTRA_DIST = COPYING autogen.sh gtk-doc.make HACKING BUG-REPORTING
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 pkgconfig_DATA = liboil-$(LIBOIL_MAJORMINOR).pc
