Index: pcre-7.2/pcre-config.in
===================================================================
--- pcre-7.2.orig/pcre-config.in	2007-07-16 08:33:37.000000000 +0200
+++ pcre-7.2/pcre-config.in	2007-07-16 08:33:37.000000000 +0200
@@ -21,6 +21,10 @@
   libR=" -Wl,-R@libdir@"
   ;;
 esac
+libS=
+if test @libdir@ != /usr/lib ; then
+  libS=-L@libdir@
+fi
 
 while test $# -gt 0; do
   case "$1" in
@@ -55,10 +59,10 @@
       echo $includes
       ;;
     --libs-posix)
-      echo -L@libdir@$libR -lpcreposix -lpcre
+      echo $libS$libR -lpcreposix -lpcre
       ;;
     --libs)
-      echo -L@libdir@$libR -lpcre
+      echo $libS$libR -lpcre
       ;;
     *)
       echo "${usage}" 1>&2
