diff -ur keepalived-1.1.15/configure keepalived-1.1.15.mod/configure
--- keepalived-1.1.15/configure	2006-10-10 05:20:35.000000000 -0400
+++ keepalived-1.1.15.mod/configure	2009-05-08 04:52:13.000000000 -0400
@@ -3928,7 +3928,7 @@
 
 
 
-VRRP_SUPPORT="_WITHOUT_VRRP_"
+VRRP_SUPPORT="_WITH_VRRP_"
 if test "$enable_vrrp" != "no"; then
   echo "$as_me:$LINENO: checking for kernel netlink support" >&5
 echo $ECHO_N "checking for kernel netlink support... $ECHO_C" >&6
