--- a/service-type-database/Makefile.am
+++ b/service-type-database/Makefile.am
@@ -18,13 +18,12 @@
 EXTRA_DIST=build-db.in service-types
 
 pkgdata_DATA=service-types
-pkglib_DATA=
 
 if HAVE_PYTHON
 if HAVE_GDBM
 
 noinst_SCRIPTS=build-db
-pkglib_DATA+=service-types.db
+pkgdata_DATA+=service-types.db
 
 build-db: build-db.in
 	$(AM_V_GEN)sed -e 's,@PYTHON\@,$(PYTHON),g' \
@@ -41,7 +40,7 @@ endif
 if HAVE_DBM
 
 noinst_SCRIPTS=build-db
-pkglib_DATA+=service-types.db.pag service-types.db.dir
+pkgdata_DATA+=service-types.db.pag service-types.db.dir
 
 build-db: build-db.in
 	$(AM_V_GEN)sed -e 's,@PYTHON\@,$(PYTHON),g' \
