--- a/configure
+++ b/configure
@@ -1083,7 +1083,7 @@ fi


 if test "x$ac_cv_prog_gcc" = xyes ; then
-  CFLAGS="-O2"
+  CFLAGS="-O2 ${CFLAGS}"
   case "$host_os" in
     IRIX*)
       CFLAGS="-mabi=n32 -fno-builtins ${CFLAGS}"
@@ -1097,7 +1097,7 @@ else
       LDFLAGS="-n32 -L/usr/lib32 -L/usr/freeware/lib32 ${LDFLAGS}"
       ;;
     OSF*)
-      CFLAGS="-O -assume noaligned_objects"
+      CFLAGS="-O -assume noaligned_objects ${CFLAGS}"
       ;;
   esac
 fi
