diff -urN pglinux-1.5beta/Makefile.am pglinux-1.5beta.new/Makefile.am
--- pglinux-1.5beta/Makefile.am	2005-04-02 07:39:56.000000000 +0200
+++ pglinux-1.5beta.new/Makefile.am	2008-12-14 18:25:45.000000000 +0100
@@ -3,7 +3,7 @@
 mybin_PROGRAMS = peerguardnf pgtext
 peerguardnf_SOURCES=Main.cpp PeerGuard.cpp BlockList.cpp Blocker_Linux.cpp Blocker_BSD.cpp HttpServer.cpp HttpRequest.cpp sha1.cpp
 pgtext_SOURCES=pgtext.cpp
-peerguardnf_LDADD=-lipq -lpthread
+peerguardnf_LDADD=-lnetfilter_queue_libipq -lnetfilter_queue -lnfnetlink -lpthread
 pgtext_LDADD=-lncurses -lpthread
 peerguardnf_LDFLAGS=$(all_includes) -I/usr/include/libipq
 AM_CPPFLAGS=$(ALL_INCLUDES) -I/usr/include/libipq
diff -urN pglinux-1.5beta/Makefile.in pglinux-1.5beta.new/Makefile.in
--- pglinux-1.5beta/Makefile.in	2005-03-31 09:40:29.000000000 +0200
+++ pglinux-1.5beta.new/Makefile.in	2008-12-14 18:25:25.000000000 +0100
@@ -72,7 +72,7 @@
 mybin_PROGRAMS = peerguardnf pgtext
 peerguardnf_SOURCES = Main.cpp PeerGuard.cpp BlockList.cpp Blocker_Linux.cpp Blocker_BSD.cpp HttpServer.cpp HttpRequest.cpp sha1.cpp
 pgtext_SOURCES = pgtext.cpp
-peerguardnf_LDADD = -lipq -lpthread
+peerguardnf_LDADD = -lnetfilter_queue_libipq -lnetfilter_queue -lnfnetlink -lpthread
 pgtext_LDADD = -lncurses -lpthread
 peerguardnf_LDFLAGS = $(all_includes) -I/usr/include/libipq
 AM_CPPFLAGS = $(ALL_INCLUDES) -I/usr/include/libipq
