diff -ruaN httping-1.2.2.orig/Makefile httping-1.2.2/Makefile
--- httping-1.2.2.orig/Makefile	2006-11-09 20:13:12.000000000 +0100
+++ httping-1.2.2/Makefile	2007-02-27 11:34:47.000000000 +0100
@@ -15,8 +15,8 @@
 include version
 
 DEBUG=-g # -D_DEBUG
-LDFLAGS+=-lssl -lcrypto $(DEBUG)
-CFLAGS+=-O2 -Wall -DVERSION=\"$(VERSION)\" $(DEBUG)
+LDFLAGS+=$(EXTRA_LDFLAGS) -lssl -lcrypto $(DEBUG)
+CFLAGS+=$(EXTRA_CFLAGS) -Wall -DVERSION=\"$(VERSION)\" $(DEBUG)
 
 OBJS=mem.o http.o io.o str.o error.o utils.o main.o tcp.o mssl.o res.o
 
