--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@ TARGET=httping
 
 WFLAGS=-Wall -W
 OFLAGS=-O2
-CFLAGS+=$(WFLAGS) $(OFLAGS) -DVERSION=\"$(VERSION)\"
+CFLAGS+=$(EXTRA_CFLAGS) $(WFLAGS) $(OFLAGS) -DVERSION=\"$(VERSION)\"
 
 PACKAGE=$(TARGET)-$(VERSION)
 PREFIX=/usr
@@ -47,7 +47,7 @@ ifeq ($(SSL),no)
 CFLAGS+=-DNO_SSL
 else
 OBJS+=mssl.o
-LDFLAGS+=-lssl -lcrypto
+LDFLAGS+=$(EXTRA_LDFLAGS) -lssl -lcrypto
 endif
 
 ifeq ($(DEBUG),yes)
