--- a/main.c
+++ b/main.c
@@ -40,6 +40,8 @@
 #include "utils.h"
 #include "error.h"
 
+#define VERSION "1.3.1"
+
 static volatile int stop = 0;
 
 int quiet = 0;
@@ -50,7 +52,7 @@ char last_error[ERROR_BUFFER_SIZE];
 
 void version(void)
 {
-	fprintf(stderr, "HTTPing v" VERSION ", (C) 2003-2009 folkert@vanheusden.com\n");
+	fprintf(stderr, "HTTPing v%s (C) 2003-2009 folkert@vanheusden.com\n", VERSION);
 #ifndef NO_SSL
 	fprintf(stderr, "SSL support included\n");
 #endif
@@ -449,7 +451,7 @@ int main(int argc, char *argv[])
 	if (useragent)
 		sprintf(&request[strlen(request)], "User-Agent: %s\r\n", useragent);
 	else
-		sprintf(&request[strlen(request)], "User-Agent: HTTPing v" VERSION "\r\n");
+		sprintf(&request[strlen(request)], "User-Agent: HTTPing v%s\r\n", VERSION);
 	sprintf(&request[strlen(request)], "Host: %s\r\n", hostname);
 	if (referer)
 		sprintf(&request[strlen(request)], "Referer: %s\r\n", referer);
