diff -urN httping-1.0.10/main.c httping-1.0.10.new/main.c
--- httping-1.0.10/main.c	2006-01-02 17:07:29.000000000 +0100
+++ httping-1.0.10.new/main.c	2007-02-27 11:14:32.000000000 +0100
@@ -38,6 +38,8 @@
 #include "utils.h"
 #include "error.h"
 
+#define VERSION "1.0.10"
+
 static volatile int stop = 0;
 
 int quiet = 0;
@@ -48,7 +50,7 @@
 
 void version(void)
 {
-	fprintf(stderr, "HTTPing v" VERSION ", (C) 2003-2005 folkert@vanheusden.com\n");
+	fprintf(stderr, "HTTPing v%s (C) 2003-2005 folkert@vanheusden.com\n", VERSION);
 #ifndef NO_SSL
 	fprintf(stderr, "SSL support included\n");
 #endif
@@ -382,7 +384,7 @@
 	if (useragent)
 		sprintf(&request[strlen(request)], "User-Agent: %s\r\n", useragent);
 	else
-		sprintf(&request[strlen(request)], "User-Agent: HTTPing v" VERSION "\r\n");
+		sprintf(&request[strlen(request)], "User-Agent: HTTPing v%s\r\n", VERSION);
 	if (referer)
 		sprintf(&request[strlen(request)], "Referer: %s\r\n", referer);
 	strcat(request, "\r\n");
