diff -burN privoxy-3.0.12-stable/jcc.c.rej privoxy-3.0.12-stable.patched/jcc.c.rej
--- privoxy-3.0.12-stable/jcc.c.rej	2008-11-26 18:56:54.000000000 +0100
+++ privoxy-3.0.12-stable.patched/jcc.c.rej	1970-01-01 01:00:00.000000000 +0100
@@ -1,20 +0,0 @@
-***************
-*** 3110,3118 ****
-           }
-           continue;
-        }
--       log_error(LOG_LEVEL_INFO,
--          "Shouldn't get here but did.");
--       return;
-     }
-  
-     if (csp->content_length == 0)
---- 3120,3127 ----
-           }
-           continue;
-        }
-+       mark_server_socket_tainted(csp);
-+       return; /* huh? we should never get here */
-     }
-  
-     if (csp->content_length == 0)
diff -burN privoxy-3.0.12-stable/parsers.c.rej privoxy-3.0.12-stable.patched/parsers.c.rej
--- privoxy-3.0.12-stable/parsers.c.rej	2008-09-15 19:52:19.000000000 +0200
+++ privoxy-3.0.12-stable.patched/parsers.c.rej	1970-01-01 01:00:00.000000000 +0100
@@ -1,16 +0,0 @@
-***************
-*** 986,991 ****
-  const add_header_func_ptr add_client_headers[] = {
-     client_host_adder,
-     client_cookie_adder,
-     client_xtra_adder,
-     /* Temporarily disabled:    client_accept_encoding_adder, */
-     connection_close_adder,
---- 983,989 ----
-  const add_header_func_ptr add_client_headers[] = {
-     client_host_adder,
-     client_cookie_adder,
-+    client_x_forwarded_adder,
-     client_xtra_adder,
-     /* Temporarily disabled:    client_accept_encoding_adder, */
-     connection_close_adder,
