--- linux-2.4.30/arch/mips/bcm947xx/gpio.c.orig	2006-03-26 06:43:24 -05:00
+++ linux-2.4.30/arch/mips/bcm947xx/gpio.c	2006-03-26 06:44:00 -05:00
@@ -135,7 +135,7 @@
 		gpio_file[i].handle = devfs_register(gpio_dir,
 						     gpio_file[i].name,
 						     DEVFS_FL_DEFAULT, gpio_major, i,
-						     S_IFCHR | S_IRUGO | S_IWUGO,
+						     S_IFCHR | S_IRUSR | S_IWUSR,
 						     &gpio_fops, NULL);
 	}
 
