diff -urN fluxbox-1.0.0.orig/util/Makefile.in fluxbox-1.0.0/util/Makefile.in
--- fluxbox-1.0.0.orig/util/Makefile.in	2007-12-30 18:54:47.000000000 +0100
+++ fluxbox-1.0.0/util/Makefile.in	2007-12-30 18:55:41.000000000 +0100
@@ -276,7 +276,7 @@
 	-test -z "$(bin_PROGRAMS)" || rm -f $(bin_PROGRAMS)
 fbsetroot$(EXEEXT): $(fbsetroot_OBJECTS) $(fbsetroot_DEPENDENCIES) 
 	@rm -f fbsetroot$(EXEEXT)
-	$(CXXLINK) $(fbsetroot_OBJECTS) $(fbsetroot_LDADD) $(LIBS)
+	$(CXXLD) $(AM_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) -o $@ $(fbsetroot_OBJECTS) $(fbsetroot_LDADD) $(LDFLAGS) $(LIBS)
 fluxbox-remote$(EXEEXT): $(fluxbox_remote_OBJECTS) $(fluxbox_remote_DEPENDENCIES) 
 	@rm -f fluxbox-remote$(EXEEXT)
 	$(CXXLINK) $(fluxbox_remote_OBJECTS) $(fluxbox_remote_LDADD) $(LIBS)
