diff -urN xapian-core-1.0.7/backends/flint/flint_version.cc xapian-core-1.0.7.new/backends/flint/flint_version.cc
--- xapian-core-1.0.7/backends/flint/flint_version.cc	2008-07-16 06:42:34.000000000 +0200
+++ xapian-core-1.0.7.new/backends/flint/flint_version.cc	2010-03-29 09:59:58.000000000 +0200
@@ -19,6 +19,7 @@
  */
 
 #include <config.h>
+#include <stdio.h>
 
 #include "safeerrno.h"
 
diff -urN xapian-core-1.0.7/bin/quartzdump.cc xapian-core-1.0.7.new/bin/quartzdump.cc
--- xapian-core-1.0.7/bin/quartzdump.cc	2008-07-16 06:42:28.000000000 +0200
+++ xapian-core-1.0.7.new/bin/quartzdump.cc	2010-03-29 10:18:26.000000000 +0200
@@ -29,6 +29,7 @@
 #include <vector>
 #include <string>
 #include <ctype.h>
+#include <stdio.h>
 
 #include "autoptr.h"
 #include "gnu_getopt.h"
--- xapian-core-1.0.7/bin/xapian-progsrv.cc	2008-07-16 06:42:28.000000000 +0200
+++ xapian-core-1.0.7.new/bin/xapian-progsrv.cc	2010-03-29 10:31:16.000000000 +0200
@@ -19,6 +19,7 @@
  */
 
 #include <config.h>
+#include <stdio.h>
 
 #include "remoteserver.h"
 
--- xapian-core-1.0.7/bin/xapian-tcpsrv.cc	2008-07-16 06:42:28.000000000 +0200
+++ xapian-core-1.0.7.new/bin/xapian-tcpsrv.cc	2010-03-29 10:41:56.000000000 +0200
@@ -21,6 +21,7 @@
  */
 
 #include <config.h>
+#include <stdio.h>
 
 #include <stdlib.h>
 
diff -urN xapian-core-1.0.7/examples/delve.cc xapian-core-1.0.7.new/examples/delve.cc
--- xapian-core-1.0.7/examples/delve.cc	2008-07-16 06:42:35.000000000 +0200
+++ xapian-core-1.0.7.new/examples/delve.cc	2010-03-29 11:14:26.000000000 +0200
@@ -21,6 +21,7 @@
  */
 
 #include <config.h>
+#include <stdio.h>
 
 #include <xapian.h>
 
--- xapian-core-1.0.7/examples/quest.cc	2008-07-16 06:42:35.000000000 +0200
+++ xapian-core-1.0.7.new/examples/quest.cc	2010-03-29 11:33:03.000000000 +0200
@@ -19,6 +19,7 @@
  */
 
 #include <config.h>
+#include <stdio.h>
 
 #include <xapian.h>
 
