diff -ur speex.old/libspeex/Makefile.in speex.dev/libspeex/Makefile.in
--- speex.old/libspeex/Makefile.in	2006-10-14 15:16:02.000000000 -0700
+++ speex.dev/libspeex/Makefile.in	2006-10-14 15:14:35.000000000 -0700
@@ -107,7 +107,7 @@
 
 EXTRA_DIST = testenc.c testenc_wb.c testenc_uwb.c testdenoise.c testecho.c
 
-INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir) @OGG_CFLAGS@
+INCLUDES = -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_builddir)
 
 lib_LTLIBRARIES = libspeex.la
 
