--- a/configure
+++ b/configure
@@ -10811,7 +10811,7 @@ if grep -q "CentOS release 5\." /etc/red
 fi;
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking kernel version that we will build against" >&5
 $as_echo_n "checking kernel version that we will build against... " >&6; }
-krel="$(make -sC ${kbuilddir} kernelrelease)";
+krel="$(make -sC ${kbuilddir} kernelversion)";
 krel="${krel%%-*}";
 kmajor="${krel%%.*}";
 krel="${krel#*.}";
--- a/configure.ac
+++ b/configure.ac
@@ -49,7 +49,7 @@ if grep -q "CentOS release 5\." /etc/red
 	echo "WARNING: This distribution's shipped kernel is not supported.";
 fi;
 AC_MSG_CHECKING([kernel version that we will build against])
-krel="$(make -sC ${kbuilddir} kernelrelease)";
+krel="$(make -sC ${kbuilddir} kernelversion)";
 krel="${krel%%-*}";
 kmajor="${krel%%.*}";
 krel="${krel#*.}";
