diff -ruN paroli-20090104.orig/setup.py paroli-20090104/setup.py
--- paroli-20090104.orig/setup.py	2009-01-05 08:38:30.000000000 +0100
+++ paroli-20090104/setup.py	2009-01-05 08:45:09.000000000 +0100
@@ -74,7 +74,7 @@
 dbus_data = [
     (os.path.join(sys.prefix, 'share/dbus-1/system-services/'),
      ['data/dbus/org.tichy.launcher.service']),
-    ('../../etc/dbus-1/system.d/',
+    ('../etc/dbus-1/system.d/',
      ['data/dbus/tichy.conf'])]
 
 setup(name='Paroli',
@@ -94,7 +94,7 @@
                       'data/paroli-dialer.desktop']),
                     (os.path.join(sys.prefix, 'share/pixmaps/'),
                      ['data/tichy']),
-                    ('../../etc/paroli/', ['data/paroli.cfg'])] \
+                    ('../etc/paroli/', ['data/paroli.cfg'])] \
           + plugins_files('paroli-services', 'paroli/services') \
           + plugins_files('paroli-applications', 'paroli/applications') \
           + dbus_data,
