diff -ruN paroli.orig/data/msgs.desktop paroli/data/msgs.desktop
--- paroli.orig/data/msgs.desktop	1970-01-01 01:00:00.000000000 +0100
+++ paroli/data/msgs.desktop	2009-02-23 22:38:37.000000000 +0100
@@ -0,0 +1,13 @@
+[Desktop Entry]
+GenericName=Messages
+Name=Messages
+Encoding=UTF-8
+Version=1.0
+Comment=Paroli Messages
+StartupNotify=false
+Exec=dbus-send --system --print-reply --dest=org.tichy.launcher /Launcher org.tichy.Launcher.Launch string:Msgs
+Icon=/usr/share/icons/paroli/msgs.png
+Terminal=false
+Type=Application
+SingleInstance=true
+Categories=Telephony;TelephonyTools
diff -ruN paroli.orig/data/parolid.desktop paroli/data/parolid.desktop
--- paroli.orig/data/parolid.desktop	1970-01-01 01:00:00.000000000 +0100
+++ paroli/data/parolid.desktop	2009-02-23 22:38:37.000000000 +0100
@@ -0,0 +1,13 @@
+[Desktop Entry]
+GenericName=Paroli
+Name=Paroli
+Encoding=UTF-8
+Version=1.0
+Comment=Paroli Daemon
+StartupNotify=false
+Exec=paroli-launcher
+Icon=/usr/share/icons/paroli/parolid.png
+Terminal=false
+Type=Application
+SingleInstance=true
+Categories=Telephony;TelephonyTools
diff -ruN paroli.orig/data/people.desktop paroli/data/people.desktop
--- paroli.orig/data/people.desktop	1970-01-01 01:00:00.000000000 +0100
+++ paroli/data/people.desktop	2009-02-23 22:38:37.000000000 +0100
@@ -0,0 +1,13 @@
+[Desktop Entry]
+GenericName=People
+Name=People
+Encoding=UTF-8
+Version=1.0
+Comment=Paroli People
+StartupNotify=false
+Exec=dbus-send --system --print-reply --dest=org.tichy.launcher /Launcher org.tichy.Launcher.Launch string:People
+Icon=/usr/share/icons/paroli/people.png
+Terminal=false
+Type=Application
+SingleInstance=true
+Categories=Telephony;TelephonyTools
diff -ruN paroli.orig/data/tele.desktop paroli/data/tele.desktop
--- paroli.orig/data/tele.desktop	1970-01-01 01:00:00.000000000 +0100
+++ paroli/data/tele.desktop	2009-02-23 22:38:37.000000000 +0100
@@ -0,0 +1,13 @@
+[Desktop Entry]
+GenericName=Tele
+Name=Tele
+Encoding=UTF-8
+Version=1.0
+Comment=Paroli Tele
+StartupNotify=false
+Exec=dbus-send --system --print-reply --dest=org.tichy.launcher /Launcher org.tichy.Launcher.Launch string:Tele
+Icon=/usr/share/icons/paroli/tele.png
+Terminal=false
+Type=Application
+SingleInstance=true
+Categories=Telephony;TelephonyTools
diff -ruN paroli.orig/setup.py paroli/setup.py
--- paroli.orig/setup.py	2009-02-23 13:38:05.000000000 +0100
+++ paroli/setup.py	2009-02-23 22:40:23.000000000 +0100
@@ -19,9 +19,6 @@
 #    You should have received a copy of the GNU General Public License
 #    along with Paroli.  If not, see <http://www.gnu.org/licenses/>.
 
-# XXX: This file only work when used by OpenEmbedded.  We should fix
-#      that !
-
 import sys
 import os
 
@@ -87,7 +84,10 @@
       scripts= ['paroli-scripts/paroli-launcher'],
       # XXX: Those locations may not work on the neo !
       data_files = [('applications',
-                     ['data/paroli-launcher.desktop',
+                     ['data/parolid.desktop',
+                      'data/tele.desktop',
+                      'data/msgs.desktop',
+                      'data/people.desktop',
                       #'data/paroli-io.desktop',
                       #'data/paroli-contacts.desktop',
                       #'data/paroli-msgs.desktop',
