diff -ruN paroli/data/paroli-contacts.desktop paroli.mod/data/paroli-contacts.desktop
--- paroli/data/paroli-contacts.desktop	2009-01-23 13:35:16.000000000 +0100
+++ paroli.mod/data/paroli-contacts.desktop	2009-01-23 13:33:12.000000000 +0100
@@ -1,16 +1,13 @@
 [Desktop Entry]
-Name=People
-Name[fr]=Calculatrice
-Name[it]=Calcolatrice
-Name[ru]=Калькулятор
-Name[sv]=Miniräknare
+GenericName=Contacts
+Name=Contacts
 Encoding=UTF-8
 Version=1.0
-Comment=Paroli Dialer
+Comment=Paroli Contacts
 StartupNotify=false
 Exec=dbus-send --system --print-reply --dest=org.tichy.launcher /Launcher org.tichy.Launcher.Launch string:Paroli-Contacts
-Icon=openmoko-calculator
-Terminal=true
+Icon=paroli-contacts.png
+Terminal=false
 Type=Application
 SingleInstance=true
-Categories=GTK;Application;Office;
\ No newline at end of file
+Categories=Telephony;TelephonyTools
diff -ruN paroli/data/paroli-dialer.desktop paroli.mod/data/paroli-dialer.desktop
--- paroli/data/paroli-dialer.desktop	2009-01-23 13:35:16.000000000 +0100
+++ paroli.mod/data/paroli-dialer.desktop	2009-01-23 13:33:24.000000000 +0100
@@ -1,16 +1,13 @@
 [Desktop Entry]
-Name=Tele
-Name[fr]=Calculatrice
-Name[it]=Calcolatrice
-Name[ru]=Калькулятор
-Name[sv]=Miniräknare
+GenericName=Dialer
+Name=Dialer
 Encoding=UTF-8
 Version=1.0
 Comment=Paroli Dialer
 StartupNotify=false
 Exec=dbus-send --system --print-reply --dest=org.tichy.launcher /Launcher org.tichy.Launcher.Launch string:Paroli-Dialer
-Icon=openmoko-calculator
-Terminal=true
+Icon=paroli-dialer.png
+Terminal=false
 Type=Application
 SingleInstance=true
-Categories=GTK;Application;Office;
\ No newline at end of file
+Categories=Telephony;TelephonyTools
diff -ruN paroli/data/paroli-io.desktop paroli.mod/data/paroli-io.desktop
--- paroli/data/paroli-io.desktop	2009-01-23 13:35:16.000000000 +0100
+++ paroli.mod/data/paroli-io.desktop	2009-01-23 13:33:33.000000000 +0100
@@ -1,16 +1,13 @@
 [Desktop Entry]
+GenericName=I/O
 Name=I/O
-Name[fr]=Calculatrice
-Name[it]=Calcolatrice
-Name[ru]=Калькулятор
-Name[sv]=Miniräknare
 Encoding=UTF-8
 Version=1.0
 Comment=Paroli I/O
 StartupNotify=false
 Exec=dbus-send --system --print-reply --dest=org.tichy.launcher /Launcher org.tichy.Launcher.Launch string:Paroli-I/O
-Icon=openmoko-calculator
-Terminal=true
+Icon=paroli-i-o.png
+Terminal=false
 Type=Application
 SingleInstance=true
-Categories=GTK;Application;Office;
\ No newline at end of file
+Categories=Telephony;TelephonyTools
diff -ruN paroli/data/paroli-launcher.desktop paroli.mod/data/paroli-launcher.desktop
--- paroli/data/paroli-launcher.desktop	2009-01-23 13:35:16.000000000 +0100
+++ paroli.mod/data/paroli-launcher.desktop	2009-01-23 13:33:50.000000000 +0100
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Name=PLauncher
 TryExec=paroli-launcher
-Icon=tichy
+Icon=tichy.png
 GenericName=paroli-launcher
 Comment=Paroli Launcher
 StartupNotify=false
 Exec=paroli-launcher
 Terminal=false
 Type=Application
-Categories=Office;
+Categories=Telephony;TelephonyTools
diff -ruN paroli/data/paroli-msgs.desktop paroli.mod/data/paroli-msgs.desktop
--- paroli/data/paroli-msgs.desktop	2009-01-23 13:35:16.000000000 +0100
+++ paroli.mod/data/paroli-msgs.desktop	2009-01-23 13:37:18.000000000 +0100
@@ -1,16 +1,13 @@
 [Desktop Entry]
-Name=Msgs
-Name[fr]=Calculatrice
-Name[it]=Calcolatrice
-Name[ru]=Калькулятор
-Name[sv]=Miniräknare
+GenericName=Messages
+Name=Messages
 Encoding=UTF-8
 Version=1.0
 Comment=Paroli Messages
 StartupNotify=false
 Exec=dbus-send --system --print-reply --dest=org.tichy.launcher /Launcher org.tichy.Launcher.Launch string:Paroli-Msgs
-Icon=openmoko-calculator
-Terminal=true
+Icon=paroli-msgs.png
+Terminal=false
 Type=Application
 SingleInstance=true
-Categories=GTK;Application;Office;
\ No newline at end of file
+Categories=Telephony;TelephonyTools
