diff -ruN paroli-20090112.orig/data/paroli-launcher.desktop paroli-20090112/data/paroli-launcher.desktop
--- paroli-20090112.orig/data/paroli-launcher.desktop	2009-01-17 05:05:49.000000000 +0100
+++ paroli-20090112/data/paroli-launcher.desktop	2009-01-20 21:20:35.000000000 +0100
@@ -5,7 +5,7 @@
 GenericName=paroli-launcher
 Comment=Paroli Launcher
 StartupNotify=false
-Exec=paroli-launcher
+Exec=sleep 60 && DISPLAY=":0" LD_PRELOAD=/usr/lib/libX11.so paroli-launcher
 Terminal=false
 Type=Application
 Categories=Telephony;TelephonyTools;
