Index: gnutls-2.2.3/libextra/gnutls_extra.c
===================================================================
--- gnutls-2.2.3.orig/libextra/gnutls_extra.c	2008-05-08 21:18:29.000000000 +0200
+++ gnutls-2.2.3/libextra/gnutls_extra.c	2008-05-08 21:18:29.000000000 +0200
@@ -137,6 +137,7 @@
     }
 #endif
 
+#ifdef ENABLE_OPENPGP
   /* Register the openpgp functions. This is because some
    * of them are defined to be NULL in the main library.
    */
@@ -152,6 +153,7 @@
 				 gnutls_openpgp_crt_deinit,
 				 gnutls_openpgp_keyring_deinit,
 				 gnutls_openpgp_privkey_deinit);
+#endif
 
   return 0;
 }
