--- a/src/ssl.c
+++ b/src/ssl.c
@@ -63,13 +63,16 @@ void ssl_init(void) { /* init SSL before
 }
 
 void ssl_configure(void) { /* configure global SSL settings */
+#ifndef OPENSSL_NO_COMP
     if(options.compression!=COMP_NONE)
         init_compression();
+#endif
     if(!init_prng())
         s_log(LOG_DEBUG, "PRNG seeded successfully");
 }
 
 static void init_compression(void) {
+#ifndef OPENSSL_NO_COMP
     int id=0;
     COMP_METHOD *cm=NULL;
     char *name="unknown";
@@ -98,6 +101,7 @@ static void init_compression(void) {
         die(1);
     }
     s_log(LOG_INFO, "Compression enabled using %s method", name);
+#endif
 }
 
 static int init_prng(void) {
