diff -urN prosody-0.3.0/util-src/Makefile prosody-0.3.0.new/util-src/Makefile
--- prosody-0.3.0/util-src/Makefile	2009-02-02 19:05:21.000000000 +0100
+++ prosody-0.3.0.new/util-src/Makefile	2009-02-15 14:44:53.000000000 +0100
@@ -6,7 +6,7 @@
 LUA_LIB?=lua$(LUA_SUFFIX)
 IDN_LIB?=idn
 OPENSSL_LIB?=crypto
-
+CC=gcc
 
 
 all: encodings.so hashes.so pposix.so
@@ -21,18 +21,18 @@
 	rm -f ../util/*.so
 
 encodings.o: encodings.c
-	gcc $(CFLAGS) -I$(LUA_INCDIR) -c -o encodings.o encodings.c
+	$(CC) $(CFLAGS) -I$(LUA_INCDIR) -c -o encodings.o encodings.c
 encodings.so: encodings.o
-	export MACOSX_DEPLOYMENT_TARGET="10.3"; gcc $(LFLAGS) -o encodings.so encodings.o -L/usr/local/lib -llua$(LUA_SUFFIX) -lidn
+	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(LFLAGS) -o encodings.so encodings.o -L$(LUA_LIBDIR) -llua$(LUA_SUFFIX) -lidn
 	
 
 hashes.o: hashes.c
-	gcc $(CFLAGS) -I$(LUA_INCDIR) -c -o hashes.o hashes.c
+	$(CC) $(CFLAGS) -I$(LUA_INCDIR) -c -o hashes.o hashes.c
 hashes.so: hashes.o
-	export MACOSX_DEPLOYMENT_TARGET="10.3"; gcc $(LFLAGS) -o hashes.so hashes.o -L/usr/local/lib -llua$(LUA_SUFFIX) -lcrypto
+	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(LFLAGS) -o hashes.so hashes.o -L$(LUA_LIBDIR) -llua$(LUA_SUFFIX) -lcrypto
 
 pposix.o: pposix.c
-	gcc $(CFLAGS) -I$(LUA_INCDIR) -c -o pposix.o pposix.c
+	$(CC) $(CFLAGS) -I$(LUA_INCDIR) -c -o pposix.o pposix.c
 pposix.so: pposix.o
-	export MACOSX_DEPLOYMENT_TARGET="10.3"; gcc $(LFLAGS) -o pposix.so pposix.o -L/usr/local/lib -llua$(LUA_SUFFIX)
+	export MACOSX_DEPLOYMENT_TARGET="10.3"; $(CC) $(LFLAGS) -o pposix.so pposix.o -L$(LUA_LIBDIR) -llua$(LUA_SUFFIX)
 	
