--- ruby-1.8.6-p36.orig/ext/openssl/openssl_missing.h	2007-02-13 00:01:19.000000000 +0100
+++ ruby-1.8.6-p36/ext/openssl/openssl_missing.h	2009-01-31 11:26:31.000000000 +0100
@@ -116,8 +116,8 @@
 int BN_mod_sqr(BIGNUM *r, const BIGNUM *a, const BIGNUM *m, BN_CTX *ctx);
 int BN_mod_add(BIGNUM *r, const BIGNUM *a, const BIGNUM *b, const BIGNUM *m, BN_CTX *ctx);
 int BN_mod_sub(BIGNUM *r, const BIGNUM *a, const BIGNUM *b, const BIGNUM *m, BN_CTX *ctx);
-int BN_rand_range(BIGNUM *r, BIGNUM *range);
-int BN_pseudo_rand_range(BIGNUM *r, BIGNUM *range);
+//int BN_rand_range(BIGNUM *r, BIGNUM *range);
+//int BN_pseudo_rand_range(BIGNUM *r, BIGNUM *range);
 char *CONF_get1_default_config_file(void);
 int PEM_def_callback(char *buf, int num, int w, void *key);
 
