diff -urN dbus-1.2.4.6permissive/configure dbus-1.2.4.6permissive.new/configure
--- dbus-1.2.4.6permissive/configure	2009-05-06 19:35:38.000000000 +0200
+++ dbus-1.2.4.6permissive.new/configure	2010-03-26 17:40:13.000000000 +0100
@@ -20958,7 +20958,7 @@
   esac
   case " $CFLAGS " in
   *[\ \	]-Wno-pointer-sign[\ \	]*) ;;
-  *) CFLAGS="$CFLAGS -Wno-pointer-sign" ;;
+  *) CFLAGS="$CFLAGS" ;;
   esac
 
   # http://bugs.freedesktop.org/show_bug.cgi?id=19195
