--- expat-1.95.8/conftools/config.sub	2008-03-18 21:54:49.000000000 +0100
+++ expat-1.95.8/conftools/config.sub	2008-03-18 21:55:32.000000000 +0100
@@ -251,7 +251,7 @@
 	| a29k \
 	| alpha | alphaev[4-8] | alphaev56 | alphaev6[78] | alphapca5[67] \
 	| alpha64 | alpha64ev[4-8] | alpha64ev56 | alpha64ev6[78] | alpha64pca5[67] \
-	| arc | arm | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr \
+	| arc | arm | arm[bl]e | arme[lb] | armv[2345] | armv[345][lb] | avr | avr32 \
 	| c4x | clipper \
 	| d10v | d30v | dsp16xx \
 	| fr30 \
