--- a/src/rrdtool.c
+++ b/src/rrdtool.c
@@ -87,6 +87,7 @@ static const char *config_keys[] =
 	"HeartBeat",
 	"RRARows",
 	"RRATimespan",
+	"RRASingle",
 	"XFF"
 };
 static int config_keys_num = STATIC_ARRAY_SIZE (config_keys);
@@ -1077,6 +1078,14 @@ static int rrd_config (const char *key, 
 		}
 		xff = tmp;
 	}
+	else if (strcasecmp ("RRASingle", key) == 0)
+	{
+		if (strcasecmp("true", value) == 0)
+		{
+			rra_types_num = 1;
+			NOTICE ("rrdtool plugin: RRASingle = true: creating only AVERAGE RRAs");
+		}
+	}
 	else
 	{
 		return (-1);
