diff -urN xapian-core-1.0.7/common/serialise-double.cc xapian-core-1.0.7.new/common/serialise-double.cc
--- xapian-core-1.0.7/common/serialise-double.cc	2008-07-16 06:42:30.000000000 +0200
+++ xapian-core-1.0.7.new/common/serialise-double.cc	2010-03-29 10:07:02.000000000 +0200
@@ -56,7 +56,7 @@
 # define MAX_MANTISSA_BYTES ((DBL_MANT_DIG + 1 + 1) / 2)
 # define MAX_EXP ((DBL_MAX_EXP + 1) / 2)
 # define MAX_MANTISSA (1 << ((DBL_MAX_EXP & 1) * 4))
-#elif
+#else
 # error FLT_RADIX is a value not currently handled (not 2 or 16)
 // # define MAX_MANTISSA_BYTES (sizeof(double) + 1)
 #endif
