Index: openl2tp-1.7/plugins/ppp_unix.c
===================================================================
--- openl2tp-1.7.orig/plugins/ppp_unix.c	2010-11-01 15:56:52.150001215 +0100
+++ openl2tp-1.7/plugins/ppp_unix.c	2010-11-01 15:59:28.011000001 +0100
@@ -643,6 +643,8 @@
 
 		if (ppp->ppp_profile_data->proxy_arp) {
 			argv[arg++] = "proxyarp";
+		} else {
+			argv[arg++] = "noproxyarp";
 		}
 
 		argv[arg++] = "plugin";
