--- a/build/config/libcurl.m4
+++ b/build/config/libcurl.m4
@@ -71,7 +71,7 @@ AC_DEFUN([LIBCURL_CHECK_CONFIG],
      if test -d "$_libcurl_with" ; then
         LIBCURL_CPPFLAGS="-I$withval/include"
         _libcurl_ldflags="-L$withval/lib"
-        AC_PATH_PROG([_libcurl_config],["$withval/bin/curl-config"])
+        _libcurl_config="$withval/bin/curl-config"
      else
 	AC_PATH_PROG([_libcurl_config],[curl-config])
      fi
