--- a/source/popt/popt.c
+++ b/source/popt/popt.c
@@ -10,13 +10,14 @@
 
 #include "system.h"
 
-#if HAVE_FLOAT_H
+//#if HAVE_FLOAT_H
 #include <float.h>
-#endif
+//#endif
 #include <math.h>
 
 #include "findme.h"
 #include "poptint.h"
+#include "../include/config.h"
 
 #ifdef	MYDEBUG
 /*@unchecked@*/
@@ -388,7 +389,7 @@ static int execCommand(poptContext con)
 	sprintf(s, "%s/%s", con->execPath, item->argv[0]);
 	argv[argc] = s;
     } else {
-	argv[argc] = findProgramPath(item->argv[0]);
+	argv[argc] = ""/*findProgramPath(item->argv[0])*/;
     }
     if (argv[argc++] == NULL) return POPT_ERROR_NOARG;
 
@@ -1246,4 +1247,3 @@ int poptStrippedArgv(poptContext con, in
     
     return numargs;
 }
-/*@=boundswrite@*/
--- a/source/modules/vfs_default.c
+++ b/source/modules/vfs_default.c
@@ -977,7 +977,7 @@ static int vfswrap_linux_setlease(vfs_ha
 
 	START_PROFILE(syscall_linux_setlease);
 
-#ifdef LINUX
+#if defined(LINUX) && defined(HAVE_KERNEL_OPLOCKS_LINUX)
 	/* first set the signal handler */
 	if(linux_set_lease_sighandler(fd) == -1)
 		return -1;
