--- a/ncat/Makefile.in
+++ b/ncat/Makefile.in
@@ -136,7 +136,7 @@ $(NSOCKDIR)/libnsock.a: $(NSOCKDIR)/Make
 install: $(TARGET)
 	@echo Installing Ncat;
 	$(SHTOOL) mkdir -f -p -m 755 $(DESTDIR)$(bindir) $(DESTDIR)$(pkgdatadir) $(DESTDIR)$(mandir)/man1
-	$(INSTALL) -c -m 755 -s ncat $(DESTDIR)$(bindir)/ncat
+	$(INSTALL) -c -m 755 ncat $(DESTDIR)$(bindir)/ncat
 	if [ -n "$(DATAFILES)" ]; then \
 		$(INSTALL) -c -m 644 $(DATAFILES) $(DESTDIR)$(pkgdatadir)/; \
 	fi
