Index: motion-3.2.9/Makefile.in
===================================================================
--- motion-3.2.9.orig/Makefile.in
+++ motion-3.2.9/Makefile.in
@@ -33,6 +33,7 @@ examplesdir = $(docdir)/examples
 ################################################################################
 CFLAGS       = @CFLAGS@ -Wall -DVERSION=\"$(VERSION)\" -D_REENTRANT \
 	       -Dsysconfdir=\"$(sysconfdir)\"
+CPPFLAGS     = @CPPFLAGS@
 LDFLAGS      = @LDFLAGS@
 LIBS         = @LIBS@ 
 VIDEO_OBJ    = @VIDEO@
@@ -120,7 +121,7 @@ endif
 ################################################################################
 $(DEPEND_FILE): *.h $(SRC)
 	@echo "Generating dependencies, please wait..."
-	@$(CC) $(CFLAGS) -M $(SRC) > .tmp
+	@$(CC) $(CFLAGS) $(CPPFLAGS) -M $(SRC) > .tmp
 	@mv -f .tmp $(DEPEND_FILE)
 	@echo
 
