Index: motion-3.2.7/Makefile.in
===================================================================
--- motion-3.2.7.orig/Makefile.in	2007-06-11 08:26:33.000000000 +0100
+++ motion-3.2.7/Makefile.in	2007-06-11 08:27:55.000000000 +0100
@@ -32,6 +32,7 @@
 ################################################################################
 CFLAGS       = @CFLAGS@ -Wall -DVERSION=\"$(VERSION)\" -D_REENTRANT \
 	       -Dsysconfdir=\"$(sysconfdir)\"
+CPPFLAGS     = @CPPFLAGS@
 LDFLAGS      = @LDFLAGS@
 LIBS         = @LIBS@ 
 VIDEO_OBJ    = @VIDEO@
@@ -119,7 +120,7 @@
 ################################################################################
 $(DEPEND_FILE): *.h $(SRC)
 	@echo "Generating dependencies, please wait..."
-	@$(CC) $(CFLAGS) -M $(SRC) > .tmp
+	@$(CC) $(CFLAGS) $(CPPFLAGS) -M $(SRC) > .tmp
 	@mv -f .tmp $(DEPEND_FILE)
 	@echo
 
