--- mrtg-2.17.3.orig/configure.in	2011-12-25 23:54:18.284584003 +0100
+++ mrtg-2.17.3/configure.in	2011-12-25 23:54:52.377260252 +0100
@@ -19,47 +19,6 @@
 
 AC_PATH_PROGS(NROFF, groff nroff)
 
-dnl Check for long long type (64 bit rateup)
-AC_CHECK_HEADERS(inttypes.h)
-AC_CHECK_TYPE(unsigned long long)
-AC_CHECK_TYPE(long long)
-AC_CHECK_FUNCS(strtoll)
-
-# Taken from libIDL-0.8.2
-AC_MSG_CHECKING([for printf long long format specifier])
-AC_CACHE_VAL(long_long_format_specifier,[
-    for format in ll l q I64; do
-        AC_TRY_RUN([#include <stdio.h>  
-            int main()
-            {
-                long long b, a = -0x3AFAFAFAFAFAFAFALL;
-                char buffer[1000];
-                sprintf (buffer, "%${format}u", a);
-                sscanf (buffer, "%${format}u", &b);
-                exit (b!=a);
-            }
-            ],
-            long_long_format_specifier="%${format}d"
-            long_long_format="${format}d"
-            break)
-        done])
-AC_MSG_RESULT($long_long_format_specifier)
-AC_DEFINE_UNQUOTED(LLD, "$long_long_format_specifier",
-    [How to print a long long])
-AC_DEFINE_UNQUOTED(LLD_FORMAT, "$long_long_format",
-    [long long format without % specifier])
-
-AH_BOTTOM(
-#ifndef HAVE_STRTOLL     
-long long int strtoll(const char *str, char **ptr, int base)
-{
-        long long int ll;
-        sscanf(str, LLD, &ll);
-        return ll;
-}
-#endif
-)
-
 dnl Checks for libraries.
 AC_CHECK_LIB(m, pow, [ MATHLIBS="-lm" ])
 
--- mrtg-2.17.3.orig/configure	2011-12-26 00:02:43.178539074 +0100
+++ mrtg-2.17.3/configure	2011-12-26 00:04:09.056241468 +0100
@@ -3609,60 +3609,6 @@
 done
 
 
-# Taken from libIDL-0.8.2
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for printf long long format specifier" >&5
-$as_echo_n "checking for printf long long format specifier... " >&6; }
-if test "${long_long_format_specifier+set}" = set; then :
-  $as_echo_n "(cached) " >&6
-else
-
-    for format in ll l q I64; do
-        if test "$cross_compiling" = yes; then :
-  { { $as_echo "$as_me:${as_lineno-$LINENO}: error: in \`$ac_pwd':" >&5
-$as_echo "$as_me: error: in \`$ac_pwd':" >&2;}
-as_fn_error $? "cannot run test program while cross compiling
-See \`config.log' for more details" "$LINENO" 5 ; }
-else
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <stdio.h>
-            int main()
-            {
-                long long b, a = -0x3AFAFAFAFAFAFAFALL;
-                char buffer[1000];
-                sprintf (buffer, "%${format}u", a);
-                sscanf (buffer, "%${format}u", &b);
-                exit (b!=a);
-            }
-
-_ACEOF
-if ac_fn_c_try_run "$LINENO"; then :
-  long_long_format_specifier="%${format}d"
-            long_long_format="${format}d"
-            break
-fi
-rm -f core *.core core.conftest.* gmon.out bb.out conftest$ac_exeext \
-  conftest.$ac_objext conftest.beam conftest.$ac_ext
-fi
-
-        done
-fi
-
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $long_long_format_specifier" >&5
-$as_echo "$long_long_format_specifier" >&6; }
-
-cat >>confdefs.h <<_ACEOF
-#define LLD "$long_long_format_specifier"
-_ACEOF
-
-
-cat >>confdefs.h <<_ACEOF
-#define LLD_FORMAT "$long_long_format"
-_ACEOF
-
-
-
-
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for pow in -lm" >&5
 $as_echo_n "checking for pow in -lm... " >&6; }
 if test "${ac_cv_lib_m_pow+set}" = set; then :
