Index: lcd4linux-0.10.0+cvs20051015.orig/drv_RouterBoard.c
===================================================================
--- lcd4linux-0.10.0+cvs20051015.orig.orig/drv_RouterBoard.c	2007-07-07 14:02:56.000000000 +0200
+++ lcd4linux-0.10.0+cvs20051015.orig/drv_RouterBoard.c	2007-07-07 14:02:56.000000000 +0200
@@ -120,7 +120,7 @@
 #include <string.h>
 #include <errno.h>
 #include <unistd.h>
-#include <sys/io.h>
+#include <asm/io.h>
 
 #include "debug.h"
 #include "cfg.h"
