Index: lcd4linux-0.10.0+cvs20051015.orig/drv_generic_parport.c
===================================================================
--- lcd4linux-0.10.0+cvs20051015.orig.orig/drv_generic_parport.c	2007-07-07 14:02:55.000000000 +0200
+++ lcd4linux-0.10.0+cvs20051015.orig/drv_generic_parport.c	2007-07-07 14:02:57.000000000 +0200
@@ -117,16 +117,6 @@
 #include <sys/stat.h>
 #include <sys/ioctl.h>
 
-#ifdef HAVE_SYS_IO_H
-#include <sys/io.h>
-#define WITH_OUTB
-#else
-#ifdef HAVE_ASM_IO_H
-#include <asm/io.h>
-#define WITH_OUTB
-#endif
-#endif
-
 #if defined (HAVE_LINUX_PARPORT_H) && defined (HAVE_LINUX_PPDEV_H)
 #define WITH_PPDEV
 #include <linux/parport.h>
@@ -143,6 +133,11 @@
 #define PARPORT_STATUS_BUSY       0x80
 #endif
 
+#ifndef WITH_OUTB
+#define inb(foo) 0
+#define outb(foo,bar) 0
+#endif
+ 
 #if !defined(WITH_OUTB) && !defined(WITH_PPDEV)
 #error neither outb() nor ppdev() possible
 #error cannot compile parallel port driver
