--- a/src/configure
+++ b/src/configure
@@ -3610,7 +3610,7 @@ _ACEOF
 cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
-#include<sys/socket.h>
+#include <sys/socket.h>
 
 #include <$ac_header>
 _ACEOF
@@ -3669,11 +3669,11 @@ cat confdefs.h >>conftest.$ac_ext
 cat >>conftest.$ac_ext <<_ACEOF
 /* end confdefs.h.  */
 
-#include<sys/socket.h>
-#include<net/ethernet.h>
-#include<linux/if.h>
-#include<linux/in.h>
-
+#include <sys/socket.h>
+#include <net/ethernet.h>
+#include <linux/if.h>
+#include <linux/in.h>
+#include <linux/in6.h>
 
 #include <$ac_header>
 _ACEOF
@@ -6272,7 +6272,7 @@ esac
 { echo "$as_me:$LINENO: checking packing order of bit fields" >&5
 echo $ECHO_N "checking packing order of bit fields... $ECHO_C" >&6; }
 if test "$cross_compiling" = yes; then
-  $ECHO "no defaults for cross-compiling"; exit 0
+  rpppoe_cv_pack_bitfields="$ac_cv_pack_bitfields"
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
