diff -ur libexif.old/Makefile.in libexif.dev/Makefile.in
--- libexif.old/Makefile.in	2005-12-27 22:18:36.000000000 +0100
+++ libexif.dev/Makefile.in	2006-09-29 23:12:39.000000000 +0200
@@ -213,7 +213,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = m4m po libexif test doc binary
+SUBDIRS = m4m po libexif test binary
 @SHIP_BINARIES_TRUE@README_W32_XDIST = README-Win32.txt
 EXTRA_DIST = @PACKAGE_TARNAME@.spec autogen.sh $(README_W32_XDIST)
 ACLOCAL_AMFLAGS = -I m4m
