diff -ruN python-ecore-r37637.orig/setup.py python-ecore-r37637/setup.py
--- python-ecore-r37637.orig/setup.py	2009-01-04 12:06:41.000000000 +0100
+++ python-ecore-r37637/setup.py	2009-01-04 12:09:58.000000000 +0100
@@ -1,10 +1,7 @@
 import sys
 import os
 
-from ez_setup import use_setuptools
-use_setuptools('0.6c3')
-
-from setuptools import setup, find_packages, Extension
+from distutils.core import setup, Extension
 import subprocess
 import shlex
 
@@ -143,7 +140,7 @@
       long_description=long_description,
       keywords='wrapper binding enlightenment abstraction event ecore',
       classifiers=trove_classifiers,
-      packages=find_packages(),
+      packages=['ecore', 'ecore.x', 'ecore.evas', 'ecore.imf'],
       install_requires=['python-evas>=0.3.1'],
       setup_requires=['python-evas>=0.3.1'],
       ext_modules=[ecoremodule, ecoreevasmodule, ecorexmodule,
